---
output:
  rmarkdown::html_document:
    highlight: pygments
    toc: true
    css: style.css
chunk_output_type: console
header-includes:
- \pagenumbering{gobble}
params:
    set_title: "Figure-7"
title: "`r params$set_title`"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE, autodep = knitr::dep_prev())
```

***

\pagebreak

# Fig 7A to Figure 7D

This data was analyzed in python using scanpy. The corresponding html files for each panel is available separately.

***

\pagebreak

# Fig 7E

To reproduce the Figures, each oncogenic dataset was recovered from the files listed in our article then, using a standardized analysis workflow based on MAESTRO v1.4.0, all the collected datasets were processed for including quality control, batch effect removal, cell clustering, differential expression analysis, cell-type annotation, malignant cell classification and gene set enrichment analysis that internally uses the Seurat package.

```{r, echo=T,eval=F}
#####Generating standarized RData file for each single cell dataset
####THe following code is an example for one of the 41 oncogenic datasets used in our article
library(Seurat)
library(ggplot2)
library(ggpubr)
library(Rmagic)
library(MAESTRO)
###Setwd to where you downloaded the data set, 
setwd("TISCH_scRNA_datasets/CRC_GSE146771/")

metadata <- read.delim("CRC_GSE146771_Smartseq2_CellMetainfo_table.tsv")

CRC_GSE146771.counts <- Read10X_h5(filename = "CRC_GSE146771_Smartseq2_expression.h5")

CRC_GSE146771.res <- RNARunSeurat(inputMat = CRC_GSE146771.counts,
                                  project = "CRC_GSE146771",
                                  orig.ident = NULL,
                                  min.c = 3,
                                  min.g = 200,
                                  mito = TRUE,
                                  mito.cutoff = 0.05,
                                  variable.genes = 2000,
                                  organism = "GRCh38",
                                  dims.use = 1:15,
                                  cluster.res = 0.6,
                                  only.pos = FALSE,
                                  genes.test.use = "presto",
                                  genes.cutoff = 1e-05,
                                  genes.pct = 0.1,
                                  genes.logfc = 0.25)

# subset cells metadata and in scRNA dataset
metadata2 <- subset(metadata, metadata$Cell %in% names(CRC_GSE146771.res$RNA$orig.ident))

CRC_GSE146771.res$RNA$lineage <- metadata2$Celltype..major.lineage.

# prepare DotPlot genes
Idents(CRC_GSE146771.res$RNA) <- "lineage"


AID_lineage <- DotPlot(CRC_GSE146771.res$RNA, features = c("AICDA", "UNG", "MSH2", "MSH6", "NHEJ1", "TP53BP1")) + RotatedAxis()
AID_lineage_data_CRC_GSE146771 <- ggplot_build(AID_lineage)$plot$data

AID_lineage_table_CRC_GSE146771 <- table(CRC_GSE146771.res$RNA$lineage)

s.genes <- cc.genes.updated.2019$s.genes
g2m.genes <- cc.genes.updated.2019$g2m.genes

CRC_GSE146771.res$RNA <- ScaleData(CRC_GSE146771.res$RNA, features = c(s.genes, g2m.genes))
CRC_GSE146771.res$RNA <- RunPCA(CRC_GSE146771.res$RNA, features = c(s.genes, g2m.genes))
DimPlot(CRC_GSE146771.res$RNA)

CRC_GSE146771.res$RNA <- CellCycleScoring(CRC_GSE146771.res$RNA, 
                                          s.features = s.genes, 
                                          g2m.features = g2m.genes, 
                                          set.ident = TRUE)
AID_cc <- DotPlot(CRC_GSE146771.res$RNA, 
                  features = c("AICDA", "UNG", "MSH2", "MSH6", "NHEJ1", "TP53BP1")) + 
  RotatedAxis()
AID_cc_data_CRC_GSE146771 <- ggplot_build(AID_cc)$plot$data

#save(CRC_GSE146771.res,file =  "CRC_GSE146771_res.RData")
save(AID_cc_data_CRC_GSE146771, 
     AID_lineage_data_CRC_GSE146771, 
     AID_lineage_table_CRC_GSE146771, 
     file =  "CRC_GSE146771_all_figure.RData") ##This is the Final RData needed for subsequent analyses




###The final RData contains:
#1: Data frame with the expression of AICDA, MSH2, MSH6, etc accounting for the cell cycle stages (G1, S, G2/M)
#2: Data frame with the expression of AICDA, MSH2, MSH6, and others per cell type within the study
#3:Data frame with the number of cells per cell type within the study


#####Repeat for each of the Datasets....
#Data sets were named as:
#[1] "ALL_GSE132509_data_figure.RData"                             "AML_GSE116256_data_figure.RData"                            
#[3] "BCC_GSE123813_aPD1_data_figure.RData"                        "BRCA_GSE114727_non_malignant_data_figure.RData"             
#[5] "BRCA_GSE136206_mouse_aPD1aCTLA4_data_figure.RData"           "BRCA_GSE138536_non_malignant_data_figure.RData"             
#[7] "CHOL_GSE125449_aPD1aPDL1aCTLA4_data_figure.RData"            "CLL_GSE111014_non_malignant_data_figure.RData"              
#[9] "CRC_GSE108989_non_malignant_data_figure.RData"               "CRC_GSE136394_non_malignant_data_immunotherapy_figure.RData"
#[11] "CRC_GSE139555_non_malignant_data_immunotherapy_figure.RData" "CRC_GSE146771_10X_non_malignant_figure.RData"               
#[13] "CRC_GSE146771_all_figure.RData"                              "DLBCL_res_all_figure.RData"                                 
#[15] "Glioma_GSE135437_all_figure.RData"                           "Glioma_GSE148842_all_figure.RData"                          
#[17] "Glioma_GSE70630_all_figure.RData"                            "Glioma_GSE84465_all_figure.RData"                           
#[19] "Glioma_GSE89567_all_figure.RData"                            "HNSC_GSE103322_all_figure.RData"                            
#[21] "HNSC_GSE139324_immune_figure.RData"                          "KIRC_GSE111360_immune_figure.RData"                         
#[23] "KIRC_GSE139555_immune_figure.RData"                          "LIHC_GSE125449_aPDL1aCTLA4_all_figure.RData"                
#[25] "LIHC_GSE140228_immune_10X_figure.RData"                      "LIHC_GSE140228_immune_Smart_figure.RData"                   
#[27] "LIHC_GSE98638_immune_CD3T_figure.RData"                      "MB_GSE119926_all_figure.RData"                              
#[29] "NHL_GSE128531.res_immune_figure.RData"                       "NSCLC_EMTAB6149.res_all_figure.RData"                       
#[31] "NSCLC_GSE127465.res_all_figure.RData"                        "NSCLC_GSE131907.res_immune_figure.RData"                    
#[33] "NSCLC_GSE139555.res_immune_figure.RData"                     "NSCLC_GSE99254.res_Tcell_figure.RData"                      
#[35] "OSCC_GSE117872_immune_figure.RData"                          "PAAD_CRA001160.res_all_figure.RData"                        
#[37] "SKCM_GSE115978_aPD1.res_all_figure.RData"                    "SKCM_GSE120575_aPD1aCTLA4.res_immune_figure.RData"          
#[39] "SKCM_GSE123139.res_immune_figure.RData"                      "SKCM_GSE148190.res_immune_figure.RData"                     
#[41] "SKCM_GSE72056.res_immune_figure.RData"

```


```{r panel7E_cals,echo=T,eval=F,fig.show="hide"}
###
##Import data
data_dir<-"/media/user/seagate_ICM/AICDA_analysis/Github_repository/" #Set to where you downloaded the data
setwd(data_dir)
source("r/prerequisites.R")

#
#Import datasets...
list_of_files <- list.files(path = "/media/user/seagate_ICM/AICDA_analysis/Single_cellStudies/RData_rawobjects", recursive = TRUE, pattern = "*.RData", full.names = TRUE)
tmp_names<-gsub("/media/user/seagate_ICM/AICDA_analysis/Single_cellStudies/RData_rawobjects/","",list_of_files[])
tmp_names<-gsub('_([^_]*)$', '',gsub('_([^_]*)$', '', tmp_names[]))
tmp_names<-gsub("_data","",tmp_names)

single_cell_design<-data.frame(Dataset=tmp_names,Tumor_type=gsub("_.*","",tmp_names),organism=c(rep("Human",4),"Mouse",rep("Human",36)),Treatment=c(rep("None",2),"aPD1",rep("None",1),"aPD1-aCTLA4",rep("None",1),"aPD1-aPDL1-aCTLA4","None","None","Immunotherapy","Immunotherapy",rep("None",12),"aPDL1-aCTLA4",rep("None",12),"aPD1","aPD1-aCTLA4",rep("None",3)),Sorting=c(rep("All",3),rep("Non-tumoral",1),rep("All",1),rep("Non-tumoral",1),"All",rep("Non-tumoral",5),rep("All",8),rep("Immune",3),rep("All",1),rep("Immune",3),rep("All",1), "Immune","All","All", "Immune","Immune", "T-cell","Immune",rep("All",2),rep("Immune",3),"All"))
###
single_cell_design$Dataset[25:26]<-c("LIHC_GSE140228_immune10X","LIHC_GSE140228_immune_Smart")
####Do design table with: name dataset, organism, treatment, sorting, tumor type


###Import data sets by importing into a temporary env..
###
ss_cellnumbers<-list()
ss_cellAICDA<-list()
ss_cellcycle<-list()

for (i in 1:length(list_of_files)) {
  tmp_env <- new.env() #Create tmp_env object
load(list_of_files[i], tmp_env)
ss_cellnumbers[[i]]<-get(ls(tmp_env)[3], envir=tmp_env) 
tmp<-get(ls(tmp_env)[2], envir=tmp_env) 
ss_cellAICDA[[i]]<-tmp[tmp$features.plot=="AICDA",]
if (i==5) {
  ss_cellAICDA[[i]]<-tmp[tmp$features.plot=="Aicda",] #For the mouse dataset
}
ss_cellcycle[[i]]<-get(ls(tmp_env)[1], envir=tmp_env) 
}
names(ss_cellnumbers)<-single_cell_design$Dataset
names(ss_cellAICDA)<-single_cell_design$Dataset
names(ss_cellcycle)<-single_cell_design$Dataset

####Get data frame of number of cells per study...
single_cell_design$N_cells<-sapply(ss_cellnumbers, sum)

####Get data for AICDA expression graph
ss_cellcycle2<-as.data.frame(rbindlist(map(ss_cellcycle, as.data.table), fill = TRUE, idcol = T))
##Change names of mouse genes to upper letters have same cathegories
ss_cellcycle2$features.plot<-toupper(ss_cellcycle2$features.plot)
##Eliminate CRC_GSE139555 cause is not well annotated
ss_cellcycle2<-ss_cellcycle2[ss_cellcycle2$.id%nin%"CRC_GSE139555_non_malignant",]
##Re-order before dcast according to cell cycle G1, S, G2/M
ss_cellcycle2$id<-factor(ss_cellcycle2$id,levels = c("G1","S","G2M"))

###Reduce using dcast
ss_cellcycle2<-reshape2::dcast(data = ss_cellcycle2, formula = features.plot ~id+.id , value.var = "avg.exp",fill = 0,fun.aggregate = sum,drop = T)

###Expression values: The values in the single-cell level expression matrix are normalized. We employed the global-scaling normalization method ('NormalizeData' function) in Seurat to scale the raw counts (UMI) in each cell to 10,000, and then log-transformed the results.

##For heatmaps.. 
#we converted raw count or FPKM, which depends on the available data, to TPM to ensure the expression level is relatively comparable between different datasets. The expression of a gene in the cell was quantified as log2(TPM+1).
rownames(ss_cellcycle2)<-ss_cellcycle2$features.plot
ss_cellcycle2<-ss_cellcycle2[,-1]

ss_cellcycle2_norm<-log2(ss_cellcycle2+1)
ss_cellcycle2_norm<-as.data.frame(ss_cellcycle2_norm)
ss_cellcycle2_norm<-na.omit(ss_cellcycle2_norm)
##Re-order rows
ss_cellcycle2_norm<-ss_cellcycle2_norm[order(factor(rownames(ss_cellcycle2_norm), levels = c("AICDA","UNG","MSH2","MSH6","NHEJ1","TP53BP1","TRP53BP1"))),]

#Re-order columns
single_cell_design2<-single_cell_design[single_cell_design$Dataset%nin%"CRC_GSE139555_non_malignant",]
single_cell_design2<-single_cell_design2[order(factor(single_cell_design2$Tumor_type, levels = c("DLBCL","SKCM","Glioma","HNSC","BRCA","MB","PAAD","CRC","NSCLC","CHOL","CLL","LIHC","ALL","AML","KIRC","NHL","BCC","OSCC"))),]
single_cell_design2$Tumor_type<-factor(single_cell_design2$Tumor_type, levels = c("DLBCL","SKCM","Glioma","HNSC","BRCA","MB","PAAD","CRC","NSCLC","CHOL","CLL","LIHC","ALL","AML","KIRC","NHL","BCC","OSCC"))


ss_cellcycle2_norm<-ss_cellcycle2_norm[,c(paste0("G1_",single_cell_design2$Dataset),paste0("S_",single_cell_design2$Dataset),paste0("G2M_",single_cell_design2$Dataset))]


##Add expression value of S cycle and order according to that
single_cell_design2$exp<-t(ss_cellcycle2_norm[1,1:40])
##Reorder
single_cell_design2<-single_cell_design2 %>% arrange((Tumor_type),desc(exp) )

##
ss_cellcycle2_norm<-ss_cellcycle2_norm[,c(paste0("G1_",single_cell_design2$Dataset),paste0("S_",single_cell_design2$Dataset),paste0("G2M_",single_cell_design2$Dataset))]


###

col_fun3 = colorRamp2(c(1.8,.5,.1,0.001,0), c("firebrick",scales::alpha("darkorange",.8),"yellow2","grey","ivory"))

#Quick visualization

Heatmap(ss_cellcycle2_norm[,1:40],col = col_fun3,name = "Log2(TPM+1)",column_names_side = "top", row_names_gp = gpar(fontsize=10),column_names_rot = 45,row_names_side = "left",show_column_dend = T,show_row_dend =T, cluster_columns = F,cluster_rows = F,column_title_gp = gpar(fontsize = 18, fontface = "bold",col="black"),column_names_gp = gpar(fontsize=10),border = TRUE,show_column_names = T, heatmap_legend_param = list(title= "Log2(TPM+1)", at = rev(c(1.8,.5,.1,0.001,0)), labels = rev(c(1.8,.5,.1,0.001,0))),column_gap = unit(2, "mm"),right_annotation = rowAnnotation(foo = anno_text(rownames(ss_cellcycle2_norm), location = 0, rot = 0, just = "left", gp = gpar(fontsize = 10))))


```

```{r panel_7E, fig.width=16,fig.height=12,fig.show=T}
##Import data
data_dir<-"/media/user/seagate_ICM/AICDA_analysis/Github_repository/" #Set to where you downloaded the data
setwd(data_dir)
source("r/prerequisites.R")
#Import needed data
ss_cellcycle2_norm<-as.data.frame(read.delim("Data/Figure7/Figure7E_1.tab"))
single_cell_design2<-as.data.frame(read.delim("Data/Figure7/Figure7E_2.tab"))
#

####plot
##Split each 40
split_n = c(rep("G1",40),rep("S",40),rep("G2M",40))
split_n = factor(split_n, levels = unique(split_n))
ss_cellcycle2_norm2<-t(ss_cellcycle2_norm)

##Annotations..
col_tumor_type = c("ALL"="brown4","AML"="darkorange4","BCC"="gray","BRCA"="hotpink","CHOL"="lightsalmon","CLL"="plum3","CRC"="slategray1","DLBCL"="firebrick","Glioma"="dodgerblue","HNSC"="yellowgreen","KIRC"="wheat1","LIHC"="sienna3","MB"="white","NHL"="maroon4","NSCLC"="coral1","OSCC"="aliceblue","PAAD"="black","SKCM"="rosybrown1")
tmp_tumortype<-data.frame(tumor=rep(single_cell_design2$Tumor_type,3),row.names = rownames(ss_cellcycle2_norm2))
cell_cycle = data.frame(phase = factor(c(rep("G1",40), rep("S",40), rep("G2M",40)), levels = c("G1", "S", "G2M")),row.names = rownames(ss_cellcycle2_norm2))
color = c("#66C2A5", "#FC8D62", "#8DA0CB")


####Annotations first
circos.clear()
circlize_plot = function() {
circos.par(start.degree = 140, gap.degree = 10,gap.after = c(10,10,30))
##For tumor type
circos.heatmap(tmp_tumortype, col = col_tumor_type, track.height = 0.01,split = split_n,cluster = F,rownames.side = "outside",show.sector.labels = TRUE,rownames.cex = .3)
##For organism
col_org = c("Human" = "gray16", "Mouse" = "gray91")
tmp_org<-data.frame(tumor=rep(single_cell_design2$organism,3),row.names = rownames(ss_cellcycle2_norm2))


circos.heatmap(tmp_org[,], col = col_org,split = split_n,cluster = F, track.height = 0.01, show.sector.labels = F)
#For sorting
col_sor = c("All"="steelblue","Non-tumoral"="cyan4","Immune"="lightcoral","T-cell"="lightgreen")
tmp_sor<-data.frame(tumor=rep(single_cell_design2$Sorting,3),row.names = rownames(ss_cellcycle2_norm2))


circos.heatmap(tmp_sor[,], col = col_sor,split = split_n,cluster = F, track.height = 0.01, show.sector.labels = F)

#For treatment
col_tx = c("None" = "white", "aPD1-aCTLA4" = "orange2","aPD1-aPDL1-aCTLA4"="blueviolet","Immunotherapy"="green1","aPDL1-aCTLA4"="darkred","aPD1"="darksalmon")
tmp_tx<-data.frame(tumor=rep(single_cell_design2$Treatment,3),row.names = rownames(ss_cellcycle2_norm2))


circos.heatmap(tmp_tx[,], col = col_tx,split = split_n,cluster = F, track.height = 0.01, show.sector.labels = F)

##Heatmap
circos.heatmap(ss_cellcycle2_norm2[,], col = col_fun3,split = split_n,cluster = F, track.height = 0.2, bg.border = "black",bg.lwd = 2, bg.lty = 2, show.sector.labels = F)

#Add gene names
circos.track(track.index = get.current.track.index(), panel.fun = function(x, y) {
    if(CELL_META$sector.numeric.index == 3) { # the last sector
        cn = colnames(ss_cellcycle2_norm2)
        n = length(cn)
        circos.text(rep(CELL_META$cell.xlim[2], n) +rev(c(.3,.2,.2,.1,.6,.7,.2))+ convert_x(5, "mm"), 
                    (1:n - .5)*1, rev(cn), adj =par("adj"),font = 2,col = rev(c("red",rep("black",6))),
                    cex = 0.45, facing = "inside")
    }
}, bg.border = NA)

circos.track(ylim = c(0, 1), panel.fun = function(x, y) {
    circos.arrow(CELL_META$xlim[1], CELL_META$xlim[2], 
        arrow.head.width = CELL_META$yrange*0.8, arrow.head.length = cm_x(0.5),
        col = color[CELL_META$sector.numeric.index])
    circos.text(CELL_META$xcenter, CELL_META$ycenter, CELL_META$sector.index, font = 2,cex=.8,
        facing = "bending.inside")
    
}, bg.border = NA, track.height = 0.075)

circos.clear()

}

##Cell cycle
##Now the legends
col_org = c("Human" = "gray16", "Mouse" = "gray91")
col_sor = c("All"="steelblue","Non-tumoral"="cyan4","Immune"="lightcoral","T-cell"="lightgreen")
col_tx = c("None" = "white", "aPD1-aCTLA4" = "orange2","aPD1-aPDL1-aCTLA4"="blueviolet","Immunotherapy"="green1","aPDL1-aCTLA4"="darkred","aPD1"="darksalmon")
lg_circle<-packLegend(Legend(title = "Log2(TPM+1)",col_fun = col_fun3,at = rev(c(1.8,.5,.1,0.001,0)), labels = rev(c(1.8,.5,.1,0.001,0))),Legend(title = "Tumor type",legend_gp = gpar(fill = col_tumor_type),labels = names(col_tumor_type),ncol=3),Legend(title = "Organism",legend_gp = gpar(fill = col_org),labels = names(col_org),ncol=1),Legend(title = "Sorting",legend_gp = gpar(fill = col_sor),labels = names(col_sor),ncol=1),Legend(title = "Treatment",legend_gp = gpar(fill = col_tx),labels = names(col_tx),ncol=1),max_height = unit(10, "cm"),column_gap= unit(1, "cm"))

##
library(gridBase)
plot.new()
circle_size = unit(1, "snpc") # snpc unit gives you a square region

pushViewport(viewport(x = 0, y = 0.5, width = circle_size, height = circle_size,
    just = c("left", "center")))
par(omi = gridOMI(), new = TRUE)
circlize_plot()
upViewport()


draw(lg_circle, x = unit(.925, "snpc"), just = c("left"))

###



```

***

\pagebreak


# Fig 7F


```{r panel7F_cals,echo=T,eval=F,fig.show="hide"}
###This part of the code can be executed once all single cell datasets (in RData format) were obtained from the previous section:
##Import data
data_dir<-"/media/user/seagate_ICM/AICDA_analysis/Github_repository/" #Set to where you downloaded the data
setwd(data_dir)
source("r/prerequisites.R")

##
#Import datasets...
list_of_files <- list.files(path = "/media/user/seagate_ICM/AICDA_analysis/Single_cellStudies/RData_rawobjects", recursive = TRUE, pattern = "*.RData", full.names = TRUE)
tmp_names<-gsub("/media/user/seagate_ICM/AICDA_analysis/Single_cellStudies/RData_rawobjects/","",list_of_files[])
tmp_names<-gsub('_([^_]*)$', '',gsub('_([^_]*)$', '', tmp_names[]))
tmp_names<-gsub("_data","",tmp_names)

single_cell_design<-data.frame(Dataset=tmp_names,Tumor_type=gsub("_.*","",tmp_names),organism=c(rep("Human",4),"Mouse",rep("Human",36)),Treatment=c(rep("None",2),"aPD1",rep("None",1),"aPD1-aCTLA4",rep("None",1),"aPD1-aPDL1-aCTLA4","None","None","Immunotherapy","Immunotherapy",rep("None",12),"aPDL1-aCTLA4",rep("None",12),"aPD1","aPD1-aCTLA4",rep("None",3)),Sorting=c(rep("All",3),rep("Non-tumoral",1),rep("All",1),rep("Non-tumoral",1),"All",rep("Non-tumoral",5),rep("All",8),rep("Immune",3),rep("All",1),rep("Immune",3),rep("All",1), "Immune","All","All", "Immune","Immune", "T-cell","Immune",rep("All",2),rep("Immune",3),"All"))
###
single_cell_design$Dataset[25:26]<-c("LIHC_GSE140228_immune10X","LIHC_GSE140228_immune_Smart")
####Do design table with: name dataset, organism, treatment, sorting, tumor type


###Import data sets by importing into a temporary env..
###
ss_cellnumbers<-list()
ss_cellAICDA<-list()
ss_cellcycle<-list()

for (i in 1:length(list_of_files)) {
  tmp_env <- new.env() #Create tmp_env object
load(list_of_files[i], tmp_env)
ss_cellnumbers[[i]]<-get(ls(tmp_env)[3], envir=tmp_env) 
tmp<-get(ls(tmp_env)[2], envir=tmp_env) 
ss_cellAICDA[[i]]<-tmp[tmp$features.plot=="AICDA",]
if (i==5) {
  ss_cellAICDA[[i]]<-tmp[tmp$features.plot=="Aicda",] #For the mouse dataset
}
ss_cellcycle[[i]]<-get(ls(tmp_env)[1], envir=tmp_env) 
}
names(ss_cellnumbers)<-single_cell_design$Dataset
names(ss_cellAICDA)<-single_cell_design$Dataset
names(ss_cellcycle)<-single_cell_design$Dataset

####Get data frame of number of cells per study...
single_cell_design$N_cells<-sapply(ss_cellnumbers, sum)

###Get data frame of proportion of cell types per study...
data2<-lapply(ss_cellnumbers, function(x){x/sum(x)}) #Transform to proportions
data2<-as.data.frame(rbindlist(map(data2, as.data.table), fill = TRUE, idcol = T))

###Reduce using dcast
data2<-reshape2::dcast(data = data2, formula = .id ~V1 , value.var = "N",fill = 0,fun.aggregate = sum,drop = T)
colnames(data2)[1]<-"Dataset"
rownames(data2)<-data2$Dataset
data2<-data2[,-1]
#########
##
####Get data for AICDA expression graph
ss_cellAICDA2<-as.data.frame(rbindlist(map(ss_cellAICDA, as.data.table), fill = TRUE, idcol = T))

###Reduce using dcast
ss_cellAICDA2<-reshape2::dcast(data = ss_cellAICDA2, formula = .id ~id , value.var = "avg.exp",fill = 0,fun.aggregate = sum,drop = T)
colnames(ss_cellAICDA2)[1]<-"Dataset"

rownames(ss_cellAICDA2)<-ss_cellAICDA2$Dataset
ss_cellAICDA2<-ss_cellAICDA2[,-1]
###Expression values: The values in the single-cell level expression matrix are normalized. We employed the global-scaling normalization method ('NormalizeData' function) in Seurat to scale the raw counts (UMI) in each cell to 10,000, and then log-transformed the results.

##For heatmaps.. 
#we converted raw count or FPKM, which depends on the available data, to TPM to ensure the expression level is relatively comparable between different datasets. The expression of a gene in the cell was quantified as log2(TPM+1).



###
##Scale data for better visualization

ss_cellAICDA2_norm<-t(log2(ss_cellAICDA2+1))
ss_cellAICDA2_norm<-as.data.frame(ss_cellAICDA2_norm)
ss_cellAICDA2_norm<-na.omit(ss_cellAICDA2_norm)
col_fun3 = colorRamp2(c(1.8,.5,.1,0.001,0), c("firebrick",scales::alpha("darkorange",.8),"yellow2","grey","ivory"))

###Organize data...make sure they have the same order
all(colnames(ss_cellAICDA2_norm)==single_cell_design$Dataset) #Reorder needed for matching
ss_cellAICDA2_norm<-ss_cellAICDA2_norm[,single_cell_design$Dataset]

####Barplot annotation...
###Cellproportions..data2, columns need to be cell type and rows the studies
##Make sure the order is the same
all(rownames(data2)==colnames(ss_cellAICDA2_norm))
data2<-data2[colnames(ss_cellAICDA2_norm),]
  all(rownames(data2)==colnames(ss_cellAICDA2_norm))
####Reduce cell types for plotting
  data2_reduced<-data2[,colnames(data2)%in%c("Plasma","B","Tprolif","CD4Tconv","Mono/Macro","Fibroblasts","Endothelial","Neutrophils","NK","Microglia","Mast")]
  tmp<- data2 %>% group_by(rownames(data2))  %>% summarize(CD8 = sum(CD8T,CD8Tex))
  
 data2_reduced$CD8<-tmp$CD8[order(factor(tmp$`rownames(data2)`, levels = rownames(data2)))]
 #Malignant
 colnames(data2)[c(1,22,32)]<-c("ACMal","Mal","OCMal")
  tmp<- data2 %>% group_by(rownames(data2))  %>% summarize(Malignant = sum(ACMal,Mal,OCMal))
 data2_reduced$Malignant<-tmp$Malignant[order(factor(tmp$`rownames(data2)`, levels = rownames(data2)))]
 #Others...
 colnames(data2)[c(18,41)]<-c("Hepatic_progenitor","Secretory_glandular")
  tmp<- data2 %>% group_by(rownames(data2))  %>% summarize(Others2 = sum(Acinar,Alveolar,Astrocyte,DC,Ductal,Endocrine,Epithelial,EryPro,Erythrocytes,GMP,Hepatic_progenitor,HSC,ILC,Keratinocytes,Melanocytes,Myocyte,Myofibroblasts,Neuron,Oligodendrocyte,OPC,Others,pDC,Pericytes,Progenitor,Promonocyte,Secretory_glandular,SMC,Stellate,TMKI67,Tproilf,Treg,Vascular))
 data2_reduced$Others<-tmp$Others2[order(factor(tmp$`rownames(data2)`, levels = rownames(data2)))]
  rowSums(data2_reduced) #Check all are 1
  
##For number of cells, change data to log10
data3<-data.frame(N_cells=single_cell_design$N_cells,N_converted=log10(single_cell_design$N_cells),row.names = single_cell_design$Dataset)  
 

#
```

```{r panel_7F, fig.width=19,fig.height=12,fig.show=T}
##Import data
data_dir<-"/media/user/seagate_ICM/AICDA_analysis/Github_repository/" #Set to where you downloaded the data
setwd(data_dir)
source("r/prerequisites.R")

#Import needed data
single_cell_design<-as.data.frame(read.delim("Data/Figure7/Figure7F_1.tab"))
data2_reduced<-as.data.frame(read.delim("Data/Figure7/Figure7F_2.tab"))
ss_cellAICDA2_norm<-as.data.frame(read.delim("Data/Figure7/Figure7F_3.tab"))
##For number of cells, change data to log10
data3<-data.frame(N_cells=single_cell_design$N_cells,N_converted=log10(single_cell_design$N_cells),row.names = single_cell_design$Dataset)  
 

col_fun3 = colorRamp2(c(1.8,.5,.1,0.001,0), c("firebrick",scales::alpha("darkorange",.8),"yellow2","grey","ivory"))
####Colors for cell types..
colors_cells<-c("brown4","orangered1","aquamarine","darkolivegreen","darkgray","darkcyan","cornflowerblue","plum1","chartreuse","purple","gold","deeppink1","darksalmon","wheat")
names(colors_cells)<-colnames(data2_reduced)
 


ha3 <- HeatmapAnnotation(Tumor_type=single_cell_design$Tumor_type,Organism=single_cell_design$organism,Treatment=single_cell_design$Treatment,Sorting=single_cell_design$Sorting,col = list(Tumor_type=c("ALL"="brown4","AML"="darkorange4","BCC"="gray","BRCA"="hotpink","CHOL"="lightsalmon","CLL"="plum3","CRC"="slategray1","DLBCL"="firebrick","Glioma"="dodgerblue","HNSC"="yellowgreen","KIRC"="wheat1","LIHC"="sienna3","MB"="white","NHL"="maroon4","NSCLC"="coral1","OSCC"="aliceblue","PAAD"="black","SKCM"="rosybrown1"),Organism=c("Human" = "gray16", "Mouse" = "gray91"),Treatment= c("None" = "white", "aPD1-aCTLA4" = "orange2","aPD1-aPDL1-aCTLA4"="blueviolet","Immunotherapy"="green1","aPDL1-aCTLA4"="darkred","aPD1"="darksalmon"),Sorting=c("All"="steelblue","Non-tumoral"="cyan4","Immune"="lightcoral","T-cell"="lightgreen")),
                         annotation_legend_param = list(Tumor_type=list(ncol=3,title = "Tumor type",at=c(unique(single_cell_design$Tumor_type)),labels=c(unique(single_cell_design$Tumor_type))),Organism = list(
                             title = "Organism",
                             at = c("Human", "Mouse"),
                             labels = c("Human", "Mouse")
                         ),Sorting=list(
                             title = "Sorting",
                             at = c(unique(single_cell_design$Sorting)),
                             labels = c(unique(single_cell_design$Sorting))
                         ),Treatment = list(
                             title = "Treatment",
                             at = c(unique(single_cell_design$Treatment)),
                             labels = c(unique(single_cell_design$Treatment))
                         )))



####Barplot annotation...
###Cellproportions..data2, columns need to be cell type and rows the studies
##Make sure the order is the same

ha_bar<-HeatmapAnnotation(foo = anno_barplot(data2_reduced,axis_param = list(direction = "reverse",at = c(0, 0.2, 0.4, 0.6, 0.8,1), 
            labels = c("0%", "20%", "40%", "60%", "80%","100%")),gp = gpar(fill = colors_cells), 
    bar_width = 1, height = unit(3, "cm")),cells=anno_barplot(data3[,-1],axis_param = list(direction = "reverse",at = c(0, 1,2,3,4,5), 
            labels = expression("0",10^1,10^2,10^3,10^4,10^5)), 
    bar_width = 1, height = unit(2, "cm")),foo2=anno_text(paste0("n = ",data3$N_cells), location = 1, rot = 90, 
    just = "right", gp = gpar(fontsize = 6)), annotation_name_rot = 90,gap =  unit(c(4,0.4), "mm"),annotation_label = c("Cell type\nproportions", "Number of \ncells in the\nstudy",""))

####
ss_aid_heatmap<-Heatmap(ss_cellAICDA2_norm,col = col_fun3,name = "Log2(TPM+1)",column_names_side = "top", row_names_gp = gpar(fontsize=10),column_names_rot = 45,row_names_side = "left",show_column_dend = T,show_row_dend =T, cluster_columns = T,cluster_rows = T,column_title_gp = gpar(fontsize = 18, fontface = "bold",col="black"),column_names_gp = gpar(fontsize=10),border = TRUE,show_column_names = T, heatmap_legend_param = list(title= "Log2(TPM+1)", at = rev(c(1.8,.5,.1,0.001,0)), labels = rev(c(1.8,.5,.1,0.001,0))),column_gap = unit(2, "mm"), top_annotation = ha3,bottom_annotation =ha_bar,right_annotation = rowAnnotation(foo = anno_text(rownames(ss_cellAICDA2_norm), location = 0, rot = 0, just = "left", gp = gpar(fontsize = 10))))


###
lg_1<-packLegend(Legend(title = "Cell type",legend_gp = gpar(fill = colors_cells),labels = names(colors_cells),ncol=3))

ss_aid_heatmap
pushViewport(viewport(width = 0.8, height = 0.8))
draw(lg_1, x = unit(39, "cm"), y = unit(1.25, "cm"), just = c("bottom"))

#12*19 save

 
```


***

\pagebreak