---
output:
  rmarkdown::html_document:
    highlight: pygments
    toc: true
    css: style.css
chunk_output_type: console
header-includes:
- \pagenumbering{gobble}
params:
    set_title: "Figure-3"
title: "`r params$set_title`"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE, autodep = knitr::dep_prev())
```

***

\pagebreak

# Fig 3A

We downloaded the RepliSeq measurements (as wavelet-smoothed signal, ile SM923451_hg19_wgEncodeUwRepliSeqGm12878WaveSignalRep1.bigWig) of 11 ENCODE cell lines (SK-N-SH = neuroblastoma; MCF-7 = mammary gland adenocarcinoma; BJ = skin fibroblast; NHEK = epidermal keratinocytes; HepG2 = liver carcinoma; IMR90 = fetal lung fibroblasts; K562 = leukemia; HeLa-S3 = cervical carcinoma; GM12878 = lymphoblastoid; HUVEC = umbilical vein endothelial cells; BG02ES = embryonic stem cell; ) from the UCSC Genome Browser (also available in NCBI GEO as GSE34399). To avoid biasing the sample, we excluded multiple lymphoblastoid cell lines and retained Gm12878 as a representative, which was previously described in [Differential DNA mismatch repair underlies mutation rate variation across the human genome] and [Sequencing newly replicated DNA reveals widespread plasticity in human replication timing]. We computed the average RepliSeq signal within 1Mb genome windows of the remaining 11 cell lines (cross-tissue replication timing signal), having resulting values that ranged from 0-100, where higher values indicate earlier replication.

**Creating the 1 MB windows**: 

First the exon coding regions where extracted using the [UCSC browser](http://genome.ucsc.edu/cgi-bin/hgTables?command=start), options track: UCSC Genes; table: knownGene; output format: BED - browser extensible data; option Exons. Then using bedtools we masked blacklisted regions defined by Anshul Kundage and Duke [bed file here](http://genome.ucsc.edu/cgi-bin/hgTrackUi?hgsid=876251189_lEwmtar4Jcr1MJgh92cL5gL9Anaa&g=wgEncodeDacMapabilityConsensusExcludable). Then we generated 1 Mb windows covering all the genome and calculated the effective exonic length (EEL) within each window for determining SNS densities (SNV counts in each window multiplied the by EEL fraction) to get SNS per MB. Furthermore, replication timing is known to be
organized in megabase-scale domains.

```{r panel3A_cals,echo=T,eval=F,fig.show="hide"}
###Calculations for Replication timing..
bed_exonCoding<-import("Data/Figure3/Raw_Calculations/bedfilesForBins/bedfilesForBins/codingExons_hg19_NOblacklist.bed")
bed_exonCoding
seqnames(bed_exonCoding)
tmp<-as.data.frame(bed_exonCoding)
table(tmp$seqnames)
paste0("chr",seq(1,22,1))
bed_exonCoding<-keepSeqlevels(bed_exonCoding,c(paste0("chr",seq(1,22,1)),"chrX"),pruning.mode="coarse")
seqnames(bed_exonCoding)
bed_exonCoding
GenomeInfoDb::seqlengths(bed_exonCoding)
require(GenomicRanges)
require(ggbio)
require(BSgenome.Hsapiens.UCSC.hg19)

chr.lengths = seqlengths(Hsapiens)[1:23]
names(chr.lengths)<-c(paste0("chr",seq(1,22,1)),"chrX")
chr.lengths
seqlengths(bed_exonCoding)<-chr.lengths
bed_exonCoding
seqlengths(bed_exonCoding)
bins <- GenomicRanges::tileGenome(GenomeInfoDb::seqlengths(bed_exonCoding), tilewidth=1000000, cut.last.tile.in.chrom=TRUE)
bins
seqinfo(Hsapiens)


##
#####Now import repliSeq data from each cell line, combine and get median score
# repli-seq analysis
library(rtracklayer)
library(hwglabr2)

##Import as GRanges object all the repliseq data, download from: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE34399

RepliSeq_SKNSH <- import("Data/Figure3/Raw_Calculations/CellLines_repliData/GSM923441_hg19_wgEncodeUwRepliSeqSknshWaveSignalRep1.bigWig")
RepliSeq_MCF<- import("Data/Figure3/Raw_Calculations/CellLines_repliData/GSM923442_hg19_wgEncodeUwRepliSeqMcf7WaveSignalRep1.bigWig")
RepliSeq_BJ<- import("Data/Figure3/Raw_Calculations/CellLines_repliData/GSM923444_hg19_wgEncodeUwRepliSeqBjWaveSignalRep2.bigWig")
RepliSeq_NHEK<- import("Data/Figure3/Raw_Calculations/CellLines_repliData/GSM923445_hg19_wgEncodeUwRepliSeqNhekWaveSignalRep1.bigWig")
RepliSeq_HEPG<- import("Data/Figure3/Raw_Calculations/CellLines_repliData/GSM923446_hg19_wgEncodeUwRepliSeqHepg2WaveSignalRep1.bigWig")
RepliSeq_IMR<- import("Data/Figure3/Raw_Calculations/CellLines_repliData/GSM923447_hg19_wgEncodeUwRepliSeqImr90WaveSignalRep1.bigWig")
RepliSeq_K<- import("Data/Figure3/Raw_Calculations/CellLines_repliData/GSM923448_hg19_wgEncodeUwRepliSeqK562WaveSignalRep1.bigWig")
RepliSeq_HELA<- import("Data/Figure3/Raw_Calculations/CellLines_repliData/GSM923449_hg19_wgEncodeUwRepliSeqHelas3WaveSignalRep1.bigWig")

RepliSeq_GM <- import("Data/Figure3/Raw_Calculations/CellLines_repliData/GSM923451_hg19_wgEncodeUwRepliSeqGm12878WaveSignalRep1.bigWig")
RepliSeq_HUVEC <- import("Data/Figure3/Raw_Calculations/CellLines_repliData/GSM923452_hg19_wgEncodeUwRepliSeqHuvecWaveSignalRep1.bigWig")
RepliSeq_BG <- import("Data/Figure3/Raw_Calculations/CellLines_repliData/GSM923453_hg19_wgEncodeUwRepliSeqBg02esWaveSignalRep1.bigWig")
#It contains the RepliScore for different genomic position as ranges

###Get average of RepliScore between all cellLines
###Create data frame with all values


Repli_allScores<-data.frame(S1=RepliSeq_SKNSH$score,S2=RepliSeq_MCF$score,S3=RepliSeq_BJ$score,S4=RepliSeq_NHEK$score,S5=RepliSeq_HEPG$score,S6=RepliSeq_IMR$score,S7=RepliSeq_K$score,S8=RepliSeq_HELA$score,S9=RepliSeq_GM$score,S10=RepliSeq_HUVEC$score, S11=RepliSeq_BG$score)

Repli_allScores$score_average<-matrixStats::rowMedians(as.matrix(Repli_allScores))
##Change negative values to 0
Repli_allScores$score_average<-ifelse(Repli_allScores$score_average<0, 0, Repli_allScores$score_average) #Drop values with negatives
RepliSeq_combined<-RepliSeq_SKNSH #Just to duplicate GRanges
RepliSeq_combined$score_average<-Repli_allScores$score_average
#Check distribution of repliScores
quantile(RepliSeq_combined$score_average, probs = seq(0,1,0.1))


####
# Get signal as "RleList"; the signal is stored in the "score" metadata column
RepliSeq_combined<-sort(GenomeInfoDb::sortSeqlevels(RepliSeq_combined))
score <- GenomicRanges::coverage(RepliSeq_combined, weight="score_average")
# Compute average signal per tile
RepliSeq_combined_Mb <- GenomicRanges::binnedAverage(bins, score, "average_score_Mb")
##Add extra column to say to which decile it belongs
RepliSeq_combined_Mb2<-as.data.frame(RepliSeq_combined_Mb)
RepliSeq_combined_Mb2<-RepliSeq_combined_Mb2 %>% mutate(Decile = ntile(average_score_Mb, 10)) #Use dplyr
RepliSeq_combined_Mb$Decile<-RepliSeq_combined_Mb2$Decile##Add also to the Granges object
##To know the ranges
decil_labels<-quantile(RepliSeq_combined_Mb$average_score_Mb, probs = seq(0,1,0.1))
decil_labels


#RepliSeq_combined_Mb #Same but with the Mb bins

##Import data, this is the file with all the mutations from ICGC
icgc2_allmuts<-as.data.frame(read.delim("Data/Figure3/Raw_Calculations/icgc2_allmuts.txt")) ##Obtained from the scripts on Figure 1

require(GenomicRanges)
require(ggbio)
require(BSgenome.Hsapiens.UCSC.hg19)

chr.lengths = seqlengths(Hsapiens)[1:23]
names(chr.lengths)<-c(paste0("chr",seq(1,22,1)),"chrX")
chr.lengths
###


#######
###
library(rtracklayer)
library(hwglabr2)



##To know the ranges
decil_labels<-quantile(RepliSeq_combined_Mb$average_score_Mb, probs = seq(0,1,0.1))
decil_labels

####Add data to mafcancer 
dim(icgc2_allmuts)
tmp<-data.frame(Gene=icgc2_allmuts$Hugo_Symbol[],start=icgc2_allmuts$Start_Position[], stop=icgc2_allmuts$End_Position[], chrom=icgc2_allmuts$Chromosome[])
#Create data frame with mutations
tmp$chrom<-paste0("chr",tmp$chrom)#Add chr
dim(tmp)
tmp = makeGRangesFromDataFrame(df = tmp, start.field = "start", end.field = "stop", seqnames.field = "chrom",keep.extra.columns = T) # to Granges
#findOverlaps
olaps = findOverlaps(tmp, RepliSeq_combined_Mb,select = "arbitrary")  #This function gives the matrix positions for each mutation (positions at RepliSeq_combined_Mb)
#olaps<-as.data.frame(olaps)
##Add data to tmp
tmp<-as.data.frame(tmp)
tmp$average_score_Mb<-RepliSeq_combined_Mb2$average_score_Mb[olaps]
tmp$Decile<-RepliSeq_combined_Mb2$Decile[olaps]
tmp$Start_Mb<-RepliSeq_combined_Mb2$start[olaps]
tmp$End_Mb<-RepliSeq_combined_Mb2$end[olaps]
###Add to the maf file
icgc2_allmuts$Average_RepliScore_Mb<-tmp$average_score_Mb
icgc2_allmuts$RepliScore_Decile<-tmp$Decile
icgc2_allmuts$Start_Mb<-tmp$Start_Mb
icgc2_allmuts$End_Mb<-tmp$End_Mb
table(icgc2_allmuts$RepliScore_Decile)


repli_melt_icgc<-subset(icgc2_allmuts[,c("Tumor_type","SBS.Sig.max","Tumor_Sample_Barcode","RepliScore_Decile","Start_Mb","End_Mb","Chromosome")], icgc2_allmuts$SBS.Sig.max %in% c("AICDA_canonical","SBS2","SBS13"))
dim(repli_melt_icgc)
#repli_melt$range<-paste0(repli_melt$Start_Mb,"-",repli_melt$End_Mb,"-",repli_melt$Chromosome)
repli_melt_icgc$Tumor_Sample_Barcode<-as.character(repli_melt_icgc$Tumor_Sample_Barcode)
repli_melt_icgc$Mutations<-1 #TO allow addition in next step
##Collapse summing by tumor_type the RepliScore but separating each signature
library(dplyr)
##Melt and sum, this is the mutations per megabase for each signature for each tumor type
repli_melt_icgc2<-repli_melt_icgc %>% group_by(Tumor_type, SBS.Sig.max,Start_Mb,End_Mb,Chromosome) %>% summarize(Mutations_Mb = sum(Mutations))
repli_melt_icgc2<-as.data.frame(repli_melt_icgc2)
head(repli_melt_icgc2)
dim(repli_melt_icgc2)
###Assign decil to each mutation
repli_melt_icgc2$range<-paste0(repli_melt_icgc2$Start_Mb,"-",repli_melt_icgc2$End_Mb,"-",repli_melt_icgc2$Chromosome)
RepliSeq_combined_Mb2$range<-paste0(RepliSeq_combined_Mb2$start,"-",RepliSeq_combined_Mb2$end,"-",gsub("chr","",RepliSeq_combined_Mb2$seqnames))
tmp<-match(repli_melt_icgc2$range,RepliSeq_combined_Mb2$range)
repli_melt_icgc2$RepliScore_Decile<-RepliSeq_combined_Mb2$Decile[tmp]

#repli_melt_icgc2$exonic_effective_length<-RepliSeq_combined_Mb2$exonic_effective_length[tmp] #Add exonic effective length to each range
repli_melt_icgc2$density_Mutations_Mb<-repli_melt_icgc2$Mutations_Mb
#repli_melt_icgc2$density_Mutations_Mb<-repli_melt_icgc2$Mutations_Mb/tilewidth#Adjust by percentage of tilewidth(relative to total length)
###SUm ussing deciles
repli_melt_icgc2_deciles<-repli_melt_icgc2 %>% group_by(Tumor_type, SBS.Sig.max,RepliScore_Decile) %>% summarize(Mutations_Mb_decile = sum(density_Mutations_Mb))
dim(repli_melt_icgc2_deciles)
head(repli_melt_icgc2_deciles)
repli_melt_icgc2_deciles<-as.data.frame(repli_melt_icgc2_deciles)
###Adjust data to use the mean per tumor type
##Get number of samples per tumor type
Tumor_type_number <-repli_melt_icgc[!duplicated(repli_melt_icgc$Tumor_Sample_Barcode), ]
Tumor_type_number<-table(Tumor_type_number$Tumor_type)
repli_melt_icgc2_deciles$Mutations_Mb_decile_mean<-1 #Create the column
for (i in 1:length(Tumor_type_number)) {
repli_melt_icgc2_deciles$Mutations_Mb_decile_mean<-ifelse(repli_melt_icgc2_deciles$Tumor_type==names(Tumor_type_number)[i],yes = repli_melt_icgc2_deciles$Mutations_Mb_decile/Tumor_type_number[i],repli_melt_icgc2_deciles$Mutations_Mb_decile_mean)
}


pancancer_repli_p_icgc<-ggboxplot(data = subset(repli_melt_icgc2_deciles,!is.na(RepliScore_Decile)),x ="RepliScore_Decile" , y = "Mutations_Mb_decile_mean",add = "jitter",color = "SBS.Sig.max")+ theme_classic()+geom_smooth(data=subset(repli_melt_icgc2_deciles,!is.na(RepliScore_Decile)),aes(x = RepliScore_Decile, y = Mutations_Mb_decile_mean, color=SBS.Sig.max, fill=SBS.Sig.max), size = 0.5,show.legend = F, se=FALSE)+ labs( x="Late to early replication", y = "Mutation number mean per Mb")+ guides(color=guide_legend(title = "Signature"))

##This is the Supplementary Figure  not included in the article

###Using Medians
repli_melt_icgc3<-repli_melt_icgc %>% group_by(Tumor_Sample_Barcode,Tumor_type, SBS.Sig.max,Start_Mb,End_Mb,Chromosome) %>% summarize(Mutations_Mb = sum(Mutations))
repli_melt_icgc3<-as.data.frame(repli_melt_icgc3)
head(repli_melt_icgc3)
dim(repli_melt_icgc3)
###Assign decil to each mutation
repli_melt_icgc3$range<-paste0(repli_melt_icgc3$Start_Mb,"-",repli_melt_icgc3$End_Mb,"-",repli_melt_icgc3$Chromosome)
tmp<-match(repli_melt_icgc3$range,RepliSeq_combined_Mb2$range)
repli_melt_icgc3$RepliScore_Decile<-RepliSeq_combined_Mb2$Decile[tmp]
#repli_melt_icgc3$exonic_effective_length<-RepliSeq_combined_Mb2$exonic_effective_length[tmp] #Add exonic effective length to each range
repli_melt_icgc3$density_Mutations_Mb<-repli_melt_icgc3$Mutations_Mb#*(repli_melt_icgc3$exonic_effective_length/tilewidth) #Adjust by percentage of e


###SUm ussing deciles
repli_melt_icgc3_deciles<-repli_melt_icgc3 %>% group_by(Tumor_Sample_Barcode,Tumor_type, SBS.Sig.max,RepliScore_Decile) %>% summarize(Mutations_Mb_decile = sum(density_Mutations_Mb))
dim(repli_melt_icgc3_deciles)
head(repli_melt_icgc3_deciles)
repli_melt_icgc3_deciles<-as.data.frame(repli_melt_icgc3_deciles)
repli_melt_icgc3_deciles<-subset(repli_melt_icgc3_deciles,!is.na(RepliScore_Decile))
###Graph using median and confidence intervals 95%
##The approximate 95% confidence intervals of the median (across tumour samples: Fig. 2d and Extended Data Fig. 1e,f)for the 1 Mbwindows were estimated using theformula +-1.58*IQR/ sqrt(n_samp), as defined in the R function boxplot.stats and references therein
###Create data frame with medians to be able to lauch the regression
repli_melt4_icgc<-repli_melt_icgc3_deciles %>% group_by(Tumor_type, SBS.Sig.max,RepliScore_Decile) %>% summarize(Mutations_Mb_median = median(Mutations_Mb_decile))
repli_melt4_icgc<-as.data.frame(repli_melt4_icgc)
head(repli_melt4_icgc)
dim(repli_melt4_icgc)

pancancer_perTumor_repli_p_icgc<-ggplot(repli_melt_icgc3_deciles[,],aes(x = RepliScore_Decile, y = Mutations_Mb_decile, colour = SBS.Sig.max,fill=SBS.Sig.max))+ theme_classic()+facet_wrap(facets = vars(Tumor_type),nrow = 7,ncol = 5)+ guides(color=guide_legend(title = "Signature"))+stat_summary(fun= median,geom = "errorbar",position = position_dodge(0.95),width = .2,fun.max = function(x) median(x) + (IQR(x)*1.58 / sqrt(length(x))), fun.min = function(x) median(x) - (IQR(x)*1.58 / sqrt(length(x))))+stat_summary(fun.y = "median", geom = "point",position = position_dodge(0.95),size = 1, colour="black",show.legend = F)+geom_smooth(repli_melt4_icgc[,],mapping = aes(x = RepliScore_Decile, y = Mutations_Mb_median, colour = SBS.Sig.max,fill=SBS.Sig.max), size = 0.5,show.legend = F, se=F)+scale_x_continuous(breaks=seq(1,10,by = 1))+ labs( x="Late to early replication", y = "Sample's median mutation number per Mb")


#######
####
#Now for better visualization normalize the mutation density using the total number of mutation for each sample within signatures and then do a boxplot (by tumor)
repli_melt_icgc5<-repli_melt_icgc %>% group_by(Tumor_Sample_Barcode,Tumor_type, SBS.Sig.max) %>% summarize(Mutations_Mb = sum(Mutations))
repli_melt_icgc5<-as.data.frame(repli_melt_icgc5)
head(repli_melt_icgc5)
dim(repli_melt_icgc5)
#Get the total of mutations first
##Now get the number of mutation per decil per sample, saved in repli_melt_icgc3_deciles

repli_melt_icgc3_deciles$Mutations_Mb_decile_normalized<-repli_melt_icgc3_deciles$Mutations_Mb_decile/repli_melt_icgc5$Mutations_Mb[match(paste0(repli_melt_icgc3_deciles$Tumor_Sample_Barcode,repli_melt_icgc3_deciles$SBS.Sig.max),paste0(repli_melt_icgc5$Tumor_Sample_Barcode,repli_melt_icgc5$SBS.Sig.max))]
  
pancancer_repli_p_icgc_normalized<-ggboxplot(data = subset(repli_melt_icgc3_deciles,!is.na(RepliScore_Decile)),x ="RepliScore_Decile" , y = "Mutations_Mb_decile_normalized",add = "jitter",color = "SBS.Sig.max",fill = "white",size = 0.5)+ theme_classic()+facet_wrap(facets = vars(Tumor_type),nrow = 7,ncol = 5)+geom_smooth(data=subset(repli_melt_icgc3_deciles,!is.na(RepliScore_Decile)),aes(x = RepliScore_Decile, y = Mutations_Mb_decile_normalized, color=SBS.Sig.max, fill=SBS.Sig.max), size = 0.5,show.legend = F, se=FALSE)+ labs( x="Late to early replication", y = "Normalized mutation density")+ guides(color=guide_legend(title = "Signature"))+scale_y_continuous(limits = c(0,0.5))
  
  
 ####Now try to generate biomodal distributions and evaluate skewness 
###Sum mutations per sample per tumor type agroupating by late (Decile 1-5) or early (Decile 6-10), then multiply all "Lates" by -1 (also this is per signature)
head(repli_melt_icgc3_deciles) #Use this data
repli_melt_icgc3_deciles$Replication_Zone<-ifelse(repli_melt_icgc3_deciles$RepliScore_Decile<=5,"Late","Early")
repli_melt_icgc3_deciles2<-repli_melt_icgc3_deciles %>% group_by(Tumor_Sample_Barcode,Tumor_type, SBS.Sig.max,Replication_Zone) %>% summarize(Mutations_Mb_decile_zone = sum(Mutations_Mb_decile)) #Sum
repli_melt_icgc3_deciles2<-as.data.frame(repli_melt_icgc3_deciles2)
#Multiple by -1 the Late zones
repli_melt_icgc3_deciles2$Mutations_Mb_decile_zone_histo<-ifelse(repli_melt_icgc3_deciles2$Replication_Zone=="Late",repli_melt_icgc3_deciles2$Mutations_Mb_decile_zone*-1,repli_melt_icgc3_deciles2$Mutations_Mb_decile_zone)


  
```

```{r panel_3A, fig.width=14,fig.height=10}
#
##For main
library(ggpubr)
library(reshape2)
##Import data
data_dir<-"/media/user/seagate_ICM/AICDA_analysis/Github_repository/" #Set to where you downloaded the data
setwd(data_dir)
##Import needed data calculated in previous chunk
repli_melt_icgc3_deciles2<-as.data.frame(read.delim("Data/Figure3/Figure3A.tab"))

#AICDA
library(lawstat)
library(e1071)
skew<-skewness(subset(repli_melt_icgc3_deciles2$Mutations_Mb_decile_zone_histo,repli_melt_icgc3_deciles2$SBS.Sig.max%in%"AICDA_canonical"))
t<-symmetry.test(subset(repli_melt_icgc3_deciles2$Mutations_Mb_decile_zone_histo,repli_melt_icgc3_deciles2$SBS.Sig.max%in%"AICDA_canonical"),side =  "left",boot = TRUE,B = 1000,q = 8/9)


#Or, #The p-value is computed by Monte Carlo simulation, if p val <0.05 then data is skewed

#skewness.norm.test(subset(repli_melt_icgc3_deciles2$Mutations_Mb_decile_zone_histo,repli_melt_icgc3_deciles2$SBS.Sig.max%in%"AICDA_canonical"))

#If skewness is less than -1 or greater than 1, the distribution is highly skewed.
#If skewness is between -1 and -0.5 or between 0.5 and 1, the distribution is moderately skewed.
#If skewness is between -0.5 and 0.5, the distribution is approximately symmetric.

###

expSup <- function(w, digits=0) {
    sprintf(paste0("%.", digits, "f*x*10^{%d}"), w/10^floor(log10(abs(w))), floor(log10(abs(w))))
}

t<-paste0("pval == ",expSup(pnorm(t$statistic),digits = 2))

p_tmp1<-ggplot(subset(repli_melt_icgc3_deciles2,repli_melt_icgc3_deciles2$SBS.Sig.max%in%"AICDA_canonical"), aes(x=Mutations_Mb_decile_zone_histo, color=Replication_Zone,fill=Replication_Zone)) +geom_histogram( alpha=0.5, position="identity",bins=70)+theme_classic()+labs(title = "AICDA signature",x="Late to early mutations per Mb", y = "n samples",fill = "Replication time zone",color="Replication time zone")+scale_color_manual(values=alpha(c("#F8766D", "darkred"), c(1,.5)))+ scale_fill_manual(values=alpha(c("#F8766D", "darkred"), c(1,.5)))+annotate("text", label=paste0("Skewness = ",round(skew,2)), x=2500, y=1800, fontface =1,size=3)+annotate("text", label =t,  x=2500, y=1775, fontface =2,size=3,parse =TRUE)+theme(plot.title = element_text(hjust = 0.5,face="bold"))


#Now for SBS2
skew<-skewness(subset(repli_melt_icgc3_deciles2$Mutations_Mb_decile_zone_histo,repli_melt_icgc3_deciles2$SBS.Sig.max%in%"SBS2"))
t<-symmetry.test(subset(repli_melt_icgc3_deciles2$Mutations_Mb_decile_zone_histo,repli_melt_icgc3_deciles2$SBS.Sig.max%in%"SBS2"),side =  "left",boot = TRUE,B = 1000,q = 8/9)
t<-paste0("pval == ",expSup(pnorm(t$statistic),digits = 2))

p_tmp2<-ggplot(subset(repli_melt_icgc3_deciles2,repli_melt_icgc3_deciles2$SBS.Sig.max%in%"SBS2"), aes(x=Mutations_Mb_decile_zone_histo, color=Replication_Zone,fill=Replication_Zone)) +geom_histogram( alpha=0.5, position="identity",bins=70)+theme_classic()+labs(title = "SBS2 signature",x="Late to early mutations per Mb", y = "n samples",fill = "Replication time zone",color="Replication time zone")+scale_color_manual(values=alpha(c("#619CFF", "darkblue"), c(1,.5)))+ scale_fill_manual(values=alpha(c("#619CFF", "darkblue"), c(1,.5)))+annotate("text", label=paste0("Skewness = ",round(skew,2)), x=10000, y=280, fontface =1,size=3)+annotate("text", label =t,  x=10000, y=275, fontface =2,size=3,parse =TRUE)+theme(plot.title = element_text(hjust = 0.5,face="bold"))
#Now for SBS13
#Now for SBS13
skew<-skewness(subset(repli_melt_icgc3_deciles2$Mutations_Mb_decile_zone_histo,repli_melt_icgc3_deciles2$SBS.Sig.max%in%"SBS13"))
t<-symmetry.test(subset(repli_melt_icgc3_deciles2$Mutations_Mb_decile_zone_histo,repli_melt_icgc3_deciles2$SBS.Sig.max%in%"SBS13"),side =  "left",boot = TRUE,B = 1000,q = 8/9)
t<-paste0("pval == ",expSup(pnorm(t$statistic),digits = 2))

p_tmp3<-ggplot(subset(repli_melt_icgc3_deciles2,repli_melt_icgc3_deciles2$SBS.Sig.max%in%"SBS13"), aes(x=Mutations_Mb_decile_zone_histo, color=Replication_Zone,fill=Replication_Zone)) +geom_histogram( alpha=0.5, position="identity",bins=70)+theme_classic()+labs(title = "SBS13 signature",x="Late to early mutations per Mb", y = "n samples",fill = "Replication time zone",color="Replication time zone")+scale_color_manual(values=alpha(c("#00BA38", "darkgreen"), c(1,.5)))+ scale_fill_manual(values=alpha(c("#00BA38", "darkgreen"), c(1,.5)))+annotate("text", label=paste0("Skewness = ",round(skew,2)), x=10000, y=330, fontface =1,size=3)+annotate("text", label =t,  x=10000, y=325, fontface =2,size=3,parse =TRUE)+theme(plot.title = element_text(hjust = 0.5,face="bold"))


###Arrange Together
p_histo_repli_all<- ggarrange(p_tmp1,p_tmp2,p_tmp3,ncol = 3,align = "hv",legend = "bottom")

p_histo_repli_all
#

```

***

\pagebreak

# Supplementary Figure 11

```{r, echo=T,eval=F,fig.show="hide"}
##For main
library(ggpubr)
library(reshape2)
##Import data
data_dir<-"/media/user/seagate_ICM/AICDA_analysis/Github_repository/" #Set to where you downloaded the data
setwd(data_dir)
##Import needed data calculated in previous chunk
repli_melt_icgc3_deciles2<-as.data.frame(read.delim("Data/Figure3/Figure3A.tab"))

#AICDA
library(lawstat)
library(e1071)

########Repeat per tumor type for supplementary,

expSup2 <- function(w, digits=0) {
  sprintf(paste0("%.", digits, "f*x*10^%d"), w/10^floor(log10(abs(w))), floor(log10(abs(w))))
}

###Generate values to anotate
f_labels <- data.frame(Tumor_type = unique(repli_melt_icgc3_deciles2$Tumor_type), label = unique(repli_melt_icgc3_deciles2$Tumor_type),label2 = unique(repli_melt_icgc3_deciles2$Tumor_type),label3 = unique(repli_melt_icgc3_deciles2$Tumor_type),label4 ="pval == ")

for (i in 1:length(unique(repli_melt_icgc3_deciles2$Tumor_type))) {
  tmp<-subset(repli_melt_icgc3_deciles2,repli_melt_icgc3_deciles2$Tumor_type%in%unique(repli_melt_icgc3_deciles2$Tumor_type)[i])
  skew<-skewness(subset(tmp$Mutations_Mb_decile_zone_histo[],tmp$SBS.Sig.max%in%"AICDA_canonical"))
  skew<-ifelse(is.na(skew)==TRUE,"0",round(skew,2))
  t<-ifelse(skew<=0,"left","right")
t<-symmetry.test(subset(tmp$Mutations_Mb_decile_zone_histo[],tmp$SBS.Sig.max%in%"AICDA_canonical"),side =  t,boot = TRUE,B = 2000,q = 8/9)
###
f_labels$label3[i]<-ifelse(skew<0 & pnorm(t$statistic)<0.05,"Enriched late",ifelse(skew>0 & pnorm(-t$statistic)<0.05,"Enriched early","No enrichment"))
valueP<-ifelse(skew<0 ,pnorm(t$statistic),pnorm(-t$statistic))
t<-ifelse(valueP<0.05,yes = expSup2(valueP,digits = 2),no = ,round(valueP,2))

f_labels$label[i]<-skew;f_labels$label2[i]<-t;

}


f_labels$label2<-paste0(f_labels$label4,f_labels$label2)
f_labels$label<-paste0("Skewness = ",f_labels$label)

p_tmp1_1<-
ggplot(subset(repli_melt_icgc3_deciles2,repli_melt_icgc3_deciles2$SBS.Sig.max%in%"AICDA_canonical"), aes(x=Mutations_Mb_decile_zone_histo, color=Replication_Zone,fill=Replication_Zone)) +geom_histogram( alpha=0.5, position="identity",bins=150)+theme_classic()+facet_wrap(facets = vars(Tumor_type),nrow = 7,ncol = 5)+labs(title = "AICDA signature",x="Late to early mutations per Mb", y = "n samples",fill = "Replication time zone",color="Replication time zone")+scale_color_manual(values=alpha(c("#F8766D", "darkred"), c(1,.5)))+ scale_fill_manual(values=alpha(c("#F8766D", "darkred"), c(1,.5)))+geom_text( mapping = aes(label = label,x = 2500, y = 150,), data = f_labels[,1:2],inherit.aes = F,size=3)+geom_text( mapping = aes(label = label2,x = 2500, y = 130,), data = f_labels[,c(1,3)],inherit.aes = F, size=3,parse = T)+geom_text( mapping = aes(label = label3,x = 2500, y = 112,fontface=2), data = f_labels[,c(1,4)],inherit.aes = F, size=3)+theme(plot.title = element_text(hjust = 0.5,face="bold"))


#ggsave(filename = "/media/user/seagate_ICM/AICDA_analysis/ICGC/Repliseq/Replication_enrichment_AICDA_byTumor_ICGC.pdf",p_tmp1,width = 14,height = 12)

###SBS2
###Generate values to anotate
f_labels <- data.frame(Tumor_type = unique(repli_melt_icgc3_deciles2$Tumor_type[repli_melt_icgc3_deciles2$SBS.Sig.max%in%"SBS2"]), label = unique(repli_melt_icgc3_deciles2$Tumor_type[repli_melt_icgc3_deciles2$SBS.Sig.max%in%"SBS2"]),label2 = unique(repli_melt_icgc3_deciles2$Tumor_type[repli_melt_icgc3_deciles2$SBS.Sig.max%in%"SBS2"]),label3 = unique(repli_melt_icgc3_deciles2$Tumor_type[repli_melt_icgc3_deciles2$SBS.Sig.max%in%"SBS2"]),label4 ="pval == ")

for (i in 1:length(unique(repli_melt_icgc3_deciles2$Tumor_type[repli_melt_icgc3_deciles2$SBS.Sig.max%in%"SBS2"]))) {
  tmp<-subset(repli_melt_icgc3_deciles2,repli_melt_icgc3_deciles2$Tumor_type[]%in%unique(repli_melt_icgc3_deciles2$Tumor_type[repli_melt_icgc3_deciles2$SBS.Sig.max%in%"SBS2"])[i])
  
  skew<-skewness(subset(tmp$Mutations_Mb_decile_zone_histo[],tmp$SBS.Sig.max%in%"SBS2"))
  skew<-ifelse(is.na(skew)==TRUE,"0",round(skew,2))
  t<-ifelse(skew<=0,"left","right")
 if (length(subset(tmp$Mutations_Mb_decile_zone_histo[],tmp$SBS.Sig.max%in%"SBS2"))>4) {
    t<-symmetry.test(subset(tmp$Mutations_Mb_decile_zone_histo[],tmp$SBS.Sig.max%in%"SBS2"),side =  t,boot = TRUE,B = 1000,q = 8/9)
f_labels$label3[i]<-ifelse(skew<0 & pnorm(t$statistic)<0.05,"Enriched late",ifelse(skew>0 & pnorm(-t$statistic)<0.05,"Enriched early","No enrichment"))
valueP<-ifelse(skew<0 ,pnorm(t$statistic),pnorm(-t$statistic))
t<-ifelse(valueP<0.05,yes = expSup2(valueP,digits = 2),no = ,round(valueP,2))

f_labels$label[i]<-skew;f_labels$label2[i]<-t;
  }else{
    f_labels$label[i]<-skew;f_labels$label2[i]<-0;f_labels$label3[i]<-"No enrichment"
}
}


f_labels$label2<-paste0(f_labels$label4,f_labels$label2)
f_labels$label<-paste0("Skewness = ",f_labels$label)

p_tmp2_2<-ggplot(subset(repli_melt_icgc3_deciles2,repli_melt_icgc3_deciles2$SBS.Sig.max%in%"SBS2"), aes(x=Mutations_Mb_decile_zone_histo, color=Replication_Zone,fill=Replication_Zone)) +geom_histogram( alpha=0.5, position="identity",bins=150)+theme_classic()+facet_wrap(facets = vars(Tumor_type),nrow = 7,ncol = 5)+labs(title = "SBS2 signature",x="Late to early mutations per Mb", y = "n samples",fill = "Replication time zone",color="Replication time zone")+scale_color_manual(values=alpha(c("#619CFF", "darkblue"), c(1,.5)))+ scale_fill_manual(values=alpha(c("#619CFF", "darkblue"), c(1,.5)))+geom_text( mapping = aes(label = label,x = 10000, y = 60,), data = f_labels[,1:2],inherit.aes = F,size=3)+geom_text( mapping = aes(label = label2,x = 10000, y = 55,), data = f_labels[,c(1,3)],inherit.aes = F, size=3,parse = T)+geom_text( mapping = aes(label = label3,x = 10000, y = 50,fontface=2), data = f_labels[,c(1,4)],inherit.aes = F, size=3)+theme(plot.title = element_text(hjust = 0.5,face="bold"))

#ggsave(filename = "/media/user/seagate_ICM/AICDA_analysis/ICGC/Repliseq/Replication_enrichment_SBS2_byTumor_ICGC.pdf",p_tmp1,width = 16,height = 12)


##SBS13

f_labels <- data.frame(Tumor_type = unique(repli_melt_icgc3_deciles2$Tumor_type[repli_melt_icgc3_deciles2$SBS.Sig.max%in%"SBS13"]), label = unique(repli_melt_icgc3_deciles2$Tumor_type[repli_melt_icgc3_deciles2$SBS.Sig.max%in%"SBS13"]),label2 = unique(repli_melt_icgc3_deciles2$Tumor_type[repli_melt_icgc3_deciles2$SBS.Sig.max%in%"SBS13"]),label3 = unique(repli_melt_icgc3_deciles2$Tumor_type[repli_melt_icgc3_deciles2$SBS.Sig.max%in%"SBS13"]),label4 ="pval == ")

for (i in 1:length(unique(repli_melt_icgc3_deciles2$Tumor_type[repli_melt_icgc3_deciles2$SBS.Sig.max%in%"SBS13"]))) {
  tmp<-subset(repli_melt_icgc3_deciles2,repli_melt_icgc3_deciles2$Tumor_type[]%in%unique(repli_melt_icgc3_deciles2$Tumor_type[repli_melt_icgc3_deciles2$SBS.Sig.max%in%"SBS13"])[i])
  
  skew<-skewness(subset(tmp$Mutations_Mb_decile_zone_histo[],tmp$SBS.Sig.max%in%"SBS13"))
  skew<-ifelse(is.na(skew)==TRUE,"0",round(skew,2))
  t<-ifelse(skew<=0,"left","right")
  #Note to now if do the next part
  if (length(subset(tmp$Mutations_Mb_decile_zone_histo[],tmp$SBS.Sig.max%in%"SBS13"))>4) {
    t<-symmetry.test(subset(tmp$Mutations_Mb_decile_zone_histo[],tmp$SBS.Sig.max%in%"SBS13"),side =  t,boot = TRUE,B = 1000,q = 8/9)
f_labels$label3[i]<-ifelse(skew<0 & pnorm(t$statistic)<0.05,"Enriched late",ifelse(skew>0 & pnorm(-t$statistic)<0.05,"Enriched early","No enrichment"))
valueP<-ifelse(skew<0 ,pnorm(t$statistic),pnorm(-t$statistic))
t<-ifelse(valueP<0.05,yes = expSup2(valueP,digits = 2),no = round(valueP,2))

f_labels$label[i]<-skew;f_labels$label2[i]<-t;
  }else{
    f_labels$label[i]<-skew;f_labels$label2[i]<-0;f_labels$label3[i]<-"No enrichment"
  }
###



}


f_labels$label2<-paste0(f_labels$label4,f_labels$label2)
f_labels$label<-paste0("Skewness = ",f_labels$label)

p_tmp3_3<-
ggplot(subset(repli_melt_icgc3_deciles2,repli_melt_icgc3_deciles2$SBS.Sig.max%in%"SBS13"), aes(x=Mutations_Mb_decile_zone_histo, color=Replication_Zone,fill=Replication_Zone)) +geom_histogram( alpha=0.5, position="identity",bins=150)+theme_classic()+facet_wrap(facets = vars(Tumor_type),nrow = 5,ncol = 5)+labs(title = "SBS13 signature",x="Late to early mutations per Mb", y = "n samples",fill = "Replication time zone",color="Replication time zone")+scale_color_manual(values=alpha(c("#00BA38", "darkgreen"), c(1,.5)))+ scale_fill_manual(values=alpha(c("#00BA38", "darkgreen"), c(1,.5)))+geom_text( mapping = aes(label = label,x = 10000, y = 60,), data = f_labels[,1:2],inherit.aes = F,size=3)+geom_text( mapping = aes(label = label2,x = 10000, y = 55,), data = f_labels[,c(1,3)],inherit.aes = F, size=3,parse = T)+geom_text( mapping = aes(label = label3,x = 10000, y = 50,fontface=2), data = f_labels[,c(1,4)],inherit.aes = F, size=3)+theme(plot.title = element_text(hjust = 0.5,face="bold"))




```

```{r SUP11, echo=T,eval=T,fig.height=12,fig.width=14}
p_tmp1_1

p_tmp2_2

p_tmp3_3
```

***

\pagebreak


# Fig 3B

Some preliminary calculations for the mutations sum within the 25 kb bins. Main figure generated in python, check Fi3C_script.py.

```{r,echo=T,eval=F}
########
#####
######
####
##Need to run first panel3A_cals chunk to obtain the RepliSeq_combined_Mb object
##Get Granges object by tumor type and having the number of each signature mutations by each range using 25 kb windows

tilewidth<-25000
bins <- GenomicRanges::tileGenome(GenomeInfoDb::seqlengths(RepliSeq_combined_Mb),tilewidth=tilewidth, cut.last.tile.in.chrom=TRUE)
score <- GenomicRanges::coverage(RepliSeq_combined, weight="score_average")
# Compute average signal per tile
RepliSeq_combined_25k <- GenomicRanges::binnedAverage(bins, score, "average_score_Mb")
##Add extra column to say to which decile it belongs
RepliSeq_combined_25k2<-as.data.frame(RepliSeq_combined_25k)
RepliSeq_combined_25k2<-RepliSeq_combined_25k2 %>% mutate(Decile = ntile(average_score_Mb, 10)) #Use dplyr
RepliSeq_combined_25k2$Decile<-RepliSeq_combined_25k2$Decile##Add also to the Granges object
##To know the ranges
decil_labels<-quantile(RepliSeq_combined_25k$average_score_Mb, probs = seq(0,1,0.1))
decil_labels

dim(icgc2_allmuts)
tmp<-data.frame(Gene=icgc2_allmuts$Hugo_Symbol[],start=icgc2_allmuts$Start_Position[], stop=icgc2_allmuts$End_Position[], chrom=icgc2_allmuts$Chromosome[])
#Create data frame with mutations
tmp$chrom<-paste0("chr",tmp$chrom)#Add chr
dim(tmp)
tmp = makeGRangesFromDataFrame(df = tmp, start.field = "start", end.field = "stop", seqnames.field = "chrom",keep.extra.columns = T) # to Granges
#findOverlaps
olaps = findOverlaps(tmp, RepliSeq_combined_25k,select = "arbitrary")  #This function gives the matrix positions for each mutation (positions at RepliSeq_combined_Mb)
#olaps<-as.data.frame(olaps)
##Add data to tmp
tmp<-as.data.frame(tmp)
tmp$average_score_Mb<-RepliSeq_combined_25k2$average_score_Mb[olaps]
tmp$Decile<-RepliSeq_combined_25k2$Decile[olaps]
tmp$Start_Mb<-RepliSeq_combined_25k2$start[olaps]
tmp$End_Mb<-RepliSeq_combined_25k2$end[olaps]
###Add to the maf file
icgc2_allmuts$Average_RepliScore_Mb<-tmp$average_score_Mb
icgc2_allmuts$RepliScore_Decile<-tmp$Decile
icgc2_allmuts$Start_Mb<-tmp$Start_Mb
icgc2_allmuts$End_Mb<-tmp$End_Mb
table(icgc2_allmuts$RepliScore_Decile)


repli_melt_icgc<-subset(icgc2_allmuts[,c("Tumor_type","SBS.Sig.max","Tumor_Sample_Barcode","RepliScore_Decile","Start_Mb","End_Mb","Chromosome")], icgc2_allmuts$SBS.Sig.max %in% c("AICDA_canonical","SBS2","SBS13"))
dim(repli_melt_icgc)
repli_melt_icgc$Tumor_Sample_Barcode<-as.character(repli_melt_icgc$Tumor_Sample_Barcode)
repli_melt_icgc$Mutations<-1 #TO allow addition in next step
##Collapse summing by tumor_type the RepliScore but separating each signature
library(dplyr)

repli_melt2_25k<-list()
for (i in 1:3) {
##Repeat for each signature
repli_melt2_25k[[i]]<-subset(repli_melt_icgc,repli_melt_icgc$SBS.Sig.max%in% unique(repli_melt_icgc$SBS.Sig.max)[i]) %>% group_by(Tumor_type, SBS.Sig.max,Start_Mb,End_Mb,Chromosome) %>% summarize(Mutations_Mb = sum(Mutations))
repli_melt2_25k[[i]]<-as.data.frame(repli_melt2_25k[[i]])
colnames(repli_melt2_25k[[i]])<-c(colnames(repli_melt2_25k[[i]])[1:5],paste0("Mutations_Mb_",unique(repli_melt_icgc$SBS.Sig.max)[i]))
repli_melt2_25k[[i]]$range<-paste0(repli_melt2_25k[[i]]$Start_Mb,"-",repli_melt2_25k[[i]]$End_Mb,"-",repli_melt2_25k[[i]]$Chromosome)
}

head(repli_melt2_25k[[1]])
colnames(repli_melt2_25k[[1]])
repli_melt_total<-icgc2_allmuts[,c("Tumor_type","SBS.Sig.max","Tumor_Sample_Barcode","RepliScore_Decile","Start_Mb","End_Mb","Chromosome")]
dim(repli_melt_total)
repli_melt_total<-icgc2_allmuts[,c("Tumor_type","SBS.Sig.max","Tumor_Sample_Barcode","RepliScore_Decile","Start_Mb","End_Mb","Chromosome")]
dim(repli_melt_total)
repli_melt_total$Tumor_Sample_Barcode<-as.character(repli_melt_total$Tumor_Sample_Barcode)
repli_melt_total$Mutations<-1 #TO allow addition in next step
repli_melt_total<-repli_melt_total %>% group_by(Tumor_type,Start_Mb,End_Mb,Chromosome) %>% summarize(Mutations_Mb_all = sum(Mutations))
dim(repli_melt_total)
head(repli_melt_total)
repli_melt_total<-as.data.frame(repli_melt_total)
repli_melt_total$range<-paste0(repli_melt_total$Start_Mb,"-",repli_melt_total$End_Mb,"-",repli_melt_total$Chromosome)

repli_melt2_25k_merge<-merge(repli_melt2_25k[[1]],repli_melt2_25k[[2]],by=c("Tumor_type","Start_Mb","End_Mb","Chromosome","range"),all=T)
dim(repli_melt2_25k_merge)
head(repli_melt2_25k_merge)
head(repli_melt2_25k_merge[,-c(6,8)])
repli_melt2_25k_merge<-repli_melt2_25k_merge[,-c(6,8)]
repli_melt2_25k_merge<-merge(repli_melt2_25k_merge,repli_melt2_25k[[3]],by=c("Tumor_type","Start_Mb","End_Mb","Chromosome","range"),all=T)
nrow(repli_melt2_25k[[1]]);nrow(repli_melt2_25k[[2]]);nrow(repli_melt2_25k[[3]])
head(repli_melt2_25k_merge)
repli_melt2_25k_merge<-repli_melt2_25k_merge[,-8]
tail(repli_melt2_25k_merge)

unique(repli_melt2_25k_merge$Tumor_type)

repli_melt2_25k_merge<-merge(repli_melt2_25k_merge,repli_melt_total,by=c("Tumor_type","Start_Mb","End_Mb","Chromosome","range"),all=T)
head(repli_melt2_25k_merge)
dim(repli_melt2_25k_merge)
repli_melt2_25k_merge[is.na(repli_melt2_25k_merge)] <- 0

RepliSeq_combined_25k2$range<-paste0(RepliSeq_combined_25k2$start,"-",RepliSeq_combined_25k2$end,"-",gsub("chr","",RepliSeq_combined_25k2$seqnames))
tmp<-match(repli_melt2_25k_merge$range,RepliSeq_combined_25k2$range)
repli_melt2_25k_merge$RepliScore_Decile<-RepliSeq_combined_25k2$Decile[tmp]
#repli_melt2_25k_merge$exonic_effective_length<-RepliSeq_combined_25k2$exonic_effective_length[tmp] #Add exonic effective length to each range, for WES data
repli_melt2_25k_merge$Mutations_Mb_AICDA_canonical_density<-repli_melt2_25k_merge$Mutations_Mb_AICDA_canonical/tilewidth#*(repli_melt2_25k_merge$exonic_effective_length)
repli_melt2_25k_merge$Mutations_Mb_SBS13_density<-repli_melt2_25k_merge$Mutations_Mb_SBS13/tilewidth#*(repli_melt2_25k_merge$exonic_effective_length)
repli_melt2_25k_merge$Mutations_Mb_SBS2_density<-repli_melt2_25k_merge$Mutations_Mb_SBS2/tilewidth#*(repli_melt2_25k_merge$exonic_effective_length)
repli_melt2_25k_merge$Mutations_Mb_all_density<-repli_melt2_25k_merge$Mutations_Mb_all/tilewidth#*(repli_melt2_25k_merge$exonic_effective_length)
Tumor_type_number <-repli_melt_icgc[!duplicated(repli_melt_icgc$Tumor_Sample_Barcode), ]
Tumor_type_number<-table(Tumor_type_number$Tumor_type)
for (i in 1:length(Tumor_type_number)) {
repli_melt2_25k_merge[,c("Mutations_Mb_AICDA_canonical_density","Mutations_Mb_SBS13_density","Mutations_Mb_SBS2_density","Mutations_Mb_all_density")]<-ifelse(repli_melt2_25k_merge$Tumor_type==names(Tumor_type_number)[i],yes = repli_melt2_25k_merge[,c("Mutations_Mb_AICDA_canonical_density","Mutations_Mb_SBS13_density","Mutations_Mb_SBS2_density","Mutations_Mb_all_density")]/Tumor_type_number[i],repli_melt2_25k_merge[,c("Mutations_Mb_AICDA_canonical_density","Mutations_Mb_SBS13_density","Mutations_Mb_SBS2_density","Mutations_Mb_all_density")])
}


repli_melt2_25k_merge_list<-list()
for (i in 1:length(unique(repli_melt2_25k_merge$Tumor_type))) {
repli_melt2_25k_merge_tmp<- subset(repli_melt2_25k_merge,Tumor_type%in%unique(repli_melt2_25k_merge$Tumor_type)[i])
###Add missing ranges so its the same as the RepliSeq_combined_25k2 ranges
tmp<-setdiff(RepliSeq_combined_25k2$range,repli_melt2_25k_merge_tmp$range)
tmp<-RepliSeq_combined_25k2[RepliSeq_combined_25k2$range%in%tmp,c(1:3,6,8)]
tmp<-data.frame(Tumor_type=rep(repli_melt2_25k_merge_tmp$Tumor_type[1],nrow(tmp)),Start_Mb=tmp$start,End_Mb=tmp$end,Chromosome=gsub("chr","",tmp$seqnames),range=tmp$range)
repli_melt2_25k_merge_tmp<- bind_rows(repli_melt2_25k_merge_tmp,tmp)
repli_melt2_25k_merge_tmp[is.na(repli_melt2_25k_merge_tmp)] <- 0 #Replace NA with 0
##get with ranges to add
repli_melt2_25k_merge_list[[i]] = makeGRangesFromDataFrame(df = repli_melt2_25k_merge_tmp, start.field = "Start_Mb", end.field = "End_Mb", seqnames.field = "Chromosome",keep.extra.columns = T)
##Sort
repli_melt2_25k_merge_list[[i]]<-sort(GenomeInfoDb::sortSeqlevels( repli_melt2_25k_merge_list[[i]]))
}
repli_melt2_25k_merge_list[[1]]
########save
save(repli_melt2_25k_merge_list,file = "/media/user/seagate_ICM/AICDA_analysis/ICGC/Repliseq/ICGC_tumorMutations_25k.RData")


####Generate a global with adding all tumors
for (i in 1:length(unique(repli_melt2_25k_merge$Tumor_type))) {
   if (i==1) {
  repli_melt2_25k_all<-as.data.frame(repli_melt2_25k_merge_list[[i]])[1:121465,]
}
  
  if (i!=1) {
    repli_melt2_25k_all[1:121465,8:16]<-repli_melt2_25k_all[1:121465,8:16]+as.data.frame(repli_melt2_25k_merge_list[[i]])[1:121465,8:16]
}
  
}

repli_melt2_25k_all$Tumor_type<-"All_tumors"
repli_melt2_25k_all = makeGRangesFromDataFrame(df = repli_melt2_25k_all, start.field = "start", end.field = "end", seqnames.field = "seqnames",keep.extra.columns = T)

save(repli_melt2_25k_all,file = "/media/user/seagate_ICM/AICDA_analysis/ICGC/Repliseq/ICGC_tumorMutations_25k_ALLTUMORS.RData")

```





***

\pagebreak


# Fig 3C

```{r,echo=T,eval=F}
##Calculations
####Read data
MS_analysis_PCAWG_release_v1_RIKEN <- read_excel("Data/Figure3/Raw_Calculations/MS_analysis.PCAWG_release_v1.RIKEN.xlsx", n_max = 36)
MS_analysis_PCAWG_release_v1_RIKEN<-as.data.frame(MS_analysis_PCAWG_release_v1_RIKEN)
##THe Id is the equivalent to the "submitter-sample-id" so it is not matchable directly to the Tumor_Sample_Barcode, we need to use the pcawg_sample_sheet data
library(readxl)
MS_analysis_PCAWG_release_v1_RIKEN$Tumor_Sample_Barcode<-pcawg_sample_sheet$aliquot_id[match(MS_analysis_PCAWG_release_v1_RIKEN$ID,pcawg_sample_sheet$submitter_sample_id)]##Add Tumor_Sample_Barcode to MS_analysis_PCAWG_release_v1_RIKEN

MS_analysis_PCAWG_release_v1_RIKEN$icgc_sample_id<-pcawg_sample_sheet$icgc_sample_id[match(MS_analysis_PCAWG_release_v1_RIKEN$ID,pcawg_sample_sheet$submitter_sample_id)]##icgc_sample_id since some sample have the "Tumor_Sample_Barcode" as icgc_sample_id

#subset to only MSI postive
length(intersect(MS_analysis_PCAWG_release_v1_RIKEN$Tumor_Sample_Barcode[MS_analysis_PCAWG_release_v1_RIKEN$MSI%in%"positive"],repli_melt_icgc3_deciles2$Tumor_Sample_Barcode)) #Only 8

length(intersect(MS_analysis_PCAWG_release_v1_RIKEN$icgc_sample_id[MS_analysis_PCAWG_release_v1_RIKEN$MSI%in%"positive"],repli_melt_icgc3_deciles2$Tumor_Sample_Barcode))#25

###Add a tag to MSI tumor using repli_melt_icgc3_deciles2
tmp<-intersect(MS_analysis_PCAWG_release_v1_RIKEN$Tumor_Sample_Barcode[MS_analysis_PCAWG_release_v1_RIKEN$MSI%in%"positive"],repli_melt_icgc3_deciles2$Tumor_Sample_Barcode)


repli_melt_icgc3_deciles2$MSI<-ifelse(repli_melt_icgc3_deciles2$Tumor_Sample_Barcode%in%tmp,"MSI","MSS")

#For the others..
tmp<-intersect(MS_analysis_PCAWG_release_v1_RIKEN$icgc_sample_id[MS_analysis_PCAWG_release_v1_RIKEN$MSI%in%"positive"],repli_melt_icgc3_deciles2$Tumor_Sample_Barcode)

repli_melt_icgc3_deciles2$MSI[repli_melt_icgc3_deciles2$Tumor_Sample_Barcode%in%tmp]<-"MSI"
repli_melt_icgc3_deciles2[repli_melt_icgc3_deciles2$MSI%in%"MSI",]
##Only present within AICDA signature
##Subset to tumor types in which there is MSI tumors
repli_melt_icgc3_deciles2_msi<-subset(repli_melt_icgc3_deciles2,repli_melt_icgc3_deciles2$Tumor_type%in%repli_melt_icgc3_deciles2$Tumor_type[repli_melt_icgc3_deciles2$MSI%in%"MSI"])

##Make a graph to see the differences in mutational charge across replication times

##Test normality to see which test to perform, shapiro where null hypho= data is normally distributed
shapiro.test(repli_melt_icgc3_deciles2_msi$Mutations_Mb_decile_zone[repli_melt_icgc3_deciles2_msi$MSI%in%"MSI"]) #Not normal
qqnorm(repli_melt_icgc3_deciles2_msi$Mutations_Mb_decile_zone[repli_melt_icgc3_deciles2_msi$MSI%in%"MSI"])
shapiro.test(repli_melt_icgc3_deciles2_msi$Mutations_Mb_decile_zone[repli_melt_icgc3_deciles2_msi$MSI%in%"MSS"]) #Not normal
qqnorm(repli_melt_icgc3_deciles2_msi$Mutations_Mb_decile_zone[repli_melt_icgc3_deciles2_msi$MSI%in%"MSS"])


```

```{r panel_3C, fig.width=19,fig.height=16}
#
##For main
library(ggpubr)
library(reshape2)
##Import data
data_dir<-"/media/user/seagate_ICM/AICDA_analysis/Github_repository/" #Set to where you downloaded the data
setwd(data_dir)
##Import needed data calculated in previous chunk
repli_melt_icgc3_deciles2<-as.data.frame(read.delim("Data/Figure3/Figure3A.tab"))
##Subset to tumor types in which there is MSI tumors
repli_melt_icgc3_deciles2_msi<-subset(repli_melt_icgc3_deciles2,repli_melt_icgc3_deciles2$Tumor_type%in%repli_melt_icgc3_deciles2$Tumor_type[repli_melt_icgc3_deciles2$MSI%in%"MSI"])


AICDA_MSI_repli<-ggboxplot(data = subset(repli_melt_icgc3_deciles2_msi,repli_melt_icgc3_deciles2_msi$SBS.Sig.max%in%"AICDA_canonical"),x ="Replication_Zone" , y = "Mutations_Mb_decile_zone",add = "jitter",color = "MSI")+ theme_classic()+stat_compare_means(method = "wilcox.test", aes(group = MSI),label = "p.format",label.x = 1.5)+labs( x="Replication time", y = "Mutation number mean per Mb")+ guides(color=guide_legend(title = "MSI status"))+theme(plot.title = element_text(hjust = 0.5,face="bold"))

AICDA_MSI_repli2<-ggboxplot(data = subset(repli_melt_icgc3_deciles2_msi,repli_melt_icgc3_deciles2_msi$SBS.Sig.max%in%"AICDA_canonical"),x ="Replication_Zone" , y = "Mutations_Mb_decile_zone",add = "jitter",color = "Replication_Zone",facet.by = "MSI")+ theme_classic()+stat_compare_means(method = "wilcox.test", aes(group = Replication_Zone),label = "p.format",label.x = 1.5)+labs( x="Replication time", y = "Mutation number mean per Mb")+ guides(color=guide_legend(title = "Replication Zone"))+theme(plot.title = element_text(hjust = 0.5,face="bold"))
  
AICDA_MSI_repli3<-ggarrange(AICDA_MSI_repli,AICDA_MSI_repli2+theme(axis.title.y = element_blank(),axis.line.y =  element_blank(),axis.text.y = element_blank(),axis.ticks.y = element_blank()),ncol = 2,align = "hv",common.legend = T)+theme(plot.title = element_text(hjust = 0.5,face="bold"),title = element_text("AICDA signature"))

##Now for SBS2, not included in article

SBS2_MSI_repli<-ggboxplot(data = subset(repli_melt_icgc3_deciles2_msi,repli_melt_icgc3_deciles2_msi$SBS.Sig.max%in%"SBS2"),x ="Replication_Zone" , y = "Mutations_Mb_decile_zone",add = "jitter",color = "Replication_Zone",facet.by = "MSI")+ theme_classic()+stat_compare_means(method = "wilcox.test", aes(group = Replication_Zone),label = "p.format",label.x = 1.5)+labs( x="Replication time", y = "Mutation number mean per Mb")+ guides(color=guide_legend(title = "Replication Zone"))+theme(plot.title = element_text(hjust = 0.5,face="bold"))

  


##Now SBS13, not included in the article
SBS13_MSI_repli<-ggboxplot(title = "MSS tumors",data = subset(repli_melt_icgc3_deciles2_msi,repli_melt_icgc3_deciles2_msi$SBS.Sig.max%in%c("SBS2","SBS13")),x ="Replication_Zone" , y = "Mutations_Mb_decile_zone",add = "jitter",color = "Replication_Zone",facet.by = "SBS.Sig.max")+ theme_classic()+stat_compare_means(method = "wilcox.test", aes(group = Replication_Zone),label = "p.format",label.x = 1.5)+labs( x="Replication time", y = "Mutation number mean per Mb")+ guides(color=guide_legend(title = "Replication Zone"))+theme(plot.title = element_text(hjust = 0.5,face="bold"))


#Plot
ggarrange(AICDA_MSI_repli,AICDA_MSI_repli2,ncol = 2)

```

***

\pagebreak

# Fig 3D

```{r panel_3D_calculs,echo=T,eval=F}
#Download composite mutations data from http://download.cbioportal.org/composite_mutations_maf.txt.gz

#Other data avaliable at: https://github.com/taylor-lab/composite-mutations/tree/master/data

Composite_maf<-read.maf(maf = "/media/user/seagate_ICM/AICDA_analysis/Composite_mutations/data/data_mutations.txt.gz")
length(unique(Composite_maf@data$Tumor_Sample_Barcode))
#31353
Composite_maf<-subsetMaf(Composite_maf,tsb = unique(Composite_maf@data$Tumor_Sample_Barcode)) #Eliminate duplicates
length(table(Composite_maf@data$metamaintype))#41 tumor types

##Calculate AICDA and APOBEC mutations


##Add data to clinical
Composite_clinical <- fread(here('data/data_clinical.txt.gz'))
Composite_clinical<-subset(Composite_clinical,Composite_clinical$Tumor_Sample_Barcode%in%unique(Composite_maf@data$Tumor_Sample_Barcode))#Subset


load("/media/user/seagate_ICM/AICDA_analysis/ICGC/APOBEC_enrich_function_WES_WGS.RData")
load("/media/user/seagate_ICM/AICDA_analysis/ICGC/AICDA_enrich_function_WES_WGS.RData")
Composite_AICDA<-AICDA_enrichment(Composite_maf, ref_genome = "BSgenome.Hsapiens.UCSC.hg19",prefix = 'chr', add = TRUE, background_n = 60,seq_type = "WGS") #Add the WGS flag to get silent mutations too

####Boxplot of AICDA mutations_fraction of mutations....Add this data to clinical data and also AICDA_enrichment score
dim(Composite_clinical) #
Composite_clinical$fraction_AICDA_mutations<-0
Composite_clinical$fraction_AICDA_mutations<-Composite_AICDA$AICDA_scores$fraction_AICDA_mutations[match(Composite_clinical$Tumor_Sample_Barcode,Composite_AICDA$AICDA_scores$Tumor_Sample_Barcode)]##Add


Composite_AICDA$AICDA_scores$AICDA_mutations<-Composite_AICDA$AICDA_scores$n_mutations-Composite_AICDA$AICDA_scores$non_AICDA_mutations
Composite_clinical$AICDA_mutations<-Composite_AICDA$AICDA_scores$AICDA_mutations[match(Composite_clinical$Tumor_Sample_Barcode,Composite_AICDA$AICDA_scores$Tumor_Sample_Barcode)]

Composite_APO<-APOBEC_enrichment(Composite_maf, ref_genome = "BSgenome.Hsapiens.UCSC.hg19",prefix = 'chr', add = TRUE, seq_type = "WGS") #Add the WGS fla

Composite_clinical$fraction_APOBEC_mutations<-0
Composite_clinical$fraction_APOBEC_mutations<-Composite_APO$APOBEC_scores$fraction_APOBEC_mutations[match(Composite_clinical$Tumor_Sample_Barcode,Composite_APO$APOBEC_scores$Tumor_Sample_Barcode)]##Add


Composite_clinical<-Composite_clinical[!is.na(Composite_clinical$fraction_AICDA_mutations),]##Subset since they have low mutation number...
#Subset
###Graph to see distribution of AICDA mutations and APOBEC
#Melt
t<-melt(Composite_clinical[],id.vars=c('metamaintype',"msi_high","mmr_signature"), measure.vars=c('fraction_AICDA_mutations','fraction_APOBEC_mutations'))

#Add line global median, add n samples to labels...

Composite_fractionMuts<-
  ggboxplot(data = t, y ="value", x="metamaintype",color = "variable",xlab = "Tumor type",ylab = "Fraction of SNP mutations",title = "Gorelick et al. study")+theme(axis.text.x = element_text(angle = 90,size=8),axis.title = element_text(face="bold"),plot.title = element_text(hjust = 0.5,face = "bold",size = 15))+ geom_hline(yintercept=mean(Composite_clinical$fraction_AICDA_mutations,na.rm = T), linetype="dashed",color="#F8766D")+ geom_hline(yintercept=mean(Composite_clinical$fraction_APOBEC_mutations,na.rm = T), linetype="dashed",color="#00BFC4")+ guides(color=guide_legend(title = "Mutation type"))+scale_color_discrete(labels=c(paste0("AICDA"," (mean = ", round(mean(Composite_clinical$fraction_AICDA_mutations,na.rm = T),3),")"),paste0("APOBEC"," (mean = ", round(mean(Composite_clinical$fraction_APOBEC_mutations,na.rm = T),3),")")))+scale_x_discrete(labels=paste0(unique(t$metamaintype)," (n=", table(Composite_clinical$metamaintype)[unique(t$metamaintype)],")"))

#That plot is not included in the study 




#

##Now subset to MSI tumors and plot ggplot comparing per tumor_type the fractions of AICDA and APOBEC mutations...

##Edit a little the data
t2<-subset(t,!is.na(t$msi_high))
t2$msi_high<-ifelse(t2$msi_high=="TRUE","MSI","MSS")
t2$Compare<-paste0(t2$msi_high,t2$variable)
# reorder the groups order : I change the order of the factor data$names
t2$Compare <- factor(t2$Compare , levels=c("MSSfraction_AICDA_mutations", "MSIfraction_AICDA_mutations", "MSSfraction_APOBEC_mutations", "MSIfraction_APOBEC_mutations"))
##Take out tumor types with less than 4 MSI samples...
take_out<-table(Composite_clinical$metamaintype[Composite_clinical$msi_high%in%"TRUE"&Composite_clinical$AICDA_mutations>0])[table(Composite_clinical$metamaintype[Composite_clinical$msi_high%in%"TRUE"&Composite_clinical$AICDA_mutations>0])>=4]
t2<-subset(t2,t2$metamaintype%in%names(take_out))


                                                                               


```

```{r panel_3D, fig.width=14,fig.height=12}
#Import data
Composite_clinical<-as.data.frame(read.delim("/media/user/seagate_ICM/AICDA_analysis/Github_repository/Data/Figure3/Figure3D_clinical.tab"))
t2<-as.data.frame(read.delim("/media/user/seagate_ICM/AICDA_analysis/Github_repository/Data/Figure3/Figure3D.tab"))

p_tmp<-ggboxplot(data = t2, y ="value", x="metamaintype",color = "Compare",xlab = "Tumor type",ylab = "Fraction of SNP mutations",title = "Gorelick et al. study")+theme(axis.text.x = element_text(angle = 90,size=8),axis.title = element_text(face="bold"),plot.title = element_text(hjust = 0.5,face = "bold",size = 15))+ geom_hline(yintercept=mean(t2$value[t2$Compare%in%"MSIfraction_AICDA_mutations"],na.rm = T), linetype="dashed",color="#F8766D")+ geom_hline(yintercept=mean(t2$value[t2$Compare%in%"MSIfraction_APOBEC_mutations"],na.rm=T), linetype="dashed",color="#00BFC4")+ guides(color=guide_legend(title = "Mutation type"))+scale_color_discrete(labels=c(paste0("AICDA MSS"," (mean = ", round(mean(t2$value[t2$Compare%in%"MSSfraction_AICDA_mutations"],na.rm = T),3),")"), paste0("AICDA MSI"," (mean = ", round(mean(t2$value[t2$Compare%in%"MSIfraction_AICDA_mutations"],na.rm = T),3),")"),paste0("APOBEC MSS"," (mean = ", round(mean(t2$value[t2$Compare%in%"MSSfraction_APOBEC_mutations"],na.rm = T),3),")"),paste0("APOBEC MSI"," (mean = ", round(mean(t2$value[t2$Compare%in%"MSIfraction_APOBEC_mutations"],na.rm = T),3),")"))) +scale_x_discrete(labels=paste0(unique(t2$metamaintype)," (n=",table(Composite_clinical$metamaintype)[unique(t2$metamaintype)],")"))

###Save legend
lg_msi<-get_legend(p_tmp, position = NULL)

##AICDA
t2$msi_high <- factor(t2$msi_high , levels=c("MSS","MSI"))

msi_ACIDA_compo<-ggboxplot(data = subset(t2,t2$variable%in%"fraction_AICDA_mutations"), y ="value", x="metamaintype",color = "msi_high",xlab = "Tumor type",ylab = "Fraction of SNP mutations",title = "Gorelick et al. study",palette = c("#F8766D","#7CAE00"))+scale_x_discrete(labels=paste0(unique(t2$metamaintype[t2$variable%in%"fraction_AICDA_mutations"])," (n=",table(t2$metamaintype[t2$variable%in%"fraction_AICDA_mutations"])[unique(t2$metamaintype[t2$variable%in%"fraction_AICDA_mutations"])],")"))+stat_compare_means(method = "wilcox.test", aes(group=msi_high),label.y = 1.01,label = "p.signif",label.x = 1.5,hide.ns=T)+theme(axis.title.x=element_blank(),axis.text.x=element_blank(),axis.ticks.x=element_blank(),plot.title = element_text(hjust = 0.5,face = "bold",size = 15),axis.title.y = element_text(face="bold"))+ annotate("text", x = 7.5, y = 1.06, label = paste0("global pval ",compare_means(value ~ msi_high, data = t2[t2$variable%in%"fraction_AICDA_mutations",])[6]," (Mann-Whitney U-test)"))
##Add global p value
##Add global p value
    #+ geom_hline(yintercept=mean(t2$value[t2$Compare%in%"MSIfraction_AICDA_mutations"],na.rm = T), linetype="dashed",color="#7CAE00")#+geom_hline(yintercept=mean(t2$value[t2$Compare%in%"MSSfraction_AICDA_mutations"],na.rm = T), linetype="dashed",color="#F8766D")

 
#compare_means(value ~ msi_high, data = t2[t2$variable%in%"fraction_AICDA_mutations",])
###APOBEC                                                                                                       
                                                                                                                                               msi_Apo_compo<-ggboxplot(data = subset(t2,t2$variable%in%"fraction_APOBEC_mutations"), y ="value", x="metamaintype",color = "msi_high",xlab = "Tumor type",ylab = "Fraction of SNP mutations",palette = c("#00BFC4","#C77CFF"))+theme(axis.text.x = element_text(angle = 90,size=8),axis.title = element_text(face="bold"),plot.title = element_text(hjust = 0.5,face = "bold",size = 15))+scale_x_discrete(labels=paste0(unique(t2$metamaintype[t2$variable%in%"fraction_APOBEC_mutations"])," (n=",table(t2$metamaintype[t2$variable%in%"fraction_APOBEC_mutations"])[unique(t2$metamaintype[t2$variable%in%"fraction_APOBEC_mutations"])],")"))+stat_compare_means(method = "wilcox.test", aes(group=msi_high),label = "p.signif",label.y = 1.01,hide.ns=T)+ annotate("text", x = 7.5, y = 1.06, label = paste0("global pval = ",compare_means(value ~ msi_high, data = t2[t2$variable%in%"fraction_APOBEC_mutations",])[6]," (Mann-Whitney U-test)"))
##Add global p value
#+ geom_hline(yintercept=mean(t2$value[t2$Compare%in%"MSIfraction_APOBEC_mutations"],na.rm = T), linetype="dashed",color="#C77CFF")+ geom_hline(yintercept=mean(t2$value[t2$Compare%in%"MSSfraction_APOBEC_mutations"],na.rm = T), linetype="dashed",color="#00BFC4")                                                                                                                                               

#compare_means(value ~ msi_high, data = t2[t2$variable%in%"fraction_APOBEC_mutations",])
##Merge
                                                                                                                                               
p_msi_compo<-ggarrange(msi_ACIDA_compo,msi_Apo_compo,nrow = 2,heights = c(1,1.5),legend.grob = lg_msi,legend = "bottom")                             

p_msi_compo

```


***

\pagebreak


# Fig 3E

Generated in python, check Fi3E_script.py.

***

\pagebreak

# Fig 3F

```{r,echo=T,eval=F}
##Calculations
##Import expression data
icgc_exp <- read.delim("Data/Figure3/Raw_Calculations/tophat_star_fpkm_uq.v2_aliquot_gl.tsv.gz", header=FALSE) ##Consult the manuscript to download this file from the original study, obtained from: https://dcc.icgc.org/api/v1/download?fn=/PCAWG/transcriptome/gene_expression/tophat_star_fpkm_uq.v2_aliquot_gl.tsv.gz 
#1522 samples only...
colnames(icgc_exp)<-icgc_exp[1,]
icgc_exp<-icgc_exp[-1,]
colnames(icgc_exp)[1]<-"Gene"
##Transform ensbl id to Hugo_symbol
###Collect gene names
library(biomaRt)
mart=useMart(biomart="ensembl", dataset="hsapiens_gene_ensembl")
mart_hg<-biomaRt::getBM(attributes = c("ensembl_gene_id","ensembl_gene_id_version","external_gene_name"),mart = mart)
##Add gene_symbol column
icgc_exp$Gene_symbol<-mart_hg$external_gene_name[match(icgc_exp$Gene,mart_hg$ensembl_gene_id_version)]
length(unique(icgc_exp$Gene_symbol))
[1] 25288
#Eliminate missing
icgc_exp<-icgc_exp[!is.na(icgc_exp$Gene_symbol),]

###COllapse repeated genes (because of different id.version)
#There are 25288 unique genes, keep mean values from each gene for each patient
#icgc_exp2 <- aggregate(. ~ Gene_symbol, data = icgc_exp, median)
icgc_exp<-icgc_exp[!duplicated(icgc_exp$Gene_symbol),]
rownames(icgc_exp)<-icgc_exp$Gene_symbol
icgc_exp<-icgc_exp[,-c(1,ncol(icgc_exp))] #Eliminate Gene and Gene_symbol columns

##(sample can either be the aliquot_id or library)
####Write Tumor sample barcode instead of aliquot_id to match the other sets...
length(intersect(pcawg_sample_sheet$aliquot_id,colnames(icgc_exp))) #1466/1521 in common
pcawg_specimen_histology_August2016_v9$aliquot_id<-pcawg_sample_sheet$aliquot_id[match(pcawg_specimen_histology_August2016_v9$donor_unique_id,pcawg_sample_sheet$donor_unique_id)]
pcawg_sample_sheet$Tumor_Sample_Barcode<-pcawg_specimen_histology_August2016_v9$Tumor_Sample_Barcode[match(pcawg_sample_sheet$donor_unique_id,pcawg_specimen_histology_August2016_v9$donor_unique_id)]

icgc_exp<-icgc_exp[,colnames(icgc_exp)%in%pcawg_sample_sheet$aliquot_id] #SUbset
#Change colnames to Tumor_Sample_Barcode
colnames(icgc_exp)<-pcawg_sample_sheet$Tumor_Sample_Barcode[match(colnames(icgc_exp),pcawg_sample_sheet$aliquot_id)]

#Take out duplicates
icgc_exp<-icgc_exp[,!duplicated(colnames(icgc_exp))]

##Subset clinical to same samples
ICGC_clinical_expr<-subset(ICGC_clinial_n3110[!duplicated(ICGC_clinial_n3110$Tumor_Sample_Barcode2),],ICGC_clinial_n3110$Tumor_Sample_Barcode2 %in%colnames(icgc_exp))
ICGC_clinical_expr<-ICGC_clinical_expr[ICGC_clinical_expr$Tumor_Sample_Barcode2%in%ICGC_clinial_palim_annot$Sample,] #Samples included in the muts
###Now subset expression to those samples too
icgc_exp<-icgc_exp[,colnames(icgc_exp)%in%ICGC_clinical_expr$Tumor_Sample_Barcode2]
ICGC_clinical_expr<-ICGC_clinical_expr[ICGC_clinical_expr$Tumor_Sample_Barcode2%in%colnames(icgc_exp),]

nrow(ICGC_clinical_expr)==ncol(icgc_exp) #Check dimensions are equal, 1130 samples

##How many tumor types
length(table(ICGC_clinical_expr$histology_abbreviation))
#24
##SUbset mutations to same samples as expression data

length(unique(icgc2_allmuts_clonality$Sample))
[1] 2707
###
length(unique(icgc2_allmuts_clonality$Sample[icgc2_allmuts_clonality_loops$Sample%in%ICGC_clinical_expr$Tumor_Sample_Barcode2]))
[1] 1130

icgc2_allmuts_clonality_exp<-icgc2_allmuts_clonality[icgc2_allmuts_clonality$Sample%in%ICGC_clinical_expr$Tumor_Sample_Barcode2,]

########
#########
####Perform correlations...
##Transpose expression data
icgc_exp<-t(icgc_exp)

###Get the number of AICDA mutations per sample, per gene, per tumor...
rna_melt<-icgc2_allmuts_clonality_exp[icgc2_allmuts_clonality_exp$AICDA_mutation=="YES",] %>% group_by(Sample,Hugo_Symbol,Tumor_type) %>% dplyr::summarise(Mutations_Mb = sum(Mutations))
rna_melt<-as.data.frame(rna_melt)
head(rna_melt)
dim(rna_melt)

length(unique(rna_melt$Hugo_Symbol))
[1] 21341 ##Mutated genes due AICDA

21341/34596
[1] 0.6168632 #Fraction genes touched by AICDA

###SUbset expression of genes to the ones also appearing in at least one sample
icgc_exp2<-icgc_exp[,colnames(icgc_exp)%in%unique(rna_melt$Hugo_Symbol)] ##only 1403 genes!!


1403/21341###Fraction of mutated genes being expressed

1403/25287##Fraction of expressed genes being mutated

##Add tumor type to exp
icgc_exp2<-as.data.frame(icgc_exp2)

icgc_exp2$Tumor_type<-rna_melt$Tumor_type[match(rownames(icgc_exp2),rna_melt$Sample)]


###Put expression and mutations in the same sample order

###%
tumors_list<-table(ICGC_clinical_expr$histology_abbreviation)
##Table to replace results
correlations_table<-as.data.frame(matrix(data=NA,nrow=ncol(icgc_exp2),ncol = 48,dimnames = list(colnames(icgc_exp2),c(paste0("p_val_",names(tumors_list)),paste0("Rho_",names(tumors_list))))))
correlations_table<-correlations_table[-nrow(correlations_table),]
for (i in 1:length(tumors_list)) {
  ##subset data to tumor type [i]
  icgc_exp2_tmp<-icgc_exp2[icgc_exp2$Tumor_type%in%names(tumors_list)[i],]
  rna_melt_tmp<-rna_melt[rna_melt$Tumor_type%in%names(tumors_list)[i],]
  ICGC_clinical_expr_tmp<-ICGC_clinical_expr[ICGC_clinical_expr$histology_abbreviation%in%names(tumors_list)[i],]
  
  ##Other cycle to do by gene
  for (j in 1:(ncol(icgc_exp2)-1)) {
    ICGC_clinical_expr_tmp$exp_tmp<-icgc_exp2_tmp[match(ICGC_clinical_expr_tmp$Tumor_Sample_Barcode2,rownames(icgc_exp2_tmp)),j] #Add expression Gene j
    rna_melt_tmp2<-rna_melt_tmp[rna_melt_tmp$Hugo_Symbol%in%colnames(icgc_exp2)[j],]
    
    ICGC_clinical_expr_tmp$mut_tmp<-rna_melt_tmp2$Mutations_Mb[match(ICGC_clinical_expr_tmp$Tumor_Sample_Barcode2,rna_melt_tmp2$Sample)] #Add muts
    ICGC_clinical_expr_tmp$mut_tmp[is.na(ICGC_clinical_expr_tmp$mut_tmp)]<-0##Replace with 0 in case there were not mutations detected
    test_tmp<- cor.test(x=as.numeric(ICGC_clinical_expr_tmp$exp_tmp), y =as.numeric(ICGC_clinical_expr_tmp$mut_tmp),method = "spearman" ,na.action="na.omit")

  correlations_table[j,i]<-test_tmp$p.value

    correlations_table[j,i+24]<-test_tmp$estimate

    
  }
 
}

####Modify data for  plotting
correlations_table<-correlations_table[rowSums(is.na(correlations_table)) != ncol(correlations_table), ] #remove empty rows

correlations_table_pval<-correlations_table[,1:24]
correlations_table_rho<-correlations_table[,25:48]
colnames(correlations_table_pval)<-names(tumors_list)
colnames(correlations_table_rho)<-names(tumors_list)

correlations_table_pval<-as.data.frame(t(correlations_table_pval))
correlations_table_pval$Tumor_type<-rownames(correlations_table_pval)

t<-melt(correlations_table_pval[],id.vars=c('Tumor_type'), measure.vars=c(rownames(correlations_table)))
colnames(t)<-c("Tumor_type","Gene","p_val")##Change colnames

correlations_table_rho<-as.data.frame(t(correlations_table_rho))
correlations_table_rho$Tumor_type<-rownames(correlations_table_rho)

t2<-melt(correlations_table_rho[],id.vars=c('Tumor_type'), measure.vars=c(rownames(correlations_table)))
colnames(t2)<-c("Tumor_type","Gene","Rho")

correlations_table2<-merge(t,t2, by = c("Tumor_type","Gene"))

###Take out missing values on p_val
correlations_table2<-correlations_table2[!is.na(correlations_table2$p_val),]

genes_associated<-correlations_table2$Gene[correlations_table2$p_val<0.05&correlations_table2$Rho>0]###Number of genes with significant correlation and positive

length(unique(genes_associated)) #only 81 genes
genes_associated[duplicated(genes_associated)] # 3 genes in more than one tumor type

##This is a supplementary Table
write.table(unique(genes_associated),file = "/media/user/seagate_ICM/AICDA_analysis/ICGC/Expression_muts_correlation/genes_AICDA_trans_associated.tab",row.names = F,quote = F,sep = "\t",col.names = F)
#################
##############

correlations_table2$DE<-ifelse(correlations_table2$p_val<0.05&correlations_table2$Rho>0,"Associated","Not associated")
correlations_table2$label<-paste0(correlations_table2$Gene,"-",correlations_table2$Tumor_type)
rownames(correlations_table2)<-seq(1:nrow(correlations_table2))
#Value that are 0 in pval change to a very small value
correlations_table2$p_val[correlations_table2$p_val==0]<-0.00000001


###ORA analysis using DAVID online...of genes
#Import results
DAVID_ORA_analysis <- read.delim("/media/user/seagate_ICM/AICDA_analysis/ICGC/Expression_muts_correlation/DAVID_ORA_analysis.txt")
immu_ora<-gsub((strsplit(DAVID_ORA_analysis$Genes[1], '\\s+')[[1]]),pattern = ',',replacement = "")
sensory_ora<-gsub((strsplit(DAVID_ORA_analysis$Genes[30], '\\s+')[[1]]),pattern = ',',replacement = "")
correlations_table2$ORA<-ifelse(correlations_table2$Gene%in%immu_ora&correlations_table2$p_val<0.05&correlations_table2$Rho>0,"IGv region",ifelse(correlations_table2$Gene%in%sensory_ora&correlations_table2$p_val<0.05&correlations_table2$Rho>0,"Sensory transduction",ifelse(correlations_table2$p_val<0.05,"Significant","Not significant")))


```


```{r panel_3F, fig.width=19,fig.height=11}
#
#Import data
correlations_table2<-as.data.frame(read.delim("/media/user/seagate_ICM/AICDA_analysis/Github_repository/Data/Figure3/Figure3F.tab"))
DAVID_ORA_analysis <- read.delim("/media/user/seagate_ICM/AICDA_analysis/Github_repository/Data/Figure3/Raw_Calculations/DAVID_ORA_analysis.txt")
###Volcano plot
library(ggrepel)

p_enrich_transcription<-ggplot(data=correlations_table2, aes(y=-log10(p_val), x=Rho, col=ORA,shape=DE)) +geom_point() +theme_std(base_size=14) + scale_color_manual(name="GSEA enrichment",values = c("red","grey","blue","black"),labels=c(paste0("IGv region (p.val=",scientific(DAVID_ORA_analysis$FDR[1],digits = 2),")" ),"NS",paste0("Sensory transduction (p.val=",scientific(DAVID_ORA_analysis$FDR[30],digits = 2),")" ),"Other"))  +
    geom_hline(yintercept=-log10(0.05), col="black",linetype="dotted") +
    geom_vline(xintercept=0, col="black",linetype="dotted")+xlab("Spearman Rho")+theme(legend.position = "right",legend.box = "vertical") +geom_text_repel(data = correlations_table2[correlations_table2$ORA%in%c("IGv region","Sensory transduction"),],aes(label = label,x=Rho, y=-log10(p_val)),  size = 5, box.padding = unit(0.35, "lines"),point.padding = unit(0.3, "lines"),inherit.aes = F)+ylim(c(0,15))+guides(color = guide_legend(order=2),shape = guide_legend(order=1))+ theme(legend.title = element_text(colour="black", size=10,face="bold"),legend.text = element_text(colour="black", size=8))+scale_shape_manual(name="Transcription associated",values = c(17,16),labels=c(paste0("Associated (n=",table(correlations_table2$DE)[1],")" ),paste0("Not associated (n=",table(correlations_table2$DE)[2],")" )))


###Pie chart as inset..

pie_dat<-data.frame(Tumor_type=names(table(correlations_table2$Tumor_type[correlations_table2$p_val<0.05&correlations_table2$Rho>0])),values=table(correlations_table2$Tumor_type[correlations_table2$p_val<0.05&correlations_table2$Rho>0]))
pie_dat$prop<-pie_dat$values.Freq/sum(pie_dat$values.Freq)
# Add label position
pie_dat <- pie_dat %>%
  arrange(desc(Tumor_type)) %>%
  mutate(lab.ypos = cumsum(prop) - 0.4*prop)
pie_dat
pie_dat$prop<-round(pie_dat$prop,digits = 2)
pie_dat$prop_labl<-paste0(pie_dat$prop*100,"%")

  
color = grDevices::colors()[grep('gr(a|e)y', grDevices::colors(), invert = T)]
cols<-sample(color, nrow(pie_dat))
             
p_pie<-ggplot(pie_dat, aes(x = "", y = prop, fill = Tumor_type)) +
    geom_bar(width = 1, stat = "identity", color = "white") +
    coord_polar("y", start = 0)+
    geom_text_repel(aes(y = lab.ypos, label = prop_labl,x = 1.5), color = "black", 
                    segment.size = .4,nudge_x = .05, 
                    show.legend = FALSE)+
    theme_void()+scale_fill_manual(values = cols,name="Tumor type")+theme(legend.title = element_text(size = 9,face="bold"),legend.key.size = unit(0.5, "cm"),legend.text = element_text(size = 9))+ guides(fill = guide_legend(ncol = 2)) 



###Insert inset
##Merge inside the other
##Grab
# called a "grop" in Grid terminology
xbp_grob <- ggplotGrob(p_pie)

p_enrich_transcription2<-p_enrich_transcription + annotation_custom(grob = xbp_grob, xmin = 0, xmax = 2.15,  ymin = 11, ymax = 15)+annotate("segment", x=0.7, y=15, xend=1.45, yend=15,col="black", linetype="dashed")+annotate("segment", x=0.7, y=11, xend=1.45, yend=11,col="black", linetype="dashed")+annotate("segment", x=0.7, y=15, xend=.7, yend=11,col="black", linetype="dashed")+annotate("segment", x=1.45, y=15, xend=1.45, yend=11,col="black", linetype="dashed")+annotate("segment", x=1.125, y=9.5, xend=1.425, yend=9.5,col="black", linetype="dashed")+annotate("segment", x=1.125, y=8, xend=1.425, yend=8,col="black", linetype="dashed")+annotate("segment", x=1.125, y=8, xend=1.125, yend=9.5,col="black", linetype="dashed")+annotate("segment", x=1.425, y=8, xend=1.425, yend=9.5,col="black", linetype="dashed")+annotate("segment", x=.7, y=11, xend=1.125, yend=9.5,col="black", linetype="dashed")+annotate("segment", x=1.425, y=9.5, xend=1.45, yend=11,col="black", linetype="dashed")+coord_cartesian(xlim = c(-1, 1), clip="off",ylim=c(0,15))



p_enrich_transcription2+theme(plot.margin = unit(c(0,2,0,0), "cm"))


########
#####Plot for "pipeline"


```

***

\pagebreak


# Fig 3G

```{r,echo=T,eval=F}
###Needed to have run previuos chunks
###Clonality calculations
###Import Sex data
sex_data<-read.delim("Data/Figure3/Raw_Calculations/donor.tsv")
##2734 patients
##Add this Tumor_Sample_Barcode tag to be able to match with the previous clinical information
sex_data$Tumor_Sample_Barcode<-pcawg_specimen_histology_August2016_v9$Tumor_Sample_Barcode[match(paste0(sex_data$icgc_donor_id,sex_data$submitted_donor_id),paste0(pcawg_specimen_histology_August2016_v9$icgc_donor_id,pcawg_specimen_histology_August2016_v9$submitted_donor_id))]##Add Tumor_Sample_Barcode to pcawg_specimen_histology_August2016_v9
pcawg_specimen_histology_August2016_v9$Gender<-sex_data$donor_sex[match(pcawg_specimen_histology_August2016_v9$Tumor_Sample_Barcode,sex_data$Tumor_Sample_Barcode)]
####Write other Tumor_Sample_Barcode code since previous one was not correctly assigned so it matches cnv data
ICGC_clinial_n3110$Tumor_Sample_Barcode2<-pcawg_specimen_histology_August2016_v9$Tumor_Sample_Barcode[match(paste0(ICGC_clinial_n3110$icgc_sample_id,ICGC_clinial_n3110$icgc_donor_id),paste0(pcawg_specimen_histology_August2016_v9$icgc_sample_id,pcawg_specimen_histology_August2016_v9$icgc_donor_id))]

#Add to clinical, only avaliable for 2532 samples...
ICGC_clinial_n3110$Gender<-pcawg_specimen_histology_August2016_v9$Gender[match(ICGC_clinial_n3110$Tumor_Sample_Barcode2,pcawg_specimen_histology_August2016_v9$Tumor_Sample_Barcode)]
###There is low number of samples having gender data which is used to calculated clonality upon sex linked chromosomes, assign arbitrary but dont include chromosome X or Y in the analysis....
ICGC_clinial_n3110$Gender[is.na(ICGC_clinial_n3110$Gender)]<-"female" 
ICGC_clinial_n3110$Gender[ICGC_clinial_n3110$Gender==""]<-"female"  #Empty data

###Import purity and ploidy, download from: https://dcc.icgc.org/releases/PCAWG/consensus_cnv/consensus.20170217.purity.ploidy.txt.gz 
Purity_ploidy<-read.delim("Data/Figure3/Raw_Calculations/consensus.20170217.purity.ploidy.txt.gz")

#Add to clinical
ICGC_clinial_n3110$Purity<-Purity_ploidy$purity[match(ICGC_clinial_n3110$Tumor_Sample_Barcode2,Purity_ploidy$samplename)]
##2778 samples only....
##Import CN data...it was first merge by sample...
#Download CN data from ICGC portal..from: https://dcc.icgc.org/releases/PCAWG/consensus_cnv/consensus.2 0170119.so matic.cna.ic gc.controlle d.tar.gz 

list_of_files <- list.files(path = "/media/user/seagate_ICM/AICDA_analysis/ICGC/copy_numbers/consensus.20170119.somatic.cna.annotated", recursive = TRUE,
                            pattern = "cna.annotated.txt", 
                            full.names = TRUE)

CN_data_ic<-rbindlist(sapply(list_of_files, fread, simplify = FALSE),            use.names = TRUE, idcol = "FileName")
##Clean sample name
colnames(CN_data_ic)[1]<-"Tumor_Sample_Barcode"
  CN_data_ic$Tumor_Sample_Barcode<-gsub(pattern = "/media/user/seagate_ICM/AICDA_analysis/Github_repository/Data/Figure3/Raw_Calculations/consensus.20170119.somatic.cna.annotated/",x = CN_data_ic$Tumor_Sample_Barcode,replacement = "")
  CN_data_ic$Tumor_Sample_Barcode<-gsub(pattern = ".consensus.20170119.somatic.cna.annotated.txt",x = CN_data_ic$Tumor_Sample_Barcode,replacement = "")
  
  #Sanity check 
  length(unique(CN_data_ic$Tumor_Sample_Barcode))
  [1] 2778
##Add ploidy
CN_data_ic$Ploidy<- Purity_ploidy$ploidy[match(CN_data_ic$Tumor_Sample_Barcode,Purity_ploidy$samplename)] 
###Add LogR data  
  # logR 
CN_data_ic$logR <- log2((CN_data_ic$total_cn + 0.3 ) / (CN_data_ic$Ploidy ))

###SUbset only to needed columns...
CN_data_ic<-CN_data_ic[,c("Tumor_Sample_Barcode","chromosome","start","end","logR","total_cn","major_cn","minor_cn","Ploidy")]
colnames(CN_data_ic)<-c("Sample","CHROM","POS_START","POS_END","LogR","ntot","Nmin","Nmaj","Ploidy")#Change colnames
##Subset clinical data to same samples as CN_data_ic into new object..
ICGC_clinial_palim_annot<-subset(ICGC_clinial_n3110[,c("Tumor_Sample_Barcode2","Gender","Purity")],ICGC_clinial_n3110$Tumor_Sample_Barcode2%in%unique(CN_data_ic$Sample))

colnames(ICGC_clinial_palim_annot)<-c("Sample", "Gender", "Purity")
dim(ICGC_clinial_palim_annot)
[1] 2709    3
###Change Gender to F or M
ICGC_clinial_palim_annot$Gender<-ifelse(ICGC_clinial_palim_annot$Gender=="female","F","M")
##Subset cn_data to this also
CN_data_ic_palim<-subset(CN_data_ic,CN_data_ic$Sample%in%ICGC_clinial_palim_annot$Sample)
length(unique(CN_data_ic_palim$Sample))
[1] 2707



#####vcf data...subset to contain same samples
icgc2_allmuts$Tumor_Sample_Barcode2<-ICGC_clinial_n3110$Tumor_Sample_Barcode2[match(paste0(icgc2_allmuts$Tumor_Sample_Barcode,icgc2_allmuts$Donor_ID),paste0(ICGC_clinial_n3110$Tumor_Sample_Barcode,ICGC_clinial_n3110$icgc_donor_id))]

icgc2_allmuts_clonality<-subset(icgc2_allmuts,icgc2_allmuts$Tumor_Sample_Barcode2%in%ICGC_clinial_palim_annot$Sample)



colnames(icgc2_allmuts_clonality)[c(75,7,2,3,8,10,13,74)]<-c("Sample","Type","CHROM","POS","REF","ALT","Tumor_Varcount","Tumor_Depth")
###Change colnames 
##Add Normal_Depth arbitrary, it is used internally by palimpsest to calculate log2R but it does not affect clonality calculation
icgc2_allmuts_clonality$Normal_Depth<-60
#Re-order some columns
icgc2_allmuts_clonality<-icgc2_allmuts_clonality[,c("Sample","Type","CHROM","POS","REF","ALT","Tumor_Varcount","Tumor_Depth","Normal_Depth",setdiff(colnames(icgc2_allmuts_clonality),c("Sample","Type","CHROM","POS","REF","ALT","Tumor_Varcount","Tumor_Depth","Normal_Depth")))]

# add chr in chromosome's name
icgc2_allmuts_clonality$CHROM <- sub("^", "chr", icgc2_allmuts_clonality$CHROM )    
CN_data_ic_palim$CHROM <- sub("^", "chr", CN_data_ic_palim$CHROM )   
##########################################
##########################################
##                                     ###
##          clonality analysis         ###
##                                     ###
##########################################
##########################################  


###Check there are not NA within columns needed for calculations (columns 1:9)
icgc2_allmuts_clonality<-na.omit(icgc2_allmuts_clonality)
###Omit mutations where there is not Tumor_Varcount or Tumor_Depth


# it takes a while,  !
icgc2_allmuts_clonality_CCF <- cnaCCF_annot(vcf=icgc2_allmuts_clonality[,], annot_data = ICGC_clinial_palim_annot,cna_data = CN_data_ic_palim,CCF_boundary=0.95) 

icgc2_allmuts_clonality<-icgc2_allmuts_clonality_CCF ##Doble save


##Subset only to needed columns
icgc2_allmuts_clonality_CCF<-icgc2_allmuts_clonality[,c(1:10,20:22,70,73,74,79,90)]          
####Take out chr X and Y since the gender info was not avaliable for all tumors and we similated some to get the calculations...
icgc2_allmuts_clonality_CCF<-icgc2_allmuts_clonality_CCF[icgc2_allmuts_clonality_CCF$CHROM %nin% c("chrX"),]
##Subset to only SNP
icgc2_allmuts_clonality_CCF<-icgc2_allmuts_clonality_CCF[icgc2_allmuts_clonality_CCF$Type %in% c("SNP"),]
table(icgc2_allmuts_clonality_CCF$Clonality)/nrow(icgc2_allmuts_clonality_CCF)
#Clonal and subclonal mutations %
####################
#####Figure Xa

  
  
###  Calculate which samples have measurable changes in the overall mutation spectra.... by permutating per sample...
###First get the mutation spectra (96 nt changes matrix) of clonal and subclonal mutations per sample
 # Caclulate trinucleotide substitution spectra across all samples and all timing categories.
  
#Annotate context
icgc2_allmuts_clonality_CCF$Type<-as.character(icgc2_allmuts_clonality_CCF$Type)
icgc2_allmuts_clonality_CCF$Type <- ifelse(icgc2_allmuts_clonality_CCF$Type == "SNP", yes = "SNV",icgc2_allmuts_clonality_CCF$Type)
#Change

icgc2_allmuts_clonality_CCF<-annotate_VCF(vcf=icgc2_allmuts_clonality_CCF,ref_genome =BSgenome.Hsapiens.UCSC.hg19, ref_fasta=NULL , add_ID_cats=F,add_DBS_cats = F)
 
#icgc2_allmuts_clonality_CCF_tmp<-annotate_VCF(vcf=icgc2_allmuts_clonality_CCF[icgc2_allmuts_clonality_CCF$Sample%in%unique(icgc2_allmuts_clonality_CCF$Sample)[1:4],],ref_genome =BSgenome.Hsapiens.UCSC.hg19, ref_fasta=NULL , add_ID_cats=F,add_DBS_cats = F) #For testing
 ##Get the signature matrix

###########"Fig xa, global numbers

icgc2_allmuts_clonality_CCF<-icgc2_allmuts_clonality_CCF[!is.na(icgc2_allmuts_clonality_CCF$Clonality),]



m <- as.data.frame.matrix(xtabs(~Clonality +SBS.Sig.max , data=icgc2_allmuts_clonality_CCF[])) ##Number per signature
m$Other_signatures<-rowSums(m[,colnames(m)%nin%c("SBS2","SBS13","AICDA_canonical")]) ##Put the other signatures together
m<-m[,c("SBS2","SBS13","AICDA_canonical","Other_signatures")]

m$overall<-rowSums(m) 

tst<-fisher.test(m[,c(3,4)])
pval <- tst$p.value
pval_label<- paste0('p ~',prettyNum(pval,digits=1))
tst<-fisher.test(m[,c(1,4)])
pval <- tst$p.value
pval_label2<- paste0('p ~',prettyNum(pval,digits=1))
tst<-fisher.test(m[,c(2,4)])
pval <- tst$p.value
pval_label3<- paste0('p ~',prettyNum(pval,digits=1))
###
#####

res2 <- m[,1:4] / m$overall
res2 <-cbind(Clonality=rownames(m),res2)
dat2 <- melt(res2,id.var='Clonality')
dat2$Clonality <- factor(dat2$Clonality,levels=rownames(m))
dat2$value <- 100*dat2$value
m<-m[,1:4]
ci <- rbind(binom.confint(m$AICDA_canonical,(m$SBS2+m$SBS13+m$AICDA_canonical+m$Other_signatures),method='exact') ,binom.confint(m$SBS2,(m$SBS2+m$SBS13+m$AICDA_canonical+m$Other_signatures),method='exact') ,binom.confint(m$SBS13,(m$SBS2+m$SBS13+m$AICDA_canonical+m$Other_signatures),method='exact') ,binom.confint(m$Other_signatures,(m$SBS2+m$SBS13+m$AICDA_canonical+m$Other_signatures),method='exact') )#For AICDA


ci$Clonality <- res2$Clonality
ci$Clonality <- factor(ci$Clonality, levels=c('clonal','subclonal'))
setnames(ci,'mean','value')
ci$value <- 100*ci$value
ci$lower <- 100*ci$lower
ci$upper <- 100*ci$upper
ci$variable<-c("AICDA_canonical","AICDA_canonical","SBS2","SBS2","SBS13","SBS13","Other_signatures","Other_signatures")
theme_std <- function(base_size = 11, base_line_size = base_size/22, base_rect_size = base_size/22) {
    require(ggplot2)
    #theme_classic(base_size = base_size, base_family = 'ArialMT', base_line_size = base_line_size, base_rect_size = base_rect_size)  %+replace%
    theme_classic(base_size = base_size, base_family = 'ArialMT')  %+replace%
        theme(
            line = element_line(colour = "black", size = base_line_size, linetype = 1, lineend = "round"),
            text = element_text(family = 'ArialMT', face = "plain",
                                colour = "black", size = base_size, lineheight = 0.9,
                                hjust = 0.5, vjust = 0.5, angle = 0, margin = margin(), debug=F),
            axis.text = element_text(colour = "black", family='ArialMT', size=rel(0.8)),
            axis.ticks = element_line(colour = "black", size=rel(1)),
            panel.border = element_blank(), panel.grid.major = element_blank(),
            panel.grid.minor = element_blank(),
            axis.line = element_line(colour = "black", size = rel(1)),
            legend.key = element_blank(),
            strip.background = element_blank())
}
dat2$variable <- factor(dat2$variable,levels=c("Other_signatures","AICDA_canonical","SBS2","SBS13"))
dat2$variable <- factor(dat2$variable,levels=c("Other_signatures","AICDA_canonical","SBS2","SBS13"))
ci$variable <- factor(ci$variable,levels=c("Other_signatures","AICDA_canonical","SBS2","SBS13"))

ci<-rbind(ci[ci$Clonality=="clonal",],ci[ci$Clonality=="subclonal",])
ci$mean_bar<-c(cumsum(ci$value[ci$Clonality=="clonal"]),cumsum(ci$value[ci$Clonality=="subclonal"]))
ci$lower<-c(cumsum(ci$lower[ci$Clonality=="clonal"]),cumsum(ci$lower[ci$Clonality=="subclonal"]))
ci$upper<-c(cumsum(ci$upper[ci$Clonality=="clonal"]),cumsum(ci$upper[ci$Clonality=="subclonal"]))


p1 <- ggplot(ci, aes(x=Clonality,y=value)) +
  geom_bar(stat='identity',aes(fill=variable)) +
  geom_signif(comparisons=list(c("clonal", "subclonal")), y_position =105, tip_length = 0, annotations=pval_label) + 
  scale_fill_manual(values=c("lightgray", "#F8766D","#00BFC4","#7CAE00"),name='Signature') +
  geom_errorbar(data=ci,aes(min=lower,max=mean_bar),color='white',width=0,size=1) +
  geom_errorbar(data=ci,aes(min=mean_bar,max=upper),color='black',width=0,size=1) +
  scale_y_continuous(expand=c(0,0),limits=c(0,115),breaks=seq(0,100,by=25),position='right') +
  labs(x=NULL,y='% of all mutants') +
  theme_std(base_size=14) +
  theme(legend.position='bottom') +
  coord_flip()+annotate("text", label=c(paste0("All mutations n = ",ci$n[5]),paste0("All mutations n = ",ci$n[1])), x=c(1.9,0.9),y=c(ci$value[8]-2,ci$value[8]-2), fontface =2,size=5)
#This is Supplementary Figure 14F
##
##############
###########################
###

#Get clonal matrix
icgc2_allmuts_clonality_CCF_clonal<-palimpsest_input(vcf = icgc2_allmuts_clonality_CCF[icgc2_allmuts_clonality_CCF$Clonality%in%"clonal",], Type = "SBS", mutypes = NA)
##Get only number of each mutation category and transpose data to have samples in rows
icgc2_allmuts_clonality_CCF_clonal<-as.data.frame(t(icgc2_allmuts_clonality_CCF_clonal$mut_nums))
icgc2_allmuts_clonality_CCF_clonal$Sample<-rownames(icgc2_allmuts_clonality_CCF_clonal)
rownames(icgc2_allmuts_clonality_CCF_clonal)<-paste0(rownames(icgc2_allmuts_clonality_CCF_clonal),"_clonal")
#Subclonal matrix
icgc2_allmuts_clonality_CCF_subclonal<-palimpsest_input(vcf = icgc2_allmuts_clonality_CCF[icgc2_allmuts_clonality_CCF$Clonality%in%"subclonal",], Type = "SBS", mutypes = NA)

icgc2_allmuts_clonality_CCF_subclonal<-as.data.frame(t(icgc2_allmuts_clonality_CCF_subclonal$mut_nums))
icgc2_allmuts_clonality_CCF_subclonal$Sample<-rownames(icgc2_allmuts_clonality_CCF_subclonal)
rownames(icgc2_allmuts_clonality_CCF_subclonal)<-paste0(rownames(icgc2_allmuts_clonality_CCF_subclonal),"_subclonal")
###Test for spectral changes per sample

tncTime<-rbind(icgc2_allmuts_clonality_CCF_clonal,icgc2_allmuts_clonality_CCF_subclonal)

tncLrt <- function(dat){
  y=dat[grep("subclonal",rownames(dat)),1:96] #Recover sub_clonal
  x=dat[rownames(dat)%nin%rownames(y),1:96]#Recover clonal
    l1 <- dmultinom(x=as.numeric(x[1,]), prob=as.numeric(x[1,]), log=TRUE) + dmultinom(x=as.numeric(y[1,]), prob=as.numeric(y[1,]), log=TRUE)
    l0 <- dmultinom(x=as.numeric(x[1,]), prob=as.numeric(x[1,])+as.numeric(y[1,]), log=TRUE) + dmultinom(x=as.numeric(y[1,]), prob=as.numeric(x[1,])+as.numeric(y[1,]), log=TRUE)
    chisq <- 2*(l1-l0)
    df <- ncol(x)-1 #Number of changes - 1 (96 spectra matrix)
    return(c(chisq=chisq, df=df, p=pchisq(chisq, df=df, lower.tail=FALSE)))
}

w <- which(rowSums(icgc2_allmuts_clonality_CCF_clonal[,1:96])>0) 
z<- which(rowSums(icgc2_allmuts_clonality_CCF_subclonal[,1:96])>0)# Samples with at mutations in both categories
w<-gsub(pattern = "_clonal",replacement = "",x = names(w))
z<-gsub(pattern = "_subclonal",replacement = "",x = names(z))
w<-intersect(w,z)
##Get data per sample
tncTime<-tncTime[tncTime$Sample%in%w,] #Subset only to samples with mutations in both clonal and subclonal cathegories
pClonalSubclonal<-c()
for (i in 1:length(unique(tncTime$Sample))) {
  tncTime_tm<-subset(tncTime,tncTime$Sample%in%unique(tncTime$Sample)[i])
  if (i==1) {
    pClonalSubclonal<-tncLrt(tncTime_tm)
  }else{
    pClonalSubclonal<-rbind(pClonalSubclonal,tncLrt(tncTime_tm))
  }
  
}#Apply function per sample
rownames(pClonalSubclonal)<-unique(tncTime$Sample)


table(p.adjust(pClonalSubclonal[,"p"],"bonf")<0.05)

##Number of samples from which there is significant spectral changes between clonal and subclonal mutations

#Save which samples

sam_clonalSubclonal<-names(which(p.adjust(pClonalSubclonal[,"p"],"bonf")<0.05))
###404 samples

###Now calculate subclonal to clonal foldchange per sample...for AICDA, SBS2 and SBS13 
##Foldchange of the relative proportions..
icgc2_allmuts_clonality_CCF$Mutation<-1 ##Add to sum mutations according to conditions..

##Do the sum only for the detected samples with spectral changes
clonal_sub_melt<-icgc2_allmuts_clonality_CCF[icgc2_allmuts_clonality_CCF$Sample%in%sam_clonalSubclonal,]%>% group_by(Sample,Tumor_type, SBS.Sig.max,Clonality) %>% dplyr::summarise(Mutations_sum = sum(Mutation))

clonal_sub_melt<-as.data.frame(clonal_sub_melt)
head(clonal_sub_melt)
dim(clonal_sub_melt)
##Eliminate NAs
clonal_sub_melt<-na.omit(clonal_sub_melt)
###This is the sum per signature per clonal status and sample

##Now get the sum per clonality and per sample to then get the fractions..
clonal_sub_melt_total<-icgc2_allmuts_clonality_CCF[icgc2_allmuts_clonality_CCF$Sample%in%sam_clonalSubclonal,]%>% group_by(Sample,Tumor_type,Clonality) %>% dplyr::summarise(Mutations_sum = sum(Mutation))

clonal_sub_melt_total<-as.data.frame(clonal_sub_melt_total)
head(clonal_sub_melt_total)
clonal_sub_melt_total<-na.omit(clonal_sub_melt_total)
###Now add the total clonal/subclonal next to each sample
clonal_sub_melt$Mutation_total_clonality<-clonal_sub_melt_total$Mutations_sum[match(paste0(clonal_sub_melt$Sample,clonal_sub_melt$Clonality),paste0(clonal_sub_melt_total$Sample,clonal_sub_melt_total$Clonality))]
##Calculate, the fraction (activity)
clonal_sub_melt$activity_fraction<-clonal_sub_melt$Mutations_sum/clonal_sub_melt$Mutation_total_clonality

table(clonal_sub_melt$Clonality)
##Some signatures have only clonal signal but not subclonal..


clonal_sub_melt2<-icgc2_allmuts_clonality_CCF[icgc2_allmuts_clonality_CCF$Sample%in%sam_clonalSubclonal,]%>% group_by(Sample,Tumor_type,SBS.Sig.max) %>% dplyr::summarise(Mutations_sum = sum(Mutation)) #Just to save results
clonal_sub_melt2<-as.data.frame(clonal_sub_melt2)
head(clonal_sub_melt2)

##Eliminate NAs
clonal_sub_melt2<-na.omit(clonal_sub_melt2)
dim(clonal_sub_melt2)
length(unique(clonal_sub_melt2$Sample))
clonal_sub_melt2$Acitity_clonal<-clonal_sub_melt$activity_fraction[match(paste0(clonal_sub_melt2$Sample,clonal_sub_melt2$SBS.Sig.max),paste0(clonal_sub_melt$Sample,clonal_sub_melt$SBS.Sig.max))]

clonal_sub_melt2$Acitity_subclonal<-clonal_sub_melt$activity_fraction[match(paste0(clonal_sub_melt2$Sample,clonal_sub_melt2$SBS.Sig.max),paste0(clonal_sub_melt$Sample,clonal_sub_melt$SBS.Sig.max))+1]
##Now fold change
clonal_sub_melt2$FC_clonal_Subclonal<-(clonal_sub_melt2$Acitity_subclonal/(1-clonal_sub_melt2$Acitity_subclonal))/(clonal_sub_melt2$Acitity_clonal/(1-clonal_sub_melt2$Acitity_clonal))

##Take out signature "0" which were the ones not attribuited..
clonal_sub_melt2<-clonal_sub_melt2[clonal_sub_melt2$SBS.Sig.max%nin%"0",]

clonal_sub_melt3<-clonal_sub_melt2[clonal_sub_melt2$SBS.Sig.max%in%c("SBS2","SBS13","SBS9","AICDA_canonical"),] #Subset
clonal_sub_melt3$Tumor_type<-"All-tumors" #Do this to have the global

clonal_sub_melt3<-rbind(clonal_sub_melt2[clonal_sub_melt2$SBS.Sig.max%in%c("SBS2","SBS13","SBS9","AICDA_canonical"),],clonal_sub_melt3)



n<-table(clonal_sub_melt3$Tumor_type[!duplicated(clonal_sub_melt3$Sample)])
n<-c(n,sum(n))
names(n)[length(n)]<-"All-tumors"
##Accomodate according to lowest (more clonal) to highest (more subclonal) of AICDA mutations...
order_n<-as.data.frame(clonal_sub_melt3[clonal_sub_melt3$SBS.Sig.max%in%"AICDA_canonical",]%>% group_by(Tumor_type) %>% dplyr::summarise(Mutations_sum = median(FC_clonal_Subclonal),Q25=quantile(FC_clonal_Subclonal,probs=.25),Q75=quantile(FC_clonal_Subclonal,probs=.75)))
order_n<-order_n[order(order_n$Mutations_sum,decreasing = F),]
order_n$n_samples<-n[order_n$Tumor_type]

clonal_sub_melt3$Tumor_type<factor(clonal_sub_melt3$Tumor_type, levels=order_n$Tumor_type)

####Fig Xb

###################Calculations for including replication timing (Supplementary Figure 14G)
####################
###########
###Generate a graph with all signatures per replication type
 icgc2_allmuts_clonality_CCF$Repli_zone<-ifelse(icgc2_allmuts_clonality_CCF$RepliScore_Decile<=5,"Late","Early") 

clonal_sub_melt<-icgc2_allmuts_clonality_CCF[icgc2_allmuts_clonality_CCF$Sample%in%sam_clonalSubclonal,]%>% group_by(Sample,Tumor_type, SBS.Sig.max,Clonality,Repli_zone) %>% dplyr::summarise(Mutations_sum = sum(Mutation))

clonal_sub_melt<-as.data.frame(clonal_sub_melt)
head(clonal_sub_melt)
dim(clonal_sub_melt)
##Eliminate NAs
clonal_sub_melt<-na.omit(clonal_sub_melt)
###This is the sum per signature per clonal status and sample

##Now get the sum per clonality and per sample to then get the fractions..
clonal_sub_melt_total<-icgc2_allmuts_clonality_CCF[icgc2_allmuts_clonality_CCF$Sample%in%sam_clonalSubclonal,]%>% group_by(Sample,Tumor_type,Clonality,Repli_zone) %>% dplyr::summarise(Mutations_sum = sum(Mutation))

clonal_sub_melt_total<-as.data.frame(clonal_sub_melt_total)
head(clonal_sub_melt_total)
clonal_sub_melt_total<-na.omit(clonal_sub_melt_total)
###Now add the total clonal/subclonal next to each sample
clonal_sub_melt$Mutation_total_clonality<-clonal_sub_melt_total$Mutations_sum[match(paste0(clonal_sub_melt$Sample,clonal_sub_melt$Clonality,clonal_sub_melt$Repli_zone),paste0(clonal_sub_melt_total$Sample,clonal_sub_melt_total$Clonality,clonal_sub_melt_total$Repli_zone))]
##Calculate, the fraction (activity)
clonal_sub_melt$activity_fraction<-clonal_sub_melt$Mutations_sum/clonal_sub_melt$Mutation_total_clonality

table(clonal_sub_melt$Clonality)
##Some signatures have only clonal signal but not subclonal..


clonal_sub_melt2<-icgc2_allmuts_clonality_CCF[icgc2_allmuts_clonality_CCF$Sample%in%sam_clonalSubclonal,]%>% group_by(Sample,Tumor_type,SBS.Sig.max,Repli_zone) %>% dplyr::summarise(Mutations_sum = sum(Mutation)) #Just to save results
clonal_sub_melt2<-as.data.frame(clonal_sub_melt2)
head(clonal_sub_melt2)

##Eliminate NAs
clonal_sub_melt2<-na.omit(clonal_sub_melt2)
dim(clonal_sub_melt2)
length(unique(clonal_sub_melt2$Sample))
clonal_sub_melt2$Acitity_clonal<-clonal_sub_melt$activity_fraction[match(paste0(clonal_sub_melt2$Sample,clonal_sub_melt2$SBS.Sig.max,clonal_sub_melt2$Repli_zone),paste0(clonal_sub_melt$Sample,clonal_sub_melt$SBS.Sig.max,clonal_sub_melt$Repli_zone))]

clonal_sub_melt2$Acitity_subclonal<-clonal_sub_melt$activity_fraction[match(paste0(clonal_sub_melt2$Sample,clonal_sub_melt2$SBS.Sig.max,clonal_sub_melt2$Repli_zone),paste0(clonal_sub_melt$Sample,clonal_sub_melt$SBS.Sig.max,clonal_sub_melt$Repli_zone))+2]
##Now fold change

clonal_sub_melt2$FC_clonal_Subclonal<-(clonal_sub_melt2$Acitity_subclonal/(1-clonal_sub_melt2$Acitity_subclonal))/(clonal_sub_melt2$Acitity_clonal/(1-clonal_sub_melt2$Acitity_clonal))

##Take out signature "0" which were the ones not attribuited..
clonal_sub_melt2<-clonal_sub_melt2[clonal_sub_melt2$SBS.Sig.max%nin%"0",]

clonal_sub_melt3<-clonal_sub_melt2[clonal_sub_melt2$SBS.Sig.max%in%c("SBS2","SBS13","SBS9","AICDA_canonical"),] #Subset
clonal_sub_melt3$Tumor_type<-"All-tumors" #Do this to have the global

clonal_sub_melt3<-rbind(clonal_sub_melt2[clonal_sub_melt2$SBS.Sig.max%in%c("SBS2","SBS13","SBS9","AICDA_canonical"),],clonal_sub_melt3)


clonal_sub_melt3$Tumor_type<factor(clonal_sub_melt3$Tumor_type, levels=order_n$Tumor_type)

####Supplementary figure
mutational_changes_clonality_p_repli<-ggboxplot(data = clonal_sub_melt3, y ="FC_clonal_Subclonal", x="Tumor_type",color = "SBS.Sig.max",xlab = "Signature",ylab = "Fold Change\n\nmore clonal<------------------->more subclonal",order = order_n$Tumor_type,facet.by = "Repli_zone",nrow=2)+theme(axis.text.x = element_text(angle = 90,size=8),axis.title = element_text(face="bold"),plot.title = element_text(hjust = 0.5,face = "bold",size = 15))+ geom_hline(yintercept=1, linetype="dashed",color="black")+scale_y_continuous(limits=c(0,6),breaks = c(.1,.5,1,2,4,6))+  guides(color=guide_legend(title = "Mutation type"))+scale_x_discrete(labels=paste0(order_n$Tumor_type," (n=", order_n$n_samples,")"))

###Data for discussion




```

```{r panel_3G, fig.width=14,fig.height=12}
#The total cohort consisted of 50,631 tumor samples representing more than 60 cancer types (Fig1A).
library(tidyr)
library(ggplot2)
library(reshape2)
library(dplyr)
library(tidyverse)
#Import data
##Import data
data_dir<-"/media/user/seagate_ICM/AICDA_analysis/Github_repository/" #Set to where you downloaded the data
setwd(data_dir)

clonal_sub_melt3<-as.data.frame(read.delim("Data/Figure3/Figure3G.tab"))
order_n<-as.data.frame(clonal_sub_melt3[clonal_sub_melt3$SBS.Sig.max%in%"AICDA_canonical",]%>% group_by(Tumor_type) %>% dplyr::summarise(Mutations_sum = median(FC_clonal_Subclonal),Q25=quantile(FC_clonal_Subclonal,probs=.25),Q75=quantile(FC_clonal_Subclonal,probs=.75)))
n<-table(clonal_sub_melt3$Tumor_type[!duplicated(clonal_sub_melt3$Sample)])
n<-c(n,sum(n))
names(n)[length(n)]<-"All-tumors"

order_n<-order_n[order(order_n$Mutations_sum,decreasing = F),]
order_n$n_samples<-n[order_n$Tumor_type]

clonal_sub_melt3$Tumor_type<factor(clonal_sub_melt3$Tumor_type, levels=order_n$Tumor_type)


mutational_changes_clonality_p<-ggboxplot(data = clonal_sub_melt3, y ="FC_clonal_Subclonal", x="Tumor_type",color = "SBS.Sig.max",xlab = "Signature",ylab = "Fold Change\n\nmore clonal<------------------->more subclonal",order = order_n$Tumor_type)+theme(axis.text.x = element_text(angle = 90,size=8),axis.title = element_text(face="bold"),plot.title = element_text(hjust = 0.5,face = "bold",size = 15))+ geom_hline(yintercept=1, linetype="dashed",color="black")+scale_y_continuous(limits=c(0,6),breaks = c(.1,.5,1,2,4,6))+  guides(color=guide_legend(title = "Mutation type"))+scale_x_discrete(labels=paste0(order_n$Tumor_type," (n=", order_n$n_samples,")"))

###Data for discuss
mutational_changes_clonality_p
```

***

\pagebreak

# Fig 3H


```{r,echo=T,eval=F}
## load gene annotations including covariates

gene_info_completed <- fread(here('/media/user/seagate_ICM/AICDA_analysis/Github_repository/Data/Figure3/Raw_Calculations/Gene_info_forModeling.txt'),
                   select=c('Hugo_Symbol','Role','panel_introduced','reptime',
                            'percentage_gene_gc_content','cds_length','hic'))

d<-icgc2_allmuts_clonality[,c("Hugo_Symbol","Tumor_Sample_Barcode","AICDA_mutation","Clonality","ntot","CCF","SBS.Sig.max","Tumor_type")]
#d <- d[d$exclude==F & d$putative_resistance_mutation==F,]
d <- merge(d, gene_info_completed, by='Hugo_Symbol', all.x=T)
colnames(d)[5]<-"tcn"#Change colname
d2<-d[d$Tumor_type%in%c("Skin-Melanoma","Myeloid-MPN"),]#Subset...
d2<-d2[!is.na(d2$reptime),] #Take out genes where there is no replication time data
d2<-d2[!is.na(d2$Clonality),]
d2<-as.data.table(d2)



## get observed number of composite mutated samples and mutated samples per gene

get_composite_rate_per_gene2_clonality<- function(d,AICDA=TRUE) {

    ## get sample barcodes for subclonal/clonal/overall
    tbl <- table.freq(d$Tumor_Sample_Barcode)
    samples <- tbl$value[tbl$N > 0] #overall
    if (AICDA==TRUE) {
       ###For AICDA condition
      tbl <- table.freq(d$Tumor_Sample_Barcode[d$Clonality%in%"clonal"&d$AICDA_mutation=="YES"])
      clonal_samples <- tbl$value[tbl$N>0] #Total clonal
      clonal_samples<-clonal_samples[!is.na(clonal_samples)] #Double checking
       tbl <- table.freq(d$Tumor_Sample_Barcode[d$Clonality%in%"subclonal"&d$AICDA_mutation=="YES"])
      subclonal_samples <- tbl$value[tbl$N>0] #Subclonal
subclonal_samples<-subclonal_samples[!is.na(subclonal_samples)] #Double checking
    
    
    }
    if (AICDA==FALSE){
    tbl <- table.freq(d$Tumor_Sample_Barcode[d$Clonality%in%"clonal"])
      clonal_samples <- tbl$value[tbl$N>0] #Total clonal
      clonal_samples<-clonal_samples[!is.na(clonal_samples)] #Double checking
       tbl <- table.freq(d$Tumor_Sample_Barcode[d$Clonality%in%"subclonal"])
      subclonal_samples <- tbl$value[tbl$N>0]
      subclonal_samples<-subclonal_samples[!is.na(subclonal_samples)] #Double checking
    }
    
    ## don't overcount multiple samples per patient
    samples <- length(unique(strtrim(samples, 9)))
    clonals <- length(unique(strtrim(clonal_samples, 9)))
    subclonals <- length(unique(strtrim(subclonal_samples, 9)))

    ## calculate the average TCN to include as covariate for this gene
    tcn <- mean(d$tcn,na.rm=T) 
    CCF_subclonal<-mean(d$CCF[d$Clonality%in%"subclonal"],na.rm=T) 
    CCF_clonal<-mean(d$CCF[d$Clonality%in%"clonal"],na.rm=T) 
    list(
         subclonals=subclonals,clonals=clonals,samples=samples,
         tcn=tcn,reptime=unique(d$reptime),percentage_gene_gc_content=unique(d$percentage_gene_gc_content),cds_length=unique(d$cds_length),
         hic=unique(d$hic),
         CCF_clonal=CCF_clonal,
         CCF_subclonal=CCF_subclonal)
} #Redefine function for clonality
#res <- d[,get_composite_rate_per_gene(.SD),by=Hugo_Symbol]

res2 <- d2[,get_composite_rate_per_gene2_clonality(.SD,AICDA=TRUE),by=Hugo_Symbol]

res2<-res2[res2$clonals!=0|res2$subclonals!=0,]#Eliminate gene that does not have clonal or subclonal mutations...
## impute missing values
res2[is.nan(tcn),tcn:=2]
res2$hic[is.na(res2$hic)] <- round(mean(res2$hic,na.rm=T))



## prep for test
res2$Hugo_Symbol <- gsub('-','_',res2$Hugo_Symbol)
res2$Hugo_Symbol <- gsub(' ','_',res2$Hugo_Symbol)
res2$reptime<-as.numeric(res2$reptime)

## use negative-binomial regression to model the predicted number of composite-mutant samples per gene by chance
#m <- glm.nb(composites ~ offset(log(samples)) + reptime + cds_length + percentage_gene_gc_content + panel_introduced + tcn + hic, data=res) #Panel introduced if it is targetted seq
###For clonals...
m_clonal <- glm.nb(clonals ~ offset(log(samples)) + reptime + cds_length + percentage_gene_gc_content+CCF_clonal  + tcn + hic , data=res2)

predictions <- predict(m_clonal,newdata=res2,type='response')
res2$predicted_clonals <- predictions 
res2$observed_proportion_clonals <- res2$clonals /res2$samples
#Subclonals
m_subclonal <- glm.nb(subclonals ~ offset(log(samples)) + reptime + cds_length + percentage_gene_gc_content+CCF_subclonal  + tcn + hic , data=res2)

predictions <- predict(m_subclonal,newdata=res2,type='response')
res2$predicted_subclonals <- predictions 
res2$observed_proportion_subclonals <- res2$subclonals / res2$samples



dat2 <- res2[,c('Hugo_Symbol','clonals',"subclonals",'samples','predicted_clonals','observed_proportion_clonals',"predicted_subclonals","observed_proportion_subclonals"),with=F]


## test each gene for enrichement of clonality (clonal odds ratio > 1; subclonal <1; no enrichment ==1)
test_gene_clonality <-function(dat) {
    x <- tryCatch({
      m<-rbind(c(dat$clonals,dat$subclonals),c(dat$predicted_clonals,dat$predicted_subclonals)) #Prepare data
        tst2 <- fisher.test(m,alternative='two.sided')
        p_twosided <- tst2$p.value
        OR<-tst2$estimate ##Odds ratio clonals/subclonals to determine which side test the enrichment
        tst<-ifelse(OR>1,"greater","less")
      tst <- fisher.test(m,alternative=tst) #
        p_enriched <- tst$p.value
        
        list(p_enriched=p_enriched, p_twosided=p_twosided,OR=OR)
    },error=function(e) {
        list(p_enriched=as.numeric(NA), p_twosided=as.numeric(NA),OR=as.numeric(NA))
    })
    dat$p_enriched <- x$p_enriched
    dat$p_twosided <- x$p_twosided
   dat$OR<-x$OR
    dat
}
dat2 <- dat2[,test_gene_clonality(.SD),by=Hugo_Symbol]


#dat2$logOR_clonal <- log2(dat2$observed_proportion_clonals/dat2$predicted_proportion_clonals)
#dat2$logOR_subclonal <- log2(dat2$observed_proportion_subclonals/dat2$predicted_proportion_subclonals)


dat2 <- dat2[order(p_enriched,decreasing=F),]

dat2$q_enriched<- p.adjust(dat2$p_enriched,method='BH')
dat2$q_twosided <- p.adjust(dat2$p_twosided,method='BH')
#dat2$q_enriched_subclonals <- p.adjust(dat2$p_enriched_subclonals,method='BH')
#dat2$q_twosided_subclonals <- p.adjust(dat2$p_twosided_subclonals,method='BH')

head(dat2[q_enriched<0.01,])
#log2(10)

###Repeat for AICDA ==FALSE and save as dat2_allTumors_allmuts
#dat2_allTumors_allmuts<-dat2
dat2_allTumors_allmuts$tag<-"All_mutations"
dat2_allTumors_AID<-dat2
dat2_allTumors_AID$tag<-"AICDA_mutations"

x <- gene_info[,c('Hugo_Symbol','Role'),with=F]
x$Hugo_Symbol <- gsub('-','_',x$Hugo_Symbol)
dat2_allTumors_allmuts <- merge(dat2_allTumors_allmuts, x,by='Hugo_Symbol', all.x=T)
dat2_allTumors_AID <- merge(dat2_allTumors_AID, x,by='Hugo_Symbol', all.x=T)##Add role tag

####Merge all muts with the significant AICDA to make a volcano plot together....

dat2_v<-rbind(dat2_allTumors_AID[q_enriched<0.01,],dat2_allTumors_allmuts)
##Take out NAs on OR or q_enriched
dat2_v<-dat2_v[!is.na(dat2_v$OR)&!is.na(dat2_v$q_enriched),]

dat2_v$DE<-ifelse(dat2_v$q_enriched<0.01&dat2_v$tag=="AICDA_mutations","FDR < 0.05 (AICDA)", ifelse(dat2_v$q_enriched<0.01&dat2_v$tag=="All_mutations","FDR < 0.05 (All)","Not Sig"))

#################
#############"Supplementary tables
####
##Save table

write.table(dat2_allTumors_AID,file = "/media/user/seagate_ICM/AICDA_analysis/ICGC/mutational_signatures/Clonality/Clonal_enriched_genes_AICDA.tab",quote = F,sep = "\t",row.names = F)
write.table(dat2_allTumors_allmuts,file = "/media/user/seagate_ICM/AICDA_analysis/ICGC/mutational_signatures/Clonality/Clonal_enriched_genes_AllMutations.tab",quote = F,sep = "\t",row.names = F)
##Repeat by tumor type...not posible due to low number of AICDA mutations when separating through clonal_subclonal

#########################
############Supplementary Figures 16D
p_tm<-ggscatter(data = ICGC_clinial_n3110[!duplicated(ICGC_clinial_n3110$Tumor_Sample_Barcode2),], x = "fraction_APOBEC_mutations",y = "fraction_AICDA_mutations",facet.by ="histology_abbreviation" ,cor.coef = TRUE, cor.coeff.args=list(color="red"),cor.method = "spearman",xlab = "Fraction APOBEC mutations", ylab = "Fraction AICDA mutations",label.rectangle=TRUE, ggtheme=theme_bw(base_size = 14),size = 0.5)+ geom_pointdensity(adjust=4) + scale_color_viridis_c()+geom_smooth(color = "red",method=lm, se=FALSE,linetype="dashed") + theme(plot.margin = unit(c(2,1,1,1), "lines"))+theme(plot.title = element_text(hjust = 0.5))

ggsave(p_tm,filename = "/media/user/seagate_ICM/AICDA_analysis/ICGC/ICGC_AICDA_vs_APOBEC_fraction_spearman.pdf",width = 16,height = 14)
###N mutations
ICGC_clinial_n3110$APOBEC_mutations<-(ICGC_clinial_n3110$AICDA_mutations/ICGC_clinial_n3110$fraction_AICDA_mutations)*ICGC_clinial_n3110$fraction_APOBEC_mutations

p_tm<-ggscatter(data = ICGC_clinial_n3110[!duplicated(ICGC_clinial_n3110$Tumor_Sample_Barcode2),], x = "APOBEC_mutations",y = "AICDA_mutations",facet.by ="histology_abbreviation" ,cor.coef = TRUE, cor.coeff.args=list(color="red"),cor.method = "spearman",xlab = "Number of APOBEC mutations", ylab = "Number of AICDA mutations",label.rectangle=TRUE, ggtheme=theme_bw(base_size = 14),size = 0.5)+ geom_pointdensity(adjust=4) + scale_color_viridis_c()+geom_smooth(color = "red",method=lm, se=FALSE,linetype="dashed") + theme(plot.margin = unit(c(2,1,1,1), "lines"))+theme(plot.title = element_text(hjust = 0.5))



```

```{r panel_3H, fig.width=14,fig.height=12}
#The total cohort consisted of 50,631 tumor samples representing more than 60 cancer types (Fig1A).
#
#Import data
##Import data
data_dir<-"/media/user/seagate_ICM/AICDA_analysis/Github_repository/" #Set to where you downloaded the data
setwd(data_dir)

dat2_v<-as.data.table(read.delim("/media/user/seagate_ICM/AICDA_analysis/Github_repository/Data/Figure3/Figure3H.tab"))
###Volcano plot
library(ggrepel)
# plot adding up all layers we have seen so far
tmp_lab<-subset(dat2_v, q_enriched < 0.01)
tmp_lab <- tmp_lab[order(q_enriched,decreasing=F),]



p_enrich_clonality<-ggplot(data=dat2_v, aes(x=log2(OR), y=-log10(q_enriched), col=DE)) +geom_point() +theme_std(base_size=14) + scale_color_manual(name=NULL,values = c("blue","red", "grey"))  +
    geom_vline(xintercept=log2(1), col="black",linetype="dotted") +
    geom_hline(yintercept=-log10(0.01), col="black",linetype="dotted")+xlab("                          Subclonal<----- Log2 (Odds ratio) ----->Clonal")+theme(legend.position = "bottom") +geom_text_repel(data = tmp_lab[c(1:20,68,134,82,67),],
                                                                                                                                                                                 aes(label = Hugo_Symbol,x=log2(OR), y=-log10(q_enriched)),
                                                                                                                                                                                 size = 5,
                                                                                                                                                                                 box.padding = unit(0.35, "lines"),
                                                                                                                                                                                 point.padding = unit(0.3, "lines"),inherit.aes = F)


p_enrich_clonality

```

***

\pagebreak




