---
output:
  rmarkdown::html_document:
    highlight: pygments
    toc: true
    toc_float:
      collapsed: false
      smooth_scroll: false
    css: style.css
chunk_output_type: console
header-includes:
- \pagenumbering{gobble}
params:
    set_title: "Figure-1"
title: "`r params$set_title`"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE, autodep = knitr::dep_prev())
```

***

\pagebreak

# Fig 1a

```{r calculations_fig1a,echo=T,eval=F}
library(tidyr)
library(ggplot2)
library(reshape2)
library(dplyr)
library(tidyverse)
##Import data

data_dir<-"/media/user/seagate_ICM/AICDA_analysis/Github_repository/" #Set to where you downloaded the data
setwd(data_dir)
source('r/prerequisites.R')

###This section includes all the calculations, starting from raw data obtained from the ICGC portal, to obtain the necessary data for the plots comprising panels B,C and D of Figure 1

library(maftools)
vc_nonsyn = c("Frame_Shift_Del", 
              "Frame_Shift_Ins", 
              "Splice_Site", 
              "Translation_Start_Site",
              "Nonsense_Mutation", 
              "Nonstop_Mutation", 
              "In_Frame_Del",
              "In_Frame_Ins", 
              "Missense_Mutation")

# silent mutations
# according to TCGA format https://docs.gdc.cancer.gov/Encyclopedia/pages/Mutation_Annotation_Format_TCGAv2/
vc_all = c(vc_nonsyn, "Intron", 
                       "5'UTR", 
                       "3'UTR", 
                       "5'Flank", 
                       "3'Flank", 
                       "IGR")

icgc_file <- "/media/user/seagate_ICM/AICDA_analysis/ICGC/final_consensus_passonly.snv_mnv_indel.icgc.public.maf.gz" #Download file directly from ICGC consortium
#Download from: https://dcc.icgc.org/releases/PCAWG/consensus_snv_indel/final_consensus_snv_indel_passonly_icgc.public.tgz



icgc <- read.maf(icgc_file, vc_nonSyn = vc_all) #This contains only 1950 samples, download remaining from https://dcc.icgc.org/search?filters=%7B%22donor%22:%7B%22studies%22:%7B%22is%22:%5B%22PCAWG%22%5D%7D,%22analysisTypes%22:%7B%22is%22:%5B%22WGS%22%5D%7D,%22availableDataTypes%22:%7B%22is%22:%5B%22ssm%22,%22cnsm%22%5D%7D%7D%7D&donors=%7B%22from%22:1%7D


#Load functions:
load("r/AICDA_enrich_function_WES_WGS.RData") ##Change where you downloaded the functions...
load("r/APOBEC_enrich_function_WES_WGS.RData") ##Change where you downloaded the functions...

#

icgc.maf_remaining2<-icgcSimpleMutationToMAF(icgc = "/media/user/seagate_ICM/AICDA_analysis/ICGC/ICGC_data_remaining_n/simple_somatic_mutation.open.tsv.gz",addHugoSymbol=T ,MAFobj = T)
##349 samples
table(icgc.maf_remaining2@data$project_code[unique(icgc.maf_remaining2@data$Tumor_Sample_Barcode)])

##Load Tumor_sample_barcode of other set 1880 samples...
already_done<-read.table("Data/Figure1/Raw_calculations/AlreadyDone_TumorBarcodes_.tsv", quote="\"", comment.char="")
##Add the real Tumor_Sample_Barcode to the maf since the other is the "icgc_sample_id" according to the already compiled clinial data...
library(readxl)
pcawg_specimen_histology_August2016_v9 <- read_excel("pcawg_specimen_histology_August2016_v9.xlsx")
pcawg_specimen_histology_August2016_v9<-as.data.frame(pcawg_specimen_histology_August2016_v9)
pcawg_sample_sheet <- read.delim("Data/Figure1/Raw_calculations/pcawg_sample_sheet.tsv")
length(intersect(pcawg_specimen_histology_August2016_v9$`# icgc_specimen_id`,pcawg_sample_sheet$icgc_specimen_id))
pcawg_specimen_histology_August2016_v9$Tumor_Sample_Barcode<-0
pcawg_specimen_histology_August2016_v9$Tumor_Sample_Barcode<-pcawg_sample_sheet$aliquot_id[match(pcawg_specimen_histology_August2016_v9$`# icgc_specimen_id`,pcawg_sample_sheet$icgc_specimen_id)]##Add Tumor_Sample_Barcode to pcawg_specimen_histology_August2016_v9

#Subset to same samples as the icgc maf file
##Add the real Tumor_Sample_Barcode to the maf since the other is the "icgc_sample_id" according to the already compiled clinial data...
already_done<-data.frame(Tumor_Sample_Barcode=already_done$V1,icgc_sample_id=pcawg_specimen_histology_August2016_v9$icgc_sample_id[match(already_done$V1,pcawg_specimen_histology_August2016_v9$Tumor_Sample_Barcode)])

#icgc_sample_id==Tumor_Sample_Barcode


length(setdiff(unique(icgc.maf_remaining2@data$Tumor_Sample_Barcode),already_done$icgc_sample_id))
to_keep<-setdiff(unique(icgc.maf_remaining2@data$Tumor_Sample_Barcode),already_done$icgc_sample_id)

#228 different samples , 119 already done...
###Subset maf to these samples
icgc.maf_remaining2_2<-subsetMaf(icgc.maf_remaining2, tsb =to_keep)

##Repeat for the other set 

length(setdiff(unique(icgc.maf_remaining1@data$Tumor_Sample_Barcode),already_done$icgc_sample_id))
to_keep<-setdiff(unique(icgc.maf_remaining1@data$Tumor_Sample_Barcode),already_done$icgc_sample_id)

#1074 different samples , 97 already done...
###Subset maf to these samples
icgc.maf_remaining1_2<-subsetMaf(icgc.maf_remaining1, tsb =to_keep)


##Merge maf and save data...
icgc.maf_remaining_final<-maftools::merge_mafs(list(icgc.maf_remaining1_2,icgc.maf_remaining2_2))
#1300 samples...

save(icgc.maf_remaining_final,file = "/media/user/seagate_ICM/AICDA_analysis/ICGC/ICGC_data_remaining_n1300/ICGC_maf1300.RData")
###A

icgc.maf_remaining_final

#####################"
###########################"
###########

#Subset to same samples as the icgc maf file
ICGC_clinial<-subset(pcawg_specimen_histology_August2016_v9, pcawg_specimen_histology_August2016_v9$Tumor_Sample_Barcode %in% icgc@data$Tumor_Sample_Barcode)
#take out RNA-seq data
ICGC_clinial<-subset(ICGC_clinial, ICGC_clinial$specimen_library_strategy %in% c("WGS","WGS+RNA-Seq"))


#Extract AICDA mutations by tumor type

library( BSgenome.Hsapiens.UCSC.hg19) #Was built in this human reference build hs37d5, according to the original article
icgc_tumorType<-unique((icgc@data$Project_Code))
for (i in 1:11) {
icgc_tmp<-subsetMaf(icgc, tsb = (unique(icgc@data$Tumor_Sample_Barcode[which(icgc@data$Project_Code==icgc_tumorType[i])])))

  icgc_aicda_tmp<-AICDA_enrichment(icgc_tmp, ref_genome = "BSgenome.Hsapiens.UCSC.hg19",prefix = 'chr', add = TRUE, background_n = 60,seq_type = "WGS") #Add the WGS flag
  
  if (i==1) {
    icgc_scores<-icgc_aicda_tmp$AICDA_scores
  icgc_aicda_maf<-icgc_aicda_tmp$maf_AICDA
    
  }else{
    icgc_scores<-rbind(icgc_scores,icgc_aicda_tmp$AICDA_scores)
    icgc_aicda_maf<-rbind(icgc_aicda_maf,icgc_aicda_tmp$maf_AICDA)
  }
}
##CHeck dimensions consistency
sum(icgc_scores$n_mutations-icgc_scores$non_AICDA_mutations)
nrow(icgc_aicda_maf)


for (i in 13:length(icgc_tumorType)) {
icgc_tmp<-subsetMaf(icgc, tsb = (unique(icgc@data$Tumor_Sample_Barcode[which(icgc@data$Project_Code==icgc_tumorType[i])])))
icgc_aicda_tmp<-AICDA_enrichment(icgc_tmp, ref_genome = "BSgenome.Hsapiens.UCSC.hg19",prefix = 'chr', add = TRUE, background_n = 60,seq_type = "WGS")
  
  
  if (i==13) {
    #icgc_tmp@data<-icgc_tmp@data[-5469137,];icgc_tmp@data<-icgc_tmp@data[-5469136,]
    
    icgc_scores2<-icgc_aicda_tmp$AICDA_scores
  icgc_aicda_maf2<-icgc_aicda_tmp$maf_AICDA
    
  }else{
    #icgc_aicda_tmp<-AICDA_enrichment(icgc_tmp, ref_genome = "BSgenome.Hsapiens.UCSC.hg19",prefix = 'chr', add = TRUE, background_n = 60)
  
    icgc_scores2<-rbind(icgc_scores2,icgc_aicda_tmp$AICDA_scores)
    icgc_aicda_maf2<-rbind(icgc_aicda_maf2,icgc_aicda_tmp$maf_AICDA)
  }
}

sum(icgc_scores2$n_mutations-icgc_scores2$non_AICDA_mutations)
nrow(icgc_aicda_maf2)

##Omit row solving row 5469137 for Skin-Melanoma (i==12) because of an error during calculation
##Do Skin-Melanoma separately 
i=12
icgc_tmp<-subsetMaf(icgc, tsb = setdiff((unique(icgc@data$Tumor_Sample_Barcode[which(icgc@data$Project_Code==icgc_tumorType[i])])),c("a84915de-6562-4836-86f9-f2a05598296e","9fc5b5c7-3973-42b4-8710-454de0cb5b50")))
#icgc_tmp@data<-icgc_tmp@data[-5469137,];icgc_tmp@data<-icgc_tmp@data[-5469136,];icgc_tmp@data<-icgc_tmp@data[-5469135,];icgc_tmp@data<-icgc_tmp@data[-5469134,]
icgc_aicda_tmp<-AICDA_enrichment(icgc_tmp, ref_genome = "BSgenome.Hsapiens.UCSC.hg19",prefix = 'chr', add = TRUE, background_n = 60,seq_type = "WGS")

#Sample a84915de-6562-4836-86f9-f2a05598296e

###Skin-Melanoma was eliminated due to error for AICDA enrichment...

###Merge
icgc_scores_final<-rbind(icgc_scores,icgc_scores2)
icgc_aicda_maf_final<-rbind(icgc_aicda_maf,icgc_aicda_maf2)
####Boxplot of AICDA mutations_fraction of mutations....Add this data to clinical data and also AICDA_enrichment score
  ICGC_clinial<-subset(ICGC_clinial, !(ICGC_clinial$histology_abbreviation %in% c("Skin-Melanoma"))) #Take out skin melanoma
dim(ICGC_clinial) #1880
ICGC_clinial$fraction_AICDA_mutations<-0
ICGC_clinial$fraction_AICDA_mutations<-icgc_scores_final$fraction_AICDA_mutations[match(ICGC_clinial$Tumor_Sample_Barcode,icgc_scores_final$Tumor_Sample_Barcode)]##Add

ICGC_clinial$AICDA_Enrichment<-icgc_scores_final$AICDA_Enrichment[match(ICGC_clinial$Tumor_Sample_Barcode,icgc_scores_final$Tumor_Sample_Barcode)]
###
icgc_scores_final$AICDA_mutations<-icgc_scores_final$n_mutations-icgc_scores_final$non_AICDA_mutations
ICGC_clinial$AICDA_mutations<-icgc_scores_final$AICDA_mutations[match(ICGC_clinial$Tumor_Sample_Barcode,icgc_scores_final$Tumor_Sample_Barcode)]

  
######
###
##Make the same for APOBEC, do graph

#Extract APOBEC mutations by tumor type
rm(icgc_scores,icgc_scores2,icgc_aicda_maf,icgc_aicda_maf2)
library( BSgenome.Hsapiens.UCSC.hg19) #Was built in this human reference build hs37d5, according to the original article
icgc_tumorType<-unique((icgc@data$Project_Code))
for (i in 1:11) {
icgc_tmp<-subsetMaf(icgc, tsb = (unique(icgc@data$Tumor_Sample_Barcode[which(icgc@data$Project_Code==icgc_tumorType[i])])))

  icgc_a_tmp<-APOBEC_enrichment(icgc_tmp, ref_genome = "BSgenome.Hsapiens.UCSC.hg19",prefix = 'chr', add = TRUE, seq_type = "WGS") #Add the WGS flag
  
  if (i==1) {
    icgc_scores<-icgc_a_tmp$APOBEC_scores
  icgc_apobec_maf<-icgc_a_tmp$MAF_APOBEC
    
  }else{
    icgc_scores<-rbind(icgc_scores,icgc_a_tmp$APOBEC_scores)
    icgc_apobec_maf<-rbind(icgc_apobec_maf,icgc_a_tmp$MAF_APOBEC)
  }
}

for (i in 13:length(icgc_tumorType)) {
icgc_tmp<-subsetMaf(icgc, tsb = (unique(icgc@data$Tumor_Sample_Barcode[which(icgc@data$Project_Code==icgc_tumorType[i])])))
icgc_a_tmp<-APOBEC_enrichment(icgc_tmp, ref_genome = "BSgenome.Hsapiens.UCSC.hg19",prefix = 'chr', add = TRUE,seq_type = "WGS")
  
  
  if (i==13) {

    icgc_scores2<-icgc_a_tmp$APOBEC_scores
  icgc_apobec_maf2<-icgc_a_tmp$MAF_APOBEC
    
  }else{
    
    icgc_scores2<-rbind(icgc_scores2,icgc_a_tmp$APOBEC_scores)
    icgc_apobec_maf2<-rbind(icgc_apobec_maf2,icgc_a_tmp$MAF_APOBEC)
  }
}


###Merge
icgc_scores_apobec_final<-rbind(icgc_scores,icgc_scores2)
icgc_apobec_maf_final<-rbind(icgc_apobec_maf,icgc_apobec_maf2)
####Boxplot of AICDA mutations_fraction of mutations....Add this data to clinical data and also AICDA_enrichment score
  
ICGC_clinial$fraction_APOBEC_mutations<-0
ICGC_clinial$fraction_APOBEC_mutations<-icgc_scores_apobec_final$fraction_APOBEC_mutations[match(ICGC_clinial$Tumor_Sample_Barcode,icgc_scores_apobec_final$Tumor_Sample_Barcode)]##Add

ICGC_clinial$APOBEC_Enrichment<-icgc_scores_apobec_final$APOBEC_Enrichment[match(ICGC_clinial$Tumor_Sample_Barcode,icgc_scores_apobec_final$Tumor_Sample_Barcode)]


############################################
###################SUbsecition to analyze remaining samples (n= 1300)
#########"
##
###
#Subset to same samples as the icgc maf file
####CHarge new clinical data ##Other one was not complete (pcawg_specimen_histology_August2016_v9)
##Use icgc_sample_id since is the one common for all data..
ICGC_clinial_rem<-read.delim("/media/user/seagate_ICM/AICDA_analysis/ICGC/Complete_clinicalData/sample.tsv")
##But unfortunately tumor_type according to the already published papers is stored in:
length(table(pcawg_specimen_histology_August2016_v9$histology_abbreviation))


ICGC_clinial_rem2<-subset(ICGC_clinial_rem, ICGC_clinial_rem$icgc_sample_id %in% icgc.maf_remaining_final@data$Tumor_Sample_Barcode)#SUbset

#We need to create a common table, add the histology type to "ICGC_clinial_rem" using "pcawg_specimen_histology_August2016_v9"; but since not all tumor ids are included we have to add it according to project code

ICGC_clinial_rem2$histology_abbreviation<-pcawg_specimen_histology_August2016_v9$histology_abbreviation[match(ICGC_clinial_rem2$project_code,pcawg_specimen_histology_August2016_v9$project_code)]



#Extract AICDA mutations by tumor type

library( BSgenome.Hsapiens.UCSC.hg19) #Was built in this human reference build hs37d5, according to the original article
icgc_tumorType2<-unique((icgc.maf_remaining_final@data$project_code))
for (i in 1:2) {
icgc_tmp<-subsetMaf(icgc.maf_remaining_final, tsb = (unique(icgc.maf_remaining_final@data$Tumor_Sample_Barcode[which(icgc.maf_remaining_final@data$project_code==icgc_tumorType2[i])])))

  icgc_aicda_tmp<-AICDA_enrichment(icgc_tmp, ref_genome = "BSgenome.Hsapiens.UCSC.hg19",prefix = 'chr', add = TRUE, background_n = 60,seq_type = "WGS") #Add the WGS flag
  
  if (i==1) {
    icgc_scores<-icgc_aicda_tmp$AICDA_scores
  icgc_aicda_maf<-icgc_aicda_tmp$maf_AICDA
    
  }else{
    icgc_scores<-rbind(icgc_scores,icgc_aicda_tmp$AICDA_scores)
    icgc_aicda_maf<-rbind(icgc_aicda_maf,icgc_aicda_tmp$maf_AICDA)
  }
}

##CHeck dimensions consistency
sum(icgc_scores$n_mutations-icgc_scores$non_AICDA_mutations)
nrow(icgc_aicda_maf)

#Ignore MELA-AU because of error

for (i in 4:length(icgc_tumorType2)) {
icgc_tmp<-subsetMaf(icgc.maf_remaining_final, tsb = (unique(icgc.maf_remaining_final@data$Tumor_Sample_Barcode[which(icgc.maf_remaining_final@data$project_code==icgc_tumorType2[i])])))
icgc_aicda_tmp<-AICDA_enrichment(icgc_tmp, ref_genome = "BSgenome.Hsapiens.UCSC.hg19",prefix = 'chr', add = TRUE, background_n = 60,seq_type = "WGS")
  
  
  if (i==4) {
    #icgc_tmp@data<-icgc_tmp@data[-5469137,];icgc_tmp@data<-icgc_tmp@data[-5469136,]
    
    icgc_scores2<-icgc_aicda_tmp$AICDA_scores
  icgc_aicda_maf2<-icgc_aicda_tmp$maf_AICDA
    
  }else{
    #icgc_aicda_tmp<-AICDA_enrichment(icgc_tmp, ref_genome = "BSgenome.Hsapiens.UCSC.hg19",prefix = 'chr', add = TRUE, background_n = 60)
  
    icgc_scores2<-rbind(icgc_scores2,icgc_aicda_tmp$AICDA_scores)
    icgc_aicda_maf2<-rbind(icgc_aicda_maf2,icgc_aicda_tmp$maf_AICDA)
  }
}

sum(icgc_scores2$n_mutations-icgc_scores2$non_AICDA_mutations)
nrow(icgc_aicda_maf2)

##Omit row solving row 5469137 for MELA-AU (i==3) because of an error during calculation
##Do Skin-Melanoma separately 
i=12
icgc_tmp<-subsetMaf(icgc, tsb = setdiff((unique(icgc@data$Tumor_Sample_Barcode[which(icgc@data$Project_Code==icgc_tumorType[i])])),c("a84915de-6562-4836-86f9-f2a05598296e","9fc5b5c7-3973-42b4-8710-454de0cb5b50")))
#icgc_tmp@data<-icgc_tmp@data[-5469137,];icgc_tmp@data<-icgc_tmp@data[-5469136,];icgc_tmp@data<-icgc_tmp@data[-5469135,];icgc_tmp@data<-icgc_tmp@data[-5469134,]
icgc_aicda_tmp<-AICDA_enrichment(icgc_tmp, ref_genome = "BSgenome.Hsapiens.UCSC.hg19",prefix = 'chr', add = TRUE, background_n = 60,seq_type = "WGS")

#Sample a84915de-6562-4836-86f9-f2a05598296e

###MELA-AU project which is part of SKIN-MELANOMA was eliminated due to error for AICDA enrichment...

###Merge
icgc_scores_final_re<-rbind(icgc_scores,icgc_scores2)
icgc_aicda_maf_final_re<-rbind(icgc_aicda_maf,icgc_aicda_maf2)



####Boxplot of AICDA mutations_fraction of mutations....Add this data to clinical data and also AICDA_enrichment score
  ICGC_clinial_rem2<-subset(ICGC_clinial_rem2, !(ICGC_clinial_rem2$project_code %in% c("MELA-AU"))) #Take out skin melanoma
dim(ICGC_clinial_rem2) #1230
ICGC_clinial_rem2$fraction_AICDA_mutations<-0
###Add:
ICGC_clinial_rem2$Tumor_Sample_Barcode<-ICGC_clinial_rem2$icgc_sample_id

ICGC_clinial_rem2$fraction_AICDA_mutations<-icgc_scores_final_re$fraction_AICDA_mutations[match(ICGC_clinial_rem2$Tumor_Sample_Barcode,icgc_scores_final_re$Tumor_Sample_Barcode)]##Add

ICGC_clinial_rem2$AICDA_Enrichment<-icgc_scores_final_re$AICDA_Enrichment[match(ICGC_clinial_rem2$Tumor_Sample_Barcode,icgc_scores_final_re$Tumor_Sample_Barcode)]
###
icgc_scores_final_re$AICDA_mutations<-icgc_scores_final_re$n_mutations-icgc_scores_final_re$non_AICDA_mutations
ICGC_clinial_rem2$AICDA_mutations<-icgc_scores_final_re$AICDA_mutations[match(ICGC_clinial_rem2$Tumor_Sample_Barcode,icgc_scores_final_re$Tumor_Sample_Barcode)]

  
######
###
##Make the same for APOBEC, do graph

#Extract APOBEC mutations by tumor type
rm(icgc_scores,icgc_scores2,icgc_aicda_maf,icgc_aicda_maf2)
library( BSgenome.Hsapiens.UCSC.hg19) #Was built in this human reference build hs37d5, according to the original article


for (i in 1:3) {
icgc_tmp<-subsetMaf(icgc.maf_remaining_final, tsb = (unique(icgc.maf_remaining_final@data$Tumor_Sample_Barcode[which(icgc.maf_remaining_final@data$project_code==icgc_tumorType2[i])])))

  icgc_a_tmp<-APOBEC_enrichment(icgc_tmp, ref_genome = "BSgenome.Hsapiens.UCSC.hg19",prefix = 'chr', add = TRUE, seq_type = "WGS") #Add the WGS flag
  
  if (i==1) {
    icgc_scores<-icgc_a_tmp$APOBEC_scores
  icgc_apobec_maf<-icgc_a_tmp$MAF_APOBEC
    
  }else{
    icgc_scores<-rbind(icgc_scores,icgc_a_tmp$APOBEC_scores)
    icgc_apobec_maf<-rbind(icgc_apobec_maf,icgc_a_tmp$MAF_APOBEC)
  }
}

for (i in 4:length(icgc_tumorType2)) {
icgc_tmp<-subsetMaf(icgc.maf_remaining_final, tsb = (unique(icgc.maf_remaining_final@data$Tumor_Sample_Barcode[which(icgc.maf_remaining_final@data$project_code==icgc_tumorType2[i])])))
icgc_a_tmp<-APOBEC_enrichment(icgc_tmp, ref_genome = "BSgenome.Hsapiens.UCSC.hg19",prefix = 'chr', add = TRUE,seq_type = "WGS")
  
  
  if (i==4) {

    icgc_scores2<-icgc_a_tmp$APOBEC_scores
  icgc_apobec_maf2<-icgc_a_tmp$MAF_APOBEC
    
  }else{
    
    icgc_scores2<-rbind(icgc_scores2,icgc_a_tmp$APOBEC_scores)
    icgc_apobec_maf2<-rbind(icgc_apobec_maf2,icgc_a_tmp$MAF_APOBEC)
  }
}


###Merge
icgc_scores_apobec_final_re<-rbind(icgc_scores,icgc_scores2)
icgc_apobec_maf_final_re<-rbind(icgc_apobec_maf,icgc_apobec_maf2)
####Boxplot of AICDA mutations_fraction of mutations....Add this data to clinical data and also AICDA_enrichment score
  
ICGC_clinial_rem2$fraction_APOBEC_mutations<-0
ICGC_clinial_rem2$fraction_APOBEC_mutations<-icgc_scores_apobec_final_re$fraction_APOBEC_mutations[match(ICGC_clinial_rem2$Tumor_Sample_Barcode,icgc_scores_apobec_final_re$Tumor_Sample_Barcode)]##Add

ICGC_clinial_rem2$APOBEC_Enrichment<-icgc_scores_apobec_final_re$APOBEC_Enrichment[match(ICGC_clinial_rem2$Tumor_Sample_Barcode,icgc_scores_apobec_final_re$Tumor_Sample_Barcode)]

###

##Add data to maf, which mutations are AICDA
icgc2_allmuts_re<-rbind(as.data.frame(icgc.maf_remaining_final@data),as.data.frame(icgc.maf_remaining_final@maf.silent)) ##Add data and maf_silent info
icgc2_allmuts_re<-subset(icgc2_allmuts_re,!(icgc2_allmuts_re$project_code %in% c("MELA-AU")))#Take out MELA-AU from Skin-Melanoma


colnames(icgc2_allmuts_re)[32]<-"t_alt_count"
colnames(icgc2_allmuts_re)[31]<-"Tumor_Depth"
icgc2_allmuts_re$Mut_identifier<-paste0(icgc2_allmuts_re$Tumor_Sample_Barcode,"-",icgc2_allmuts_re$Start_Position,"-",icgc2_allmuts_re$aa_mutation,"-",icgc2_allmuts_re$t_alt_count) ##To Id AICDA mutations easier


colnames(icgc_aicda_maf_final_re)[32]<-"t_alt_count"
colnames(icgc_aicda_maf_final_re)[31]<-"Tumor_Depth"
icgc_aicda_maf_final_re$Mut_identifier<-paste0(icgc_aicda_maf_final_re$Tumor_Sample_Barcode,"-",icgc_aicda_maf_final_re$Start_Position,"-",icgc_aicda_maf_final_re$aa_mutation,"-",icgc_aicda_maf_final_re$t_alt_count)

icgc_aicda_maf_final_re$histology_abbreviation<-ICGC_clinial_rem2$histology_abbreviation[match(icgc_aicda_maf_final_re$Tumor_Sample_Barcode,ICGC_clinial_rem2$Tumor_Sample_Barcode)]##Add

icgc2_allmuts_re$histology_abbreviation<-ICGC_clinial_rem2$histology_abbreviation[match(icgc2_allmuts_re$Tumor_Sample_Barcode,ICGC_clinial_rem2$Tumor_Sample_Barcode)]##Add
##Add

icgc2_allmuts_re$AICDA_mutation<-ifelse(icgc2_allmuts_re$Mut_identifier %in% icgc_aicda_maf_final_re$Mut_identifier, "YES","NO")###Add a column next to each mutation to indicate if it is AICDA induced or not


table( icgc2_allmuts_re$AICDA_mutation)



###
##Add data to maf, which mutations are AICDA
icgc2_allmuts<-rbind(as.data.frame(icgc@data),as.data.frame(icgc@maf.silent)) ##Add data and maf_silent info
icgc2_allmuts<-subset(icgc2_allmuts,!(icgc2_allmuts$Project_Code %in% c("Skin-Melanoma")))#Take out Skin-Melanoma




icgc2_allmuts$Mut_identifier<-paste0(icgc2_allmuts$Tumor_Sample_Barcode,"-",icgc2_allmuts$Start_Position,"-",icgc2_allmuts$CONTEXT,"-",icgc2_allmuts$t_alt_count) ##To Id AICDA mutations easier

icgc_aicda_maf_final$Mut_identifier<-paste0(icgc_aicda_maf_final$Tumor_Sample_Barcode,"-",icgc_aicda_maf_final$Start_Position,"-",icgc_aicda_maf_final$CONTEXT,"-",icgc_aicda_maf_final$t_alt_count)

icgc_aicda_maf_final$histology_abbreviation<-ICGC_clinial$histology_abbreviation[match(icgc_aicda_maf_final$Tumor_Sample_Barcode,ICGC_clinial$Tumor_Sample_Barcode)]##Add


icgc2_allmuts$AICDA_mutation<-ifelse(icgc2_allmuts$Mut_identifier %in% icgc_aicda_maf_final$Mut_identifier, "YES","NO")###Add a column next to each mutation to indicate if it is AICDA induced or not

#Calculate Tumor_depth and write it to t_ref_count
colnames(icgc2_allmuts)[39]<-"Tumor_Depth"
icgc2_allmuts$Tumor_Depth<-icgc2_allmuts$Tumor_Depth+icgc2_allmuts$t_alt_count

table( icgc2_allmuts$AICDA_mutation)


####Merge ICGC_clinial (1880 samples ) and ICGC_clinial_rem2 for 1230 samples
##Needed colnames: c("icgc_sample_id","project_code","icgc_donor_id","histology_abbreviation","fraction_AICDA_mutations","Tumor_Sample_Barcode","AICDA_Enrichment","AICDA_mutations","fraction_APOBEC_mutations")
ICGC_clinial_n3110<-rbind(ICGC_clinial[,c("icgc_sample_id","project_code","icgc_donor_id","histology_abbreviation","fraction_AICDA_mutations","Tumor_Sample_Barcode","AICDA_Enrichment","AICDA_mutations","fraction_APOBEC_mutations")],ICGC_clinial_rem2[,c("icgc_sample_id","project_code","icgc_donor_id","histology_abbreviation","fraction_AICDA_mutations","Tumor_Sample_Barcode","AICDA_Enrichment","AICDA_mutations","fraction_APOBEC_mutations")])



```

```{r panel_1a,fig.height=12, fig.width=14}
library(ggpubr)
library(reshape2)
library(tidyr)
library(ggplot2)
library(reshape2)
library(dplyr)
library(tidyverse)
##Import data

data_dir<-"/media/user/seagate_ICM/AICDA_analysis/Github_repository/" #Set to where you downloaded the data
setwd(data_dir)
source('r/prerequisites.R')

##Import data
setwd(data_dir)
ICGC_clinial_n3110<-read.delim("Data/Figure1/ICGC_data_figure1a.tab")

###Graph to see distribution of AICDA mutations and APOBEC
#Melt
###Graph to see distribution of AICDA mutations and APOBEC
#Melt

###Firts merge together clinical_data....

t<-melt(ICGC_clinial_n3110[],id.vars=c('histology_abbreviation'), measure.vars=c('fraction_AICDA_mutations','fraction_APOBEC_mutations'))

#Add line global median, add n samples to labels...

icgc_p_fractionMuts<-ggboxplot(data = t, y ="value", x="histology_abbreviation",color = "variable",xlab = "Tumor type",ylab = "Fraction of SNP mutations",title = "ICGC study")+theme(axis.text.x = element_text(angle = 90,size=8),axis.title = element_text(face="bold"),plot.title = element_text(hjust = 0.5,face = "bold",size = 15))+ geom_hline(yintercept=median(ICGC_clinial_n3110$fraction_AICDA_mutations), linetype="dashed",color="#F8766D")+ geom_hline(yintercept=median(ICGC_clinial_n3110$fraction_APOBEC_mutations), linetype="dashed",color="#00BFC4")+ guides(color=guide_legend(title = "Mutation type"))+scale_color_discrete(labels=c(paste0("AICDA"," (median = ", round(median(ICGC_clinial_n3110$fraction_AICDA_mutations),3),")"),paste0("APOBEC"," (median = ", round(median(ICGC_clinial_n3110$fraction_APOBEC_mutations),3),")")))+scale_x_discrete(labels=paste0(unique(t$histology_abbreviation)," (n=", table(ICGC_clinial_n3110$histology_abbreviation)[unique(t$histology_abbreviation)],")"))

icgc_p_fractionMuts

```


***

\pagebreak


# Supplementary Figure 1a

```{r calculations_Supfig1a,echo=T,eval=F}
library(Palimpsest)
library(maftools)
###Extract non-AICDA mutational signatures....
#########
###############
####Get the signature contribution of every tumor type using also the AICDA_canonical signature
#Prepare "vcf" from maf_pancancer global mutations
# prepare data as input to dndscv package to detect gene drivers
###Add histology_abbreviation from clinical to have same "tumor types"


icgc2_allmuts$histology_abbreviation<-ICGC_clinial$histology_abbreviation[match(icgc2_allmuts$Tumor_Sample_Barcode,ICGC_clinial$Tumor_Sample_Barcode)]##Add


maf_drivers_icgc <- icgc2_allmuts[,c("Tumor_Sample_Barcode", "Variant_Type", "Chromosome", "Start_Position", "Reference_Allele", "Tumor_Seq_Allele2", "t_alt_count", "Hugo_Symbol","histology_abbreviation")]
signif_genes_icgc<-list() ;signif_genes_top_icgc<-list()                            
icgc2_allmuts$Driver<-0
icgc_tumorType<-unique((icgc2_allmuts$histology_abbreviation))

for (i in 1:length(icgc_tumorType)) {
  maf_drivers_tmp<-maf_drivers_icgc[maf_drivers_icgc$histology_abbreviation %in% icgc_tumorType[i],]
  # maf_drivers_tmp<-maf_drivers_tmp[maf_drivers_tmp$Variant_Type %in%"SNP",] #improves calculation
  ##Apply per tumor type   
  dndsout = dndscv(maf_drivers_tmp[,c(1,3,4,5,6)])
  # 
  sel_cv= dndsout$sel_cv
  signif_genes_icgc[[i]] = sel_cv[sel_cv$qglobal_cv<0.01, c("gene_name","qglobal_cv")]
  rownames(signif_genes_icgc[[i]]) = NULL
  
  signif_genes_top_icgc[[i]] <- signif_genes_icgc[[i]][1:25,1]
  
  # add driver feature in the maf file                  
  icgc2_allmuts$Driver[icgc2_allmuts$histology_abbreviation %in% icgc_tumorType[i]] <- ifelse( icgc2_allmuts$Hugo_Symbol[icgc2_allmuts$histology_abbreviation %in% icgc_tumorType[i]] %in% signif_genes_top_icgc[[i]], 1, 0)
  
}


##Myeloid-MDS could not be properly calculated, do for 1:20 and then 22:23

names(signif_genes_top_icgc)<-icgc_tumorType


#Change name of histology_abbreviation to Tumor_type, to make it simplier 
colnames(icgc2_allmuts)[47]<-"Tumor_type"

icgc2_allmuts_NO_AICDA<-icgc2_allmuts[icgc2_allmuts$AICDA_mutation=="NO",c("Tumor_Sample_Barcode", "Variant_Type", "Chromosome", "Start_Position", "Reference_Allele", "Tumor_Seq_Allele2", "t_alt_count", "Driver","Tumor_type","Hugo_Symbol")]    ##SUbset to not associated AICDA mutations and only data for palimpsest

icgc2_allmuts_NO_AICDA<-as.data.frame(icgc2_allmuts_NO_AICDA)
# New version suports SNP, DEL and INS so there is no need to subset only SNP
# load data from Palimpsest package, according to its path                  

colnames(icgc2_allmuts_NO_AICDA) <- c(colnames(vcf[c(1:7,10)]),"Tumor_type","Gene_Name")

icgc2_allmuts_NO_AICDA$Type <- ifelse(icgc2_allmuts_NO_AICDA$Type == "SNP", yes = "SNV", ifelse(icgc2_allmuts_NO_AICDA$Type == "DEL", yes = "DEL","INS"))


#Preparing input data for mutational signature analysis
##Annotate mutations
vcf_icgc2_allmuts_NO_AICDA<-annotate_VCF(vcf=icgc2_allmuts_NO_AICDA,ref_genome =BSgenome.Hsapiens.UCSC.hg19, ref_fasta=NULL , add_ID_cats=F,add_DBS_cats = F)
dim(vcf_icgc2_allmuts_NO_AICDA)






##Do a loop for each tumor type.....extracting a specific AICDA_signature per cancer type...
SBS_input_pancancer_NO_AICDA_icgc<-list(); SBS_signatures_exp_pancancer_NO_AICDA_icgc<-list()

exclude_sigs<-c("SBS27","SBS39","SBS43","SBS45" , "SBS46" , "SBS47" , "SBS48"  ,"SBS49" , "SBS50" , "SBS51" , "SBS52" , "SBS53" , "SBS54" , "SBS55","SBS56" , "SBS57" , "SBS58" , "SBS59" , "SBS60")#Signatures not to include to see if it helps the fitting,  18 are posible sequencing artefacts; SBS39 posible noise causing with AICDA signature



for (i in 1:24) {
SBS_cosmic_bytumor<-SBS_cosmic[setdiff(rownames(SBS_cosmic),exclude_sigs),]##l
##Plot cosine similarities
##Apply now to global maf  
  
vcf_icgc2_allmuts_NO_AICDA_tmp<-vcf_icgc2_allmuts_NO_AICDA[vcf_icgc2_allmuts_NO_AICDA$Tumor_type %in% icgc_tumorType[i],] #Subset by tumor type
SBS_input_pancancer_NO_AICDA_icgc[[i]]<-palimpsest_input(vcf = vcf_icgc2_allmuts_NO_AICDA_tmp, Type = "SBS") #Get tnm proportions and numbers by sample
  
resdir.2<-paste0("/media/user/seagate_ICM/AICDA_analysis/ICGC/mutational_signatures/",icgc_tumorType[i],"/");if(!file.exists(resdir.2)) dir.create(resdir.2) #Create directory if it does not exist
  
  
#sig_cols<-signature_colour_generator(rownames(SBS_cosmic_bytumor)) #Asign colors, this time used the previously defined colors
  
SBS_signatures_exp_pancancer_NO_AICDA_icgc[[i]] <- deconvolution_fit(input_vcf = vcf_icgc2_allmuts_NO_AICDA_tmp,input_matrices = SBS_input_pancancer_NO_AICDA_icgc[[i]],input_signatures = SBS_cosmic_bytumor,signature_colours = sig_cols,resdir = resdir.2,doplot = F,save_signatures_exp = F) #Calcules weight for each signature
select <- which(colMeans(SBS_signatures_exp_pancancer_NO_AICDA_icgc[[i]]$sig_nums) > 1) #Select signatures with real contribution
#sig_cols<-signature_colour_generator(names(select)), not neccesary, previously defined
  
##Re-fit using only selected signatures
#selected mutational known signatures in tumor[i] samples   
X<-as.character(names(select))
signatures3 <-SBS_cosmic_bytumor[X,]
  
SBS_signatures_exp_pancancer_NO_AICDA_icgc[[i]] <- deconvolution_fit(input_vcf = vcf_icgc2_allmuts_NO_AICDA_tmp,input_matrices = SBS_input_pancancer_NO_AICDA_icgc[[i]],input_signatures = signatures3,signature_colours = sig_cols,resdir = resdir.2,doplot = F,save_signatures_exp = T) #Calcules weight for each signature, re-fit to improve visualization
##now add a column for the AICDA_canonical using the previously obtain numbers, add at $sig_nums
SBS_signatures_exp_pancancer_NO_AICDA_icgc[[i]]$sig_nums$AICDA_canonical<-ICGC_clinial$AICDA_mutations[ICGC_clinial$Tumor_Sample_Barcode %in% rownames(SBS_signatures_exp_pancancer_NO_AICDA_icgc[[i]]$sig_nums)]

###Now recalculate each sig_props using the new mutations numbers
SBS_signatures_exp_pancancer_NO_AICDA_icgc[[i]]$sig_props$AICDA_canonical<-0
SBS_signatures_exp_pancancer_NO_AICDA_icgc[[i]]$sig_props<-round(SBS_signatures_exp_pancancer_NO_AICDA_icgc[[i]]$sig_nums[,]/rowSums(SBS_signatures_exp_pancancer_NO_AICDA_icgc[[i]]$sig_nums[,]),7)
}


names(SBS_signatures_exp_pancancer_NO_AICDA_icgc)<-icgc_tumorType[]

View(SBS_signatures_exp_pancancer_NO_AICDA_icgc)


############################################
###################SUbsecition to analyze remaining samples (n= 1230)
#########"
##
###





###Extract non-AICDA mutational signatures....

####Get the signature contribution of every tumor type using also the AICDA_canonical signature
#Prepare "vcf" from maf_pancancer global mutations
# prepare data as input to dndscv package to detect gene drivers
###Add histology_abbreviation from clinical to have same "tumor types"



maf_drivers_icgc_re <- icgc2_allmuts_re[,c("Tumor_Sample_Barcode", "Variant_Type", "Chromosome", "Start_Position", "Reference_Allele", "Tumor_Seq_Allele2", "t_alt_count", "Hugo_Symbol","histology_abbreviation")]
signif_genes_icgc_re<-list() ;signif_genes_top_icgc_re<-list()                            
icgc2_allmuts_re$Driver<-0
icgc_tumorType2<-unique((icgc2_allmuts_re$histology_abbreviation))

for (i in 24:length(icgc_tumorType2)) {
  maf_drivers_icgc_tmp<-maf_drivers_icgc_re[maf_drivers_icgc_re$histology_abbreviation %in% icgc_tumorType2[i],]
  # maf_drivers_tmp<-maf_drivers_tmp[maf_drivers_tmp$Variant_Type %in%"SNP",] #improves calculation
  ##Apply per tumor type   
  dndsout = dndscv(maf_drivers_icgc_tmp[,c(1,3,4,5,6)])
  # 
  sel_cv= dndsout$sel_cv
  signif_genes_icgc_re[[i]] = sel_cv[sel_cv$qglobal_cv<0.01, c("gene_name","qglobal_cv")]
  rownames(signif_genes_icgc_re[[i]]) = NULL
  
  signif_genes_top_icgc_re[[i]] <- signif_genes_icgc_re[[i]][1:25,1]
  
  # add driver feature in the maf file                  
  icgc2_allmuts_re$Driver[icgc2_allmuts_re$histology_abbreviation %in% icgc_tumorType2[i]] <- ifelse( icgc2_allmuts_re$Hugo_Symbol[icgc2_allmuts_re$histology_abbreviation %in% icgc_tumorType2[i]] %in% signif_genes_top_icgc_re[[i]], 1, 0)
  
}


##Myeloid-AML and Panc-Endocrine could not be properly calculated, do for 1:22 and then 24:26

names(signif_genes_top_icgc_re)<-icgc_tumorType2[1:25]


#Change name of histology_abbreviation to Tumor_type, to make it simplier 
colnames(icgc2_allmuts_re)[51]<-"Tumor_type"





icgc2_allmuts_NO_AICDA_re<-icgc2_allmuts_re[icgc2_allmuts_re$AICDA_mutation=="NO",c("Tumor_Sample_Barcode", "Variant_Type", "Chromosome", "Start_Position", "Reference_Allele", "Tumor_Seq_Allele2", "t_alt_count", "Driver","Tumor_type","Hugo_Symbol")]    ##SUbset to not associated AICDA mutations and only data for palimpsest

icgc2_allmuts_NO_AICDA_re<-as.data.frame(icgc2_allmuts_NO_AICDA_re)
# New version suports SNP, DEL and INS so there is no need to subset only SNP
# load data from Palimpsest package, according to its path                  


colnames(icgc2_allmuts_NO_AICDA_re) <- c(colnames(vcf[c(1:7,10)]),"Tumor_type","Gene_Name")

icgc2_allmuts_NO_AICDA_re$Type <- ifelse(icgc2_allmuts_NO_AICDA_re$Type == "SNP", yes = "SNV", ifelse(icgc2_allmuts_NO_AICDA_re$Type == "DEL", yes = "DEL","INS"))

icgc2_allmuts_NO_AICDA_re<-as.data.frame(icgc2_allmuts_NO_AICDA_re)
icgc2_allmuts_NO_AICDA_re$Sample<-as.character(icgc2_allmuts_NO_AICDA_re$Sample)
#Preparing input data for mutational signature analysis
##Annotate mutations
##Eliminate NA in Type
icgc2_allmuts_NO_AICDA_re<-icgc2_allmuts_NO_AICDA_re[!is.na(icgc2_allmuts_NO_AICDA_re$Type),]
##Take out wrongly format variants in which REF==ALT
icgc2_allmuts_NO_AICDA_re<-icgc2_allmuts_NO_AICDA_re[icgc2_allmuts_NO_AICDA_re$REF!=icgc2_allmuts_NO_AICDA_re$ALT,]

vcf_icgc2_allmuts_NO_AICDA_re<-annotate_VCF(vcf=icgc2_allmuts_NO_AICDA_re,ref_genome =BSgenome.Hsapiens.UCSC.hg19, ref_fasta=NULL , add_ID_cats=F,add_DBS_cats = F)
dim(vcf_icgc2_allmuts_NO_AICDA_re)

####Do the same on the icgc2_allmuts_re to have the same mutations as the annotated vcf

icgc2_allmuts_re<-icgc2_allmuts_re[!is.na(icgc2_allmuts_re$Variant_Type)&icgc2_allmuts_re$Reference_Allele!=icgc2_allmuts_re$Tumor_Seq_Allele2,]

##Do a loop for each tumor type.....extracting a specific AICDA_signature per cancer type...
SBS_input_pancancer_NO_AICDA_icgc_re<-list(); SBS_signatures_exp_pancancer_NO_AICDA_icgc_re <-list()

exclude_sigs<-c("SBS27","SBS39","SBS43","SBS45" , "SBS46" , "SBS47" , "SBS48"  ,"SBS49" , "SBS50" , "SBS51" , "SBS52" , "SBS53" , "SBS54" , "SBS55","SBS56" , "SBS57" , "SBS58" , "SBS59" , "SBS60")#Signatures not to include to see if it helps the fitting,  18 are posible sequencing artefacts; SBS39 posible noise causing with AICDA signature

###########################"
#######################"
####Edition-merge zone

###
icgc_tumorType<-unique(ICGC_clinial_n3110$histology_abbreviation)

for (i in 1:length(icgc_tumorType)) {
SBS_cosmic_bytumor<-SBS_cosmic[setdiff(rownames(SBS_cosmic),exclude_sigs),]##l
##Plot cosine similarities
##Apply now to global maf  
  
vcf_icgc2_allmuts_NO_AICDA_tmp<-vcf_icgc2_allmuts_NO_AICDA[vcf_icgc2_allmuts_NO_AICDA$Tumor_type %in% icgc_tumorType[i],] #Subset by tumor type
SBS_input_pancancer_NO_AICDA_icgc[[i]]<-palimpsest_input(vcf = vcf_icgc2_allmuts_NO_AICDA_tmp, Type = "SBS") #Get tnm proportions and numbers by sample
  
resdir.2<-paste0("/media/user/seagate_ICM/AICDA_analysis/ICGC/mutational_signatures/",icgc_tumorType[i],"/");if(!file.exists(resdir.2)) dir.create(resdir.2) #Create directory if it does not exist
load("r/Mutational_Signature_colors.RData") #Color for signatures
SBS_signatures_exp_pancancer_NO_AICDA_icgc[[i]] <- deconvolution_fit(input_vcf = vcf_icgc2_allmuts_NO_AICDA_tmp,input_matrices = SBS_input_pancancer_NO_AICDA_icgc[[i]],input_signatures = SBS_cosmic_bytumor,signature_colours = sig_cols,resdir = resdir.2,doplot = F,save_signatures_exp = F) #Calcules weight for each signature
select <- which(colMeans(SBS_signatures_exp_pancancer_NO_AICDA_icgc[[i]]$sig_nums) > 1) #Select signatures with real contribution
#sig_cols<-signature_colour_generator(names(select)), not neccesary, previously defined
  
##Re-fit using only selected signatures
#selected mutational known signatures in tumor[i] samples   
X<-as.character(names(select))
signatures3 <-SBS_cosmic_bytumor[X,]
  
SBS_signatures_exp_pancancer_NO_AICDA_icgc[[i]] <- deconvolution_fit(input_vcf = vcf_icgc2_allmuts_NO_AICDA_tmp,input_matrices = SBS_input_pancancer_NO_AICDA_icgc[[i]],input_signatures = signatures3,signature_colours = sig_cols,resdir = resdir.2,doplot = F,save_signatures_exp = T) #Calcules weight for each signature, re-fit to improve visualization
##now add a column for the AICDA_canonical using the previously obtain numbers, add at $sig_nums
SBS_signatures_exp_pancancer_NO_AICDA_icgc[[i]]$sig_nums$AICDA_canonical<-ICGC_clinial_n3110$AICDA_mutations[ICGC_clinial_n3110$Tumor_Sample_Barcode %in% rownames(SBS_signatures_exp_pancancer_NO_AICDA_icgc[[i]]$sig_nums)]

###Now recalculate each sig_props using the new mutations numbers
SBS_signatures_exp_pancancer_NO_AICDA_icgc[[i]]$sig_props$AICDA_canonical<-0
SBS_signatures_exp_pancancer_NO_AICDA_icgc[[i]]$sig_props<-round(SBS_signatures_exp_pancancer_NO_AICDA_icgc[[i]]$sig_nums[,]/rowSums(SBS_signatures_exp_pancancer_NO_AICDA_icgc[[i]]$sig_nums[,]),7)
}


names(SBS_signatures_exp_pancancer_NO_AICDA_icgc)<-icgc_tumorType[]


#######
###
###Summary graph of signatures per tumor type, using corrplot + other packages
##Get median signature weight by tumor for each signature (this is going to be represented by the circle size)

##Get median signature weight by tumor for each signature (this is going to be represented by the circle size)
signature_proportions_semi_icgc<-as.data.frame(x = matrix(data=0,nrow = 35,ncol = length(sig_cols),dimnames  = list(icgc_tumorType,names(sig_cols)))) #Create empty data frame to fill
signature_mutations_median_semi_icgc<-as.data.frame(x = matrix(data=0,nrow = 35,ncol = length(sig_cols),dimnames  = list(icgc_tumorType,names(sig_cols))))
for (i in 1:35) {
    count_tmp<-colSums(SBS_signatures_exp_pancancer_NO_AICDA_icgc[[i]]$sig_nums[,])/sum(SBS_signatures_exp_pancancer_NO_AICDA_icgc[[i]]$sig_nums)
  
    signature_proportions_semi_icgc[i,match(names(count_tmp),colnames(signature_proportions_semi_icgc))]<-count_tmp
    
    count_tmp<-SBS_signatures_exp_pancancer_NO_AICDA_icgc[[i]]$sig_nums
    for (j in 1:ncol(count_tmp)) {
        count_tmp2<-count_tmp[,j]
        count_tmp2<- count_tmp2[count_tmp2>0]
        signature_mutations_median_semi_icgc[i,match(names(count_tmp)[j],colnames(signature_mutations_median_semi_icgc))]<-median(count_tmp2)/1000 #To calculate the TMB per megabase, the total number of mutations counted is divided by the size of the coding region of the targeted territory; We used 38 Mb as the estimate of the exome size for WES.https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6350758/
        #In this case since it is WGS and we are considering all genes (coding and noCoding) it is better to use Mutational Load/1000 instead
    }
    
}




#Change a little the signatures order to improve analysis 
signature_proportions_semi_icgc2<-signature_proportions_semi_icgc[,c(1:11,47,12,13:46)] #Reorder to put AICDA next to SBS9
signature_mutations_median_semi_icgc2<-signature_mutations_median_semi_icgc[,c(1:11,47,12,13:46)]
rownames(signature_proportions_semi_icgc2)<-paste0(rownames(signature_proportions_semi_icgc2)," (n = ", total_n_icgc, " )")
signature_mutations_median_semi_icgc2[is.na(signature_mutations_median_semi_icgc2)]<-0#Change NA to 0
signature_mutations_median_semi_icgc2<-signature_mutations_median_semi_icgc2[,colnames(signature_mutations_median_semi_icgc2[colSums(signature_mutations_median_semi_icgc2)>0])]#Eliminate signatures not present in any tumor type

signature_proportions_semi_icgc2<-signature_proportions_semi_icgc2[,colnames(signature_mutations_median_semi_icgc2)]
library(corrplot)


  
```

```{r panel_sup1a_cals, eval=T,echo=T,fig.show="hide",results="hide"}
##LOad manual function for corplot 
data_dir<-"/media/user/seagate_ICM/AICDA_analysis/Github_repository/" #Set to where you downloaded the data
setwd(data_dir)
load("r/corrplot_circleSizeFunction.RData")
load("r/Mutational_Signature_colors.RData") #Color for signatures


signature_proportions_semi_icgc2<-as.data.frame(read.delim("Data/Figure1/ICGC_data_Sup_figure1a.tab"))
signature_mutations_median_semi_icgc2<-as.data.frame(read.delim("Data/Figure1/ICGC_data_Sup_figure1a_median.tab"))

##
signature_proportions_semi_icgc3<-signature_proportions_semi_icgc2#To set blanks where there is no signature signal
signature_proportions_semi_icgc3<-ifelse(signature_proportions_semi_icgc3==0, 1,0.000)

#display.brewer.all() to check color list
#708090
#For background colors where there are values
ColorScheme = c("ivory1", "ivory2")
Groups = c(rep(1,35), rep(2,35))
#) take out legend

#For background colors where there are values where there are values 0 we have to refill according to previous background color

ColorScheme_insignificant<-ifelse(signature_proportions_semi_icgc3==1, 1,NA)
ColorScheme_insignificant[,seq(2,ncol(ColorScheme_insignificant),by=2)]<-ifelse(ColorScheme_insignificant[,seq(2,ncol(ColorScheme_insignificant),by=2)]==1, "ivory2",NA)
ColorScheme_insignificant[,seq(1,ncol(ColorScheme_insignificant),by=2)]<-ifelse(ColorScheme_insignificant[,seq(1,ncol(ColorScheme_insignificant),by=2)]==1, "ivory1",NA)

ColorScheme_insignificant<-ColorScheme_insignificant[!is.na(ColorScheme_insignificant)]

###MUltiple size vector by a factor to increase circle size (since max number is .508)
corrplot_circleSize(is.corr = FALSE,corr = as.matrix(signature_mutations_median_semi_icgc2),na.label = " ",col = c(brewer.pal(n = 10, name = "Set3"),brewer.pal(n = 10, name = "Paired")), tl.col = "black", order = "original", tl.cex = 0.6,size_vector= as.numeric(as.matrix(signature_proportions_semi_icgc2[,])*1.4),method = "circle", cl.pos = "n",p.mat = signature_proportions_semi_icgc3, insig = "blank",bg = ColorScheme[Groups],bg_insig  = ColorScheme_insignificant,mar = c(0,1,0,0),at = c(0,0.05,.10,.25,.5,1,2.5,5,10,25),color_at = c("white",brewer.pal(n = 9, name = "Paired")))


##Add manual rectangles
segments(c(11.5,13.5), rep(0.5,2), c(11.5,13.5), rep(35.5,2), lwd=3) #move 1 in x per box, vertical lines (x0,y0,x1,y1)


segments(c(11.5, 11.5), c(0.5, 35.5), c(13.5, 13.5), c(0.5,35.5), lwd=3) #Horizontal lines

#Obtional, add description to each SBS text(x = 1, y = 26.2, "(H", srt = 90,cex=0.6) 

#Add text

#Grab

lgd2 = Legend(at = rev(c(0,0.05,.10,.25,.5,1,2.5,5,10,25)), labels = paste0(rev(c(0,0.05,.10,.25,.5,1,2.5,5,10,25)), " k"),legend_gp = gpar(fill = rev(c("white",brewer.pal(n = 9, name = "Paired")))), 
              title = "Median k mutations load due signature \n(among tumors with signature)", border = "black",title_gp = gpar(col = "black", fontsize = 10,fontface="bold"),ncol = 1, title_position = "leftcenter-rot",grid_height = unit(9, "mm"), grid_width = unit(9, "mm"))
## draw it, changes optional (Legend for circle colors)

lgd3 = Legend( title = "Signature weight contribution \nwithin tumor type", type = "points", at =round(seq(0,.67,by=.074),2),title_gp = gpar(col = "black", fontsize = 10,fontface="bold"), pch = 19, legend_gp = gpar(col = "ivory4",cex=round(seq(0,.67,by=.074),2)),size =unit(c(0,seq(1.2,4.4,by=.4)*2.4), "mm") ,ncol=1,border = "white",title_position = "leftcenter-rot",grid_height = unit(9, "mm"), grid_width = unit(9, "mm")) ## draw it, changes optional (Legend for sizes)
## draw it, changes optional (Legend for sizes)  

pd = packLegend( lgd2, lgd3, direction = "vertical",row_gap = unit(.8, "cm"))
pushViewport(viewport(  y = 0.358,x=0.05,height = .1,width =.1,just = c("bottom")))

grid.draw(pd)
p_pann_summary_signatures_semideconv_icgc<-recordPlot()

##Save 16*20

p_pann_summary_signatures_semideconv_icgc

```

```{r panel_1C,fig.height=16, fig.width=20}
p_pann_summary_signatures_semideconv_icgc
```


***

\pagebreak
